<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Example-unique analyses | MNet</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.33 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Example-unique analyses | MNet" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Example-unique analyses | MNet" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Tuantuan Gui and Hai Fang" />


<meta name="date" content="2023-04-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="example-rountine-analyses.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.27/datatables.js"></script>
<link href="libs/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.12.1/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-buttons-1.12.1/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-buttons-1.12.1/js/dataTables.buttons.min.js"></script>
<script src="libs/dt-ext-buttons-1.12.1/js/buttons.html5.min.js"></script>
<script src="libs/dt-ext-buttons-1.12.1/js/buttons.colVis.min.js"></script>
<script src="libs/dt-ext-buttons-1.12.1/js/buttons.print.min.js"></script>
<link href="libs/dt-ext-fixedcolumns-1.12.1/css/fixedColumns.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedcolumns-1.12.1/js/dataTables.fixedColumns.min.js"></script>
<link href="libs/dt-plugin-searchhighlight-1.12.1/source.css" rel="stylesheet" />
<script src="libs/dt-plugin-searchhighlight-1.12.1/jquery.highlight.js"></script>
<script src="libs/dt-plugin-searchhighlight-1.12.1/source.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction &amp; Installation</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#overview-of-workflow"><i class="fa fa-check"></i><b>1.1</b> Overview of Workflow</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.2</b> Installation</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html"><i class="fa fa-check"></i><b>2</b> Example-unique analyses</a>
<ul>
<li class="chapter" data-level="2.1" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#loading-the-data"><i class="fa fa-check"></i><b>2.1</b> Loading the data</a></li>
<li class="chapter" data-level="2.2" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#pathway-enrichment-analysis"><i class="fa fa-check"></i><b>2.2</b> Pathway enrichment analysis</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#extended-pathway-enrichment"><i class="fa fa-check"></i><b>2.2.1</b> Extended pathway enrichment</a></li>
<li class="chapter" data-level="2.2.2" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#metabolite-pathway-enrichment"><i class="fa fa-check"></i><b>2.2.2</b> metabolite pathway enrichment</a></li>
<li class="chapter" data-level="2.2.3" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#gene-pathway-enrichment"><i class="fa fa-check"></i><b>2.2.3</b> gene pathway enrichment</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#pathway-differential-abundance-score"><i class="fa fa-check"></i><b>2.3</b> Pathway differential abundance score</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#extended-pathway-differential-abundance-score"><i class="fa fa-check"></i><b>2.3.1</b> Extended pathway differential abundance score</a></li>
<li class="chapter" data-level="2.3.2" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#metabolite-differential-abundance-score"><i class="fa fa-check"></i><b>2.3.2</b> metabolite differential abundance score</a></li>
<li class="chapter" data-level="2.3.3" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#gene-differential-abundance-score"><i class="fa fa-check"></i><b>2.3.3</b> gene differential abundance score</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#msea-metabolite-set-enrichment-analysis"><i class="fa fa-check"></i><b>2.4</b> MSEA Metabolite Set Enrichment Analysis</a></li>
<li class="chapter" data-level="2.5" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#subnetwork-analysis-of-metabolism-and-transcriptome-data-from-tnbc"><i class="fa fa-check"></i><b>2.5</b> Subnetwork analysis of Metabolism and transcriptome data from TNBC:</a></li>
<li class="chapter" data-level="2.6" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#subnetwork-analysis-of-interested-metabolites-and-genes"><i class="fa fa-check"></i><b>2.6</b> Subnetwork analysis of interested metabolites and genes</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html"><i class="fa fa-check"></i><b>3</b> Example-rountine analyses</a>
<ul>
<li class="chapter" data-level="3.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-and-kegg-name-change"><i class="fa fa-check"></i><b>3.1</b> Metabolite and kegg name change</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-name-changed-to-refmet-name"><i class="fa fa-check"></i><b>3.1.1</b> Metabolite name changed to refmet name</a></li>
<li class="chapter" data-level="3.1.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-name-changed-to-kegg-id"><i class="fa fa-check"></i><b>3.1.2</b> Metabolite name changed to kegg id</a></li>
<li class="chapter" data-level="3.1.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-name-corresponding-to-kegg-pathway"><i class="fa fa-check"></i><b>3.1.3</b> Metabolite name corresponding to kegg pathway</a></li>
<li class="chapter" data-level="3.1.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-keggid-to-kegg-pathway"><i class="fa fa-check"></i><b>3.1.4</b> Metabolite keggid to kegg pathway</a></li>
<li class="chapter" data-level="3.1.5" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#the-pathway-information"><i class="fa fa-check"></i><b>3.1.5</b> the pathway information</a></li>
<li class="chapter" data-level="3.1.6" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathway-name-to-pathway-id"><i class="fa fa-check"></i><b>3.1.6</b> pathway name to pathway id</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#group-wise-analyses"><i class="fa fa-check"></i><b>3.2</b> Group-wise analyses</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pca-plot"><i class="fa fa-check"></i><b>3.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="3.2.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differnetial-metabolite-analysis"><i class="fa fa-check"></i><b>3.2.2</b> Differnetial metabolite analysis</a></li>
<li class="chapter" data-level="3.2.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differential-metabolites-volcano"><i class="fa fa-check"></i><b>3.2.3</b> Differential metabolites’ volcano</a></li>
<li class="chapter" data-level="3.2.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differential-metabolites-heatmap"><i class="fa fa-check"></i><b>3.2.4</b> Differential metabolites’ heatmap</a></li>
<li class="chapter" data-level="3.2.5" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differential-metabolites-zscore"><i class="fa fa-check"></i><b>3.2.5</b> Differential metabolites’ zscore</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#feature-selection"><i class="fa fa-check"></i><b>3.3</b> Feature selection</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#boruta"><i class="fa fa-check"></i><b>3.3.1</b> Boruta</a></li>
<li class="chapter" data-level="3.3.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#random-forest"><i class="fa fa-check"></i><b>3.3.2</b> Random Forest</a></li>
<li class="chapter" data-level="3.3.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#xgboost"><i class="fa fa-check"></i><b>3.3.3</b> XGBoost</a></li>
<li class="chapter" data-level="3.3.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#lasso"><i class="fa fa-check"></i><b>3.3.4</b> LASSO</a></li>
<li class="chapter" data-level="3.3.5" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#elastic-network"><i class="fa fa-check"></i><b>3.3.5</b> elastic network</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathway-analysis"><i class="fa fa-check"></i><b>3.4</b> Pathway analysis</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathview-only-metabolite"><i class="fa fa-check"></i><b>3.4.1</b> pathview only metabolite</a></li>
<li class="chapter" data-level="3.4.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathview-metabolite-and-gene"><i class="fa fa-check"></i><b>3.4.2</b> pathview metabolite and gene</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#clinical-analysis"><i class="fa fa-check"></i><b>3.5</b> Clinical analysis</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#time-series-of-clinical"><i class="fa fa-check"></i><b>3.5.1</b> Time series of clinical</a></li>
<li class="chapter" data-level="3.5.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#survival-analysis"><i class="fa fa-check"></i><b>3.5.2</b> Survival analysis</a></li>
<li class="chapter" data-level="3.5.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolites-survival-plot"><i class="fa fa-check"></i><b>3.5.3</b> Metabolites’ survival plot</a></li>
<li class="chapter" data-level="3.5.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#cox-analysis"><i class="fa fa-check"></i><b>3.5.4</b> Cox analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>4</b> Session information</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">MNet</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="example-unique-analyses" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Example-unique analyses<a href="example-unique-analyses.html#example-unique-analyses" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="loading-the-data" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Loading the data<a href="example-unique-analyses.html#loading-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The case study is an integrated metabolic and transcriptional analysis to understand the rapidly proliferating breast cancers (Liao, et al., 2022). In this paper, the authors characterized the metabolomic profile of two types of 31 breast tumors in human to uncovered pyrimidine metabolism and glutaminolysis pathways are elevated in TNBC. As an example of mixed metabolites and genes analyses, here we are going to use both the dataset of metabolite and gene expression data as reported in the paper. The metabolomics data was accessed from the Supplementary Tables S2 in the paper, and the gene expression data is from the author. They found that pyrimidine and purine metabolism, carbon metabolism, and the energy production related pathways were highly enriched in Cluster 2 while many lipid/fatty acid metabolism related pathways were enriched in Cluster 1.
We will use MNet to identify a set of features whose activity changes significantly between the two sample clusters.This result will hopefully hint at some specific biological activities that are pathologically altered in tumoral samples.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="example-unique-analyses.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb4-2"><a href="example-unique-analyses.html#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MNet)</span>
<span id="cb4-3"><a href="example-unique-analyses.html#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb4-4"><a href="example-unique-analyses.html#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb4-5"><a href="example-unique-analyses.html#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb4-6"><a href="example-unique-analyses.html#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="example-unique-analyses.html#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co">#load(&quot;data/meta_dat.rda&quot;)</span></span>
<span id="cb4-8"><a href="example-unique-analyses.html#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co">#load(&quot;data/gene_dat.rda&quot;)</span></span>
<span id="cb4-9"><a href="example-unique-analyses.html#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co">#load(&quot;data/group.rda&quot;)</span></span>
<span id="cb4-10"><a href="example-unique-analyses.html#cb4-10" aria-hidden="true" tabindex="-1"></a>diff_meta <span class="ot">&lt;-</span> <span class="fu">mlimma</span>(meta_dat,group)</span>
<span id="cb4-11"><a href="example-unique-analyses.html#cb4-11" aria-hidden="true" tabindex="-1"></a>diff_gene <span class="ot">&lt;-</span> <span class="fu">mlimma</span>(gene_dat,group)</span>
<span id="cb4-12"><a href="example-unique-analyses.html#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(diff_meta)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 8
##   logFC AveExpr     t  P.Value    adj.P.Val     B  logP name  
##   &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;        &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt; 
## 1  2.86    22.6  9.25 1.52e-10 0.0000000332 14.1   7.48 C02045
## 2  2.44    26.2  7.80 6.83e- 9 0.000000748  10.4   6.13 C00267
## 3 -1.82    27.1 -6.80 1.10e- 7 0.00000622    7.64  5.21 C00073
## 4 -3.78    20.9 -6.79 1.14e- 7 0.00000622    7.61  5.21 C05674
## 5 -2.20    21.4 -6.58 2.07e- 7 0.00000907    7.02  5.04 C00255
## 6 -2.37    21.6 -6.45 2.98e- 7 0.0000109     6.66  4.96 C00242</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="example-unique-analyses.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(diff_gene)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 8
##    logFC AveExpr      t  P.Value    adj.P.Val     B  logP name  
##    &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;        &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt; 
## 1   6.92   23.1   11.3  1.15e-12 0.0000000203  17.7  7.69 APH1B 
## 2  21.4    23.4   10.6  5.13e-12 0.0000000455  16.5  7.34 GFRA1 
## 3  -5.76   24.4  -10.1  1.85e-11 0.0000000891  15.4  7.05 RFC4  
## 4  10.5    14.8   10.1  2.01e-11 0.0000000891  15.3  7.05 FAM47E
## 5 -10.8    20.6   -9.79 3.85e-11 0.000000137   14.7  6.86 ORC6  
## 6   4.02    5.65   9.69 4.98e-11 0.000000147   14.5  6.83 FAM87B</code></pre>
</div>
<div id="pathway-enrichment-analysis" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Pathway enrichment analysis<a href="example-unique-analyses.html#pathway-enrichment-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="extended-pathway-enrichment" class="section level3 hasAnchor" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Extended pathway enrichment<a href="example-unique-analyses.html#extended-pathway-enrichment" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>KEGG metabolic pathway-based extended pathway enrichment (EPE) analysis to investigate the dysregulation of metabolic pathways considering both metabolites and genes using the function ‘PathwayAnalysis’.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="example-unique-analyses.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="do">## combine the metabolite information and gene information</span></span>
<span id="cb8-2"><a href="example-unique-analyses.html#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="example-unique-analyses.html#cb8-3" aria-hidden="true" tabindex="-1"></a>all_dat <span class="ot">&lt;-</span> <span class="fu">rbind</span>(diff_gene,diff_meta)</span>
<span id="cb8-4"><a href="example-unique-analyses.html#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="example-unique-analyses.html#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="do">## filter the increase differential genes and metabolites</span></span>
<span id="cb8-6"><a href="example-unique-analyses.html#cb8-6" aria-hidden="true" tabindex="-1"></a>all_dat1 <span class="ot">&lt;-</span> all_dat <span class="sc">%&gt;%</span></span>
<span id="cb8-7"><a href="example-unique-analyses.html#cb8-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC<span class="sc">&gt;</span><span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-8"><a href="example-unique-analyses.html#cb8-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb8-9"><a href="example-unique-analyses.html#cb8-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-10"><a href="example-unique-analyses.html#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="do">## the increase pathway analysis of both genes and metabolites</span></span>
<span id="cb8-11"><a href="example-unique-analyses.html#cb8-11" aria-hidden="true" tabindex="-1"></a>result1 <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(all_dat1<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;Extended&quot;</span>,<span class="at">p_cutoff=</span><span class="fl">0.01</span>)</span>
<span id="cb8-12"><a href="example-unique-analyses.html#cb8-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-13"><a href="example-unique-analyses.html#cb8-13" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result_v0131/EPE1.png&quot;</span>,result1<span class="sc">$</span>gp,<span class="at">width=</span><span class="dv">30</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb8-14"><a href="example-unique-analyses.html#cb8-14" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result_v0131/EPE_gene_meta_cluster1.png&quot;</span>,result1<span class="sc">$</span>p_barplot,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb8-15"><a href="example-unique-analyses.html#cb8-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-16"><a href="example-unique-analyses.html#cb8-16" aria-hidden="true" tabindex="-1"></a><span class="do">## filter the decrease differential genes and metabolites</span></span>
<span id="cb8-17"><a href="example-unique-analyses.html#cb8-17" aria-hidden="true" tabindex="-1"></a>all_dat2 <span class="ot">&lt;-</span> all_dat <span class="sc">%&gt;%</span></span>
<span id="cb8-18"><a href="example-unique-analyses.html#cb8-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-19"><a href="example-unique-analyses.html#cb8-19" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb8-20"><a href="example-unique-analyses.html#cb8-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-21"><a href="example-unique-analyses.html#cb8-21" aria-hidden="true" tabindex="-1"></a>result2 <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(all_dat2<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;Extended&quot;</span>,<span class="at">p_cutoff=</span><span class="fl">0.01</span>)</span>
<span id="cb8-22"><a href="example-unique-analyses.html#cb8-22" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result_v0131/EPE2.png&quot;</span>,result2<span class="sc">$</span>gp,<span class="at">width=</span><span class="dv">30</span>,<span class="at">height=</span><span class="dv">5</span>)</span>
<span id="cb8-23"><a href="example-unique-analyses.html#cb8-23" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result_v0131/EPE_gene_meta_cluster2.png&quot;</span>,result2<span class="sc">$</span>p_barplot,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb8-24"><a href="example-unique-analyses.html#cb8-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-25"><a href="example-unique-analyses.html#cb8-25" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> <span class="fu">list</span>(result1<span class="sc">$</span>p_barplot,result2<span class="sc">$</span>p_barplot))</span>
<span id="cb8-26"><a href="example-unique-analyses.html#cb8-26" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result_v0131/EPE_gene_meta.png&quot;</span>,p,<span class="at">width=</span><span class="dv">20</span>,<span class="at">height =</span> <span class="dv">10</span>)</span></code></pre></div>
<p><img src="result_v0131/EPE_gene_meta.png" />
<img src="result_v0131/EPE1.png" />
<img src="result_v0131/EPE2.png" /></p>
</div>
<div id="metabolite-pathway-enrichment" class="section level3 hasAnchor" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> metabolite pathway enrichment<a href="example-unique-analyses.html#metabolite-pathway-enrichment" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>KEGG metabolic pathway-based pathway enrichment (PE) analysis to investigate the dysregulation of metabolic pathways considering metabolites.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="example-unique-analyses.html#cb9-1" aria-hidden="true" tabindex="-1"></a>all_dat <span class="ot">&lt;-</span> diff_meta</span>
<span id="cb9-2"><a href="example-unique-analyses.html#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="example-unique-analyses.html#cb9-3" aria-hidden="true" tabindex="-1"></a>all_dat1 <span class="ot">&lt;-</span> all_dat <span class="sc">%&gt;%</span></span>
<span id="cb9-4"><a href="example-unique-analyses.html#cb9-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC<span class="sc">&gt;</span><span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-5"><a href="example-unique-analyses.html#cb9-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb9-6"><a href="example-unique-analyses.html#cb9-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-7"><a href="example-unique-analyses.html#cb9-7" aria-hidden="true" tabindex="-1"></a>result1 <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(all_dat1<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;metabolite&quot;</span>,<span class="at">p_cutoff=</span><span class="fl">0.1</span>)</span>
<span id="cb9-8"><a href="example-unique-analyses.html#cb9-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-9"><a href="example-unique-analyses.html#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result_v0131/PE_meta1.png&quot;</span>,result1<span class="sc">$</span>gp,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb9-10"><a href="example-unique-analyses.html#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(result1<span class="sc">$</span>output,<span class="st">&quot;result_v0131/PE_meta1.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb9-11"><a href="example-unique-analyses.html#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result_v0131/PE_meta_cluster1.png&quot;</span>,result1<span class="sc">$</span>p_barplot,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb9-12"><a href="example-unique-analyses.html#cb9-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-13"><a href="example-unique-analyses.html#cb9-13" aria-hidden="true" tabindex="-1"></a>all_dat2 <span class="ot">&lt;-</span> all_dat <span class="sc">%&gt;%</span></span>
<span id="cb9-14"><a href="example-unique-analyses.html#cb9-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-15"><a href="example-unique-analyses.html#cb9-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb9-16"><a href="example-unique-analyses.html#cb9-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-17"><a href="example-unique-analyses.html#cb9-17" aria-hidden="true" tabindex="-1"></a>result2 <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(all_dat2<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;metabolite&quot;</span>,<span class="at">p_cutoff=</span><span class="fl">0.1</span>)</span>
<span id="cb9-18"><a href="example-unique-analyses.html#cb9-18" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(result2<span class="sc">$</span>output,<span class="st">&quot;result_v0131/PE_gene2.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb9-19"><a href="example-unique-analyses.html#cb9-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-20"><a href="example-unique-analyses.html#cb9-20" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result_v0131/PE_meta2.png&quot;</span>,result2<span class="sc">$</span>gp,<span class="at">width=</span><span class="dv">15</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb9-21"><a href="example-unique-analyses.html#cb9-21" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result_v0131/PE_meta_cluster2.png&quot;</span>,result2<span class="sc">$</span>p_barplot,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb9-22"><a href="example-unique-analyses.html#cb9-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-23"><a href="example-unique-analyses.html#cb9-23" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> <span class="fu">list</span>(result1<span class="sc">$</span>p_barplot,result2<span class="sc">$</span>p_barplot))</span>
<span id="cb9-24"><a href="example-unique-analyses.html#cb9-24" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result_v0131/PE_meta.png&quot;</span>,p,<span class="at">width=</span><span class="dv">20</span>,<span class="at">height =</span> <span class="dv">10</span>)</span></code></pre></div>
<p><img src="result_v0131/PE_meta.png" />
<img src="result_v0131/PE_meta1.png" /></p>
<p><img src="result_v0131/PE_meta2.png" /></p>
</div>
<div id="gene-pathway-enrichment" class="section level3 hasAnchor" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> gene pathway enrichment<a href="example-unique-analyses.html#gene-pathway-enrichment" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>KEGG metabolic pathway-based pathway enrichment (PE) analysis to investigate the dysregulation of metabolic pathways considering genes.
因为基因差异注释后的比较少，所以选择阈值小点的</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="example-unique-analyses.html#cb10-1" aria-hidden="true" tabindex="-1"></a>all_dat <span class="ot">&lt;-</span> diff_gene</span>
<span id="cb10-2"><a href="example-unique-analyses.html#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="example-unique-analyses.html#cb10-3" aria-hidden="true" tabindex="-1"></a>all_dat1 <span class="ot">&lt;-</span> all_dat <span class="sc">%&gt;%</span></span>
<span id="cb10-4"><a href="example-unique-analyses.html#cb10-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC<span class="sc">&gt;</span><span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-5"><a href="example-unique-analyses.html#cb10-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb10-6"><a href="example-unique-analyses.html#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="example-unique-analyses.html#cb10-7" aria-hidden="true" tabindex="-1"></a>result1 <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(all_dat1<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;gene&quot;</span>,<span class="at">p_cutoff=</span><span class="fl">0.1</span>)</span>
<span id="cb10-8"><a href="example-unique-analyses.html#cb10-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-9"><a href="example-unique-analyses.html#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(result1<span class="sc">$</span>output,<span class="st">&quot;result_v0131/PE_gene1.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb10-10"><a href="example-unique-analyses.html#cb10-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-11"><a href="example-unique-analyses.html#cb10-11" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result_v0131/PE_gene1.png&quot;</span>,result1<span class="sc">$</span>gp,<span class="at">width=</span><span class="dv">30</span>,<span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb10-12"><a href="example-unique-analyses.html#cb10-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result_v0131/PE_gene_cluster1.png&quot;</span>,result1<span class="sc">$</span>p_barplot,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb10-13"><a href="example-unique-analyses.html#cb10-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-14"><a href="example-unique-analyses.html#cb10-14" aria-hidden="true" tabindex="-1"></a>all_dat2 <span class="ot">&lt;-</span> all_dat <span class="sc">%&gt;%</span></span>
<span id="cb10-15"><a href="example-unique-analyses.html#cb10-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-16"><a href="example-unique-analyses.html#cb10-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb10-17"><a href="example-unique-analyses.html#cb10-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-18"><a href="example-unique-analyses.html#cb10-18" aria-hidden="true" tabindex="-1"></a>result2 <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(all_dat2<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;gene&quot;</span>,<span class="at">p_cutoff=</span><span class="fl">0.1</span>)</span>
<span id="cb10-19"><a href="example-unique-analyses.html#cb10-19" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(result2<span class="sc">$</span>output,<span class="st">&quot;result_v0131/PE_gene2.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb10-20"><a href="example-unique-analyses.html#cb10-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-21"><a href="example-unique-analyses.html#cb10-21" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result_v0131/PE_gene2.png&quot;</span>,result2<span class="sc">$</span>gp,<span class="at">width=</span><span class="dv">25</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb10-22"><a href="example-unique-analyses.html#cb10-22" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result_v0131/PE_gene_cluster2.png&quot;</span>,result2<span class="sc">$</span>p_barplot,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb10-23"><a href="example-unique-analyses.html#cb10-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-24"><a href="example-unique-analyses.html#cb10-24" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> <span class="fu">list</span>(result1<span class="sc">$</span>p_barplot,result2<span class="sc">$</span>p_barplot))</span>
<span id="cb10-25"><a href="example-unique-analyses.html#cb10-25" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result_v0131/PE_gene.png&quot;</span>,p,<span class="at">width=</span><span class="dv">20</span>,<span class="at">height =</span> <span class="dv">10</span>)</span></code></pre></div>
<p><img src="result_v0131/PE_gene.png" />
<img src="result_v0131/PE_gene1.png" />
<img src="result_v0131/PE_gene2.png" /></p>
</div>
</div>
<div id="pathway-differential-abundance-score" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Pathway differential abundance score<a href="example-unique-analyses.html#pathway-differential-abundance-score" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="extended-pathway-differential-abundance-score" class="section level3 hasAnchor" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Extended pathway differential abundance score<a href="example-unique-analyses.html#extended-pathway-differential-abundance-score" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>KEGG metabolic pathway-based extended differential abundance (EDA) analysis investigate the dysregulation of metabolic pathways considering both metabolites and genes.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="example-unique-analyses.html#cb11-1" aria-hidden="true" tabindex="-1"></a>diff_gene_increase <span class="ot">&lt;-</span>  diff_gene <span class="sc">%&gt;%</span></span>
<span id="cb11-2"><a href="example-unique-analyses.html#cb11-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC<span class="sc">&gt;</span><span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-3"><a href="example-unique-analyses.html#cb11-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb11-4"><a href="example-unique-analyses.html#cb11-4" aria-hidden="true" tabindex="-1"></a>diff_gene_decrease <span class="ot">&lt;-</span> diff_gene <span class="sc">%&gt;%</span></span>
<span id="cb11-5"><a href="example-unique-analyses.html#cb11-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-6"><a href="example-unique-analyses.html#cb11-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb11-7"><a href="example-unique-analyses.html#cb11-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-8"><a href="example-unique-analyses.html#cb11-8" aria-hidden="true" tabindex="-1"></a>diff_meta_increase <span class="ot">&lt;-</span> diff_meta <span class="sc">%&gt;%</span></span>
<span id="cb11-9"><a href="example-unique-analyses.html#cb11-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC<span class="sc">&gt;</span><span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-10"><a href="example-unique-analyses.html#cb11-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb11-11"><a href="example-unique-analyses.html#cb11-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-12"><a href="example-unique-analyses.html#cb11-12" aria-hidden="true" tabindex="-1"></a>diff_meta_decrease <span class="ot">&lt;-</span> diff_meta <span class="sc">%&gt;%</span></span>
<span id="cb11-13"><a href="example-unique-analyses.html#cb11-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-14"><a href="example-unique-analyses.html#cb11-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb11-15"><a href="example-unique-analyses.html#cb11-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-16"><a href="example-unique-analyses.html#cb11-16" aria-hidden="true" tabindex="-1"></a> ExtendDAscore_result <span class="ot">&lt;-</span> <span class="fu">DAscore</span>(<span class="fu">c</span>(diff_gene_increase<span class="sc">$</span>name,diff_meta_increase<span class="sc">$</span>name),<span class="fu">c</span>(diff_gene_decrease<span class="sc">$</span>name,diff_meta_decrease<span class="sc">$</span>name),<span class="fu">c</span>(diff_gene<span class="sc">$</span>name,diff_meta<span class="sc">$</span>name),<span class="at">min_measured_num =</span> <span class="dv">2</span>,<span class="at">out=</span><span class="st">&quot;Extended&quot;</span>)</span>
<span id="cb11-17"><a href="example-unique-analyses.html#cb11-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-18"><a href="example-unique-analyses.html#cb11-18" aria-hidden="true" tabindex="-1"></a> <span class="fu">ggsave</span>(<span class="st">&quot;result_v0131/EDA.png&quot;</span>,ExtendDAscore_result<span class="sc">$</span>p,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">15</span>)</span></code></pre></div>
<p><img src="result_v0131/EDA.png" /></p>
</div>
<div id="metabolite-differential-abundance-score" class="section level3 hasAnchor" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> metabolite differential abundance score<a href="example-unique-analyses.html#metabolite-differential-abundance-score" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>KEGG metabolic pathway-based differential abundance (DA) analysis investigate the dysregulation of metabolic pathways considering only metabolites.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="example-unique-analyses.html#cb12-1" aria-hidden="true" tabindex="-1"></a>diff_meta_increase <span class="ot">&lt;-</span> diff_meta <span class="sc">%&gt;%</span></span>
<span id="cb12-2"><a href="example-unique-analyses.html#cb12-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC <span class="sc">&gt;</span> <span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-3"><a href="example-unique-analyses.html#cb12-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb12-4"><a href="example-unique-analyses.html#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="example-unique-analyses.html#cb12-5" aria-hidden="true" tabindex="-1"></a>diff_meta_decrease <span class="ot">&lt;-</span> diff_meta <span class="sc">%&gt;%</span></span>
<span id="cb12-6"><a href="example-unique-analyses.html#cb12-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-7"><a href="example-unique-analyses.html#cb12-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb12-8"><a href="example-unique-analyses.html#cb12-8" aria-hidden="true" tabindex="-1"></a>dascore_result_meta <span class="ot">&lt;-</span> <span class="fu">DAscore</span>(diff_meta_increase<span class="sc">$</span>name,diff_meta_decrease<span class="sc">$</span>name,diff_meta<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;metabolite&quot;</span>)</span>
<span id="cb12-9"><a href="example-unique-analyses.html#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result_v0131/DAscore_metabolite.png&quot;</span>,dascore_result_meta<span class="sc">$</span>p,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">15</span>)</span></code></pre></div>
<p><img src="result_v0131/DAscore_metabolite.png" /></p>
</div>
<div id="gene-differential-abundance-score" class="section level3 hasAnchor" number="2.3.3">
<h3><span class="header-section-number">2.3.3</span> gene differential abundance score<a href="example-unique-analyses.html#gene-differential-abundance-score" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>KEGG metabolic pathway-based differential abundance (DA) analysis investigate the dysregulation of metabolic pathways considering only genes.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="example-unique-analyses.html#cb13-1" aria-hidden="true" tabindex="-1"></a>diff_gene_increase <span class="ot">&lt;-</span> diff_gene <span class="sc">%&gt;%</span></span>
<span id="cb13-2"><a href="example-unique-analyses.html#cb13-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC <span class="sc">&gt;</span> <span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-3"><a href="example-unique-analyses.html#cb13-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb13-4"><a href="example-unique-analyses.html#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="example-unique-analyses.html#cb13-5" aria-hidden="true" tabindex="-1"></a>diff_gene_decrease <span class="ot">&lt;-</span> diff_gene <span class="sc">%&gt;%</span></span>
<span id="cb13-6"><a href="example-unique-analyses.html#cb13-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-7"><a href="example-unique-analyses.html#cb13-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb13-8"><a href="example-unique-analyses.html#cb13-8" aria-hidden="true" tabindex="-1"></a>dascore_result_gene <span class="ot">&lt;-</span> <span class="fu">DAscore</span>(diff_gene_increase<span class="sc">$</span>name,diff_gene_decrease<span class="sc">$</span>name,diff_gene<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb13-9"><a href="example-unique-analyses.html#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result_v0131/DAscore_gene.png&quot;</span>,dascore_result_gene<span class="sc">$</span>p,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">15</span>)</span></code></pre></div>
<p><img src="result_v0131/DAscore_gene.png" /></p>
</div>
</div>
<div id="msea-metabolite-set-enrichment-analysis" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> MSEA Metabolite Set Enrichment Analysis<a href="example-unique-analyses.html#msea-metabolite-set-enrichment-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>the metabolite set enrichment analysis using function “MSEA”</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="example-unique-analyses.html#cb14-1" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">MSEA</span>(sim.cpd.data)</span>
<span id="cb14-2"><a href="example-unique-analyses.html#cb14-2" aria-hidden="true" tabindex="-1"></a>result<span class="sc">$</span>leadingEdge <span class="ot">&lt;-</span> <span class="fu">as.character</span>(result<span class="sc">$</span>leadingEdge)</span>
<span id="cb14-3"><a href="example-unique-analyses.html#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(result,<span class="at">file=</span><span class="st">&quot;result_v0131/MSEA.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
<p>the msea plot of interested pathway</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="example-unique-analyses.html#cb15-1" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">pMSEA</span>(<span class="st">&quot;Butanoate metabolism&quot;</span>,sim.cpd.data)</span>
<span id="cb15-2"><a href="example-unique-analyses.html#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result_v0131/MSEA_Butanoate.png&quot;</span>,p,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="dv">4</span>)</span></code></pre></div>
<p><img src="result_v0131/MSEA_Butanoate.png" /></p>
</div>
<div id="subnetwork-analysis-of-metabolism-and-transcriptome-data-from-tnbc" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> Subnetwork analysis of Metabolism and transcriptome data from TNBC:<a href="example-unique-analyses.html#subnetwork-analysis-of-metabolism-and-transcriptome-data-from-tnbc" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>the data need to have the column name “name”, and have the “p_value”,“logFC” is alternative. If have the “logFC”, the color in the figure is the logFC, otherwise, the color “blue” in figure means metabolites and the color “red” means the genes.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="example-unique-analyses.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(diff_meta)[<span class="dv">4</span>]  <span class="ot">&lt;-</span> <span class="st">&quot;p_value&quot;</span></span>
<span id="cb16-2"><a href="example-unique-analyses.html#cb16-2" aria-hidden="true" tabindex="-1"></a>diff_metabolite <span class="ot">&lt;-</span> diff_meta <span class="sc">%&gt;%</span></span>
<span id="cb16-3"><a href="example-unique-analyses.html#cb16-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.01</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-4"><a href="example-unique-analyses.html#cb16-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">abs</span>(logFC)<span class="sc">&gt;</span><span class="dv">1</span>)</span>
<span id="cb16-5"><a href="example-unique-analyses.html#cb16-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-6"><a href="example-unique-analyses.html#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(diff_gene)[<span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="st">&quot;p_value&quot;</span></span>
<span id="cb16-7"><a href="example-unique-analyses.html#cb16-7" aria-hidden="true" tabindex="-1"></a>diff_gene1 <span class="ot">&lt;-</span> diff_gene <span class="sc">%&gt;%</span></span>
<span id="cb16-8"><a href="example-unique-analyses.html#cb16-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.01</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-9"><a href="example-unique-analyses.html#cb16-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">abs</span>(logFC)<span class="sc">&gt;</span><span class="dv">1</span>)</span>
<span id="cb16-10"><a href="example-unique-analyses.html#cb16-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-11"><a href="example-unique-analyses.html#cb16-11" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">&quot;result_v0131/subnetwork_important.pdf&quot;</span>)</span>
<span id="cb16-12"><a href="example-unique-analyses.html#cb16-12" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">pdnet</span>(diff_meta,diff_gene,<span class="at">nsize=</span><span class="dv">100</span>)</span>
<span id="cb16-13"><a href="example-unique-analyses.html#cb16-13" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<embed src="result_v0131/subnetwork_important.pdf" width="600px" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="subnetwork-analysis-of-interested-metabolites-and-genes" class="section level2 hasAnchor" number="2.6">
<h2><span class="header-section-number">2.6</span> Subnetwork analysis of interested metabolites and genes<a href="example-unique-analyses.html#subnetwork-analysis-of-interested-metabolites-and-genes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>the data need to have the column name “name”, and do not have the “p_value”</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="example-unique-analyses.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">&quot;result_v0131/subnetwork_interested.pdf&quot;</span>)</span>
<span id="cb17-2"><a href="example-unique-analyses.html#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="do">## get 500 differential expression gene</span></span>
<span id="cb17-3"><a href="example-unique-analyses.html#cb17-3" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">pdnet</span>(diff_metabolite[,<span class="dv">8</span>],diff_gene1[<span class="dv">1</span><span class="sc">:</span><span class="dv">500</span>,<span class="dv">8</span>])</span>
<span id="cb17-4"><a href="example-unique-analyses.html#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb17-5"><a href="example-unique-analyses.html#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(a,<span class="st">&quot;result_v0131/subnetwork_interested_edge.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
<embed src="result_v0131/subnetwork_interested.pdf" width="600px" style="display: block; margin: auto;" type="application/pdf" />

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="example-rountine-analyses.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/01-example_unique.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
