<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Example-unique analyses | MNet</title>
  <meta name="description" content="Chapter 3 Example-unique analyses | MNet" />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Example-unique analyses | MNet" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Example-unique analyses | MNet" />
  
  
  

<meta name="author" content="Tuantuan Gui, Xuan Liu and Hai Fang" />


<meta name="date" content="2024-02-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="database-construction.html"/>
<link rel="next" href="example-rountine-analyses.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.30/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-buttons-1.13.4/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-buttons-1.13.4/js/dataTables.buttons.min.js"></script>
<script src="libs/dt-ext-buttons-1.13.4/js/buttons.html5.min.js"></script>
<script src="libs/dt-ext-buttons-1.13.4/js/buttons.colVis.min.js"></script>
<script src="libs/dt-ext-buttons-1.13.4/js/buttons.print.min.js"></script>
<link href="libs/dt-ext-fixedcolumns-1.13.4/css/fixedColumns.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedcolumns-1.13.4/js/dataTables.fixedColumns.min.js"></script>
<link href="libs/dt-plugin-searchhighlight-1.13.4/source.css" rel="stylesheet" />
<script src="libs/dt-plugin-searchhighlight-1.13.4/jquery.highlight.js"></script>
<script src="libs/dt-plugin-searchhighlight-1.13.4/source.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction &amp; Installation</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#overview-of-workflow"><i class="fa fa-check"></i><b>1.1</b> Overview of Workflow</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.2</b> Installation</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="database-construction.html"><a href="database-construction.html"><i class="fa fa-check"></i><b>2</b> Database construction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="database-construction.html"><a href="database-construction.html#kegg-pathways-metabolite-and-gene"><i class="fa fa-check"></i><b>2.1</b> KEGG pathway’s metabolite and gene</a></li>
<li class="chapter" data-level="2.2" data-path="database-construction.html"><a href="database-construction.html#kegg-metabolite-metabolite-pairs-and-metabolite-gene-pairs"><i class="fa fa-check"></i><b>2.2</b> KEGG metabolite-metabolite pairs and metabolite-gene pairs</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="database-construction.html"><a href="database-construction.html#keggs-reaction-and-compounds-download"><i class="fa fa-check"></i><b>2.2.1</b> KEGG’s reaction and compound’s download</a></li>
<li class="chapter" data-level="2.2.2" data-path="database-construction.html"><a href="database-construction.html#downloading-the-enzyme-and-its-corresponding-gene-information-from-the-htextand-extract-the-gene-and-the-enzymes-relationship"><i class="fa fa-check"></i><b>2.2.2</b> Downloading the enzyme and its corresponding gene information from the htext,and extract the gene and the enzyme’s relationship</a></li>
<li class="chapter" data-level="2.2.3" data-path="database-construction.html"><a href="database-construction.html#extracting-the-metabolite-gene-pairs-and-metabolite-metabolite-pairs"><i class="fa fa-check"></i><b>2.2.3</b> Extracting the metabolite-gene pairs and metabolite-metabolite pairs</a></li>
<li class="chapter" data-level="2.2.4" data-path="database-construction.html"><a href="database-construction.html#conclusion"><i class="fa fa-check"></i><b>2.2.4</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="database-construction.html"><a href="database-construction.html#graphite"><i class="fa fa-check"></i><b>2.3</b> Graphite</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="database-construction.html"><a href="database-construction.html#download-all-the-data-from-graphite"><i class="fa fa-check"></i><b>2.3.1</b> Download all the data from graphite</a></li>
<li class="chapter" data-level="2.3.2" data-path="database-construction.html"><a href="database-construction.html#download-the-information-of-metabolism-pathways"><i class="fa fa-check"></i><b>2.3.2</b> Download the information of metabolism pathways</a></li>
<li class="chapter" data-level="2.3.3" data-path="database-construction.html"><a href="database-construction.html#extract-the-data-of-metabolism"><i class="fa fa-check"></i><b>2.3.3</b> Extract the data of metabolism</a></li>
<li class="chapter" data-level="2.3.4" data-path="database-construction.html"><a href="database-construction.html#combine-the-data-from-database-kegg-wikipathway-reactome-smpdb-and-then-uniq-the-data"><i class="fa fa-check"></i><b>2.3.4</b> Combine the data from database kegg, wikipathway, reactome, smpdb and then uniq the data</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="database-construction.html"><a href="database-construction.html#bigg"><i class="fa fa-check"></i><b>2.4</b> BiGG</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="database-construction.html"><a href="database-construction.html#down-all-the-models-in-the-bigg"><i class="fa fa-check"></i><b>2.4.1</b> Down all the models in the BiGG</a></li>
<li class="chapter" data-level="2.4.2" data-path="database-construction.html"><a href="database-construction.html#download-the-every-reaction"><i class="fa fa-check"></i><b>2.4.2</b> Download the every reaction</a></li>
<li class="chapter" data-level="2.4.3" data-path="database-construction.html"><a href="database-construction.html#exact-all-the-metabolites-names-in-every-model-in-bigg"><i class="fa fa-check"></i><b>2.4.3</b> Exact all the metabolites names in every model in BiGG</a></li>
<li class="chapter" data-level="2.4.4" data-path="database-construction.html"><a href="database-construction.html#the-r-script-that-change-the-metabolite-name-to-kegg-id-in-the-final-output"><i class="fa fa-check"></i><b>2.4.4</b> The R script that change the metabolite name to kegg id in the final output</a></li>
<li class="chapter" data-level="2.4.5" data-path="database-construction.html"><a href="database-construction.html#combine-the-data-from-bigg-and-graphite"><i class="fa fa-check"></i><b>2.4.5</b> Combine the data from BiGG and graphite</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html"><i class="fa fa-check"></i><b>3</b> Example-unique analyses</a>
<ul>
<li class="chapter" data-level="3.1" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#loading-the-data"><i class="fa fa-check"></i><b>3.1</b> Loading the data</a></li>
<li class="chapter" data-level="3.2" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#pathway-enrichment-analysis"><i class="fa fa-check"></i><b>3.2</b> Pathway enrichment analysis</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#gene-pathway-enrichment"><i class="fa fa-check"></i><b>3.2.1</b> gene pathway enrichment</a></li>
<li class="chapter" data-level="3.2.2" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#metabolite-pathway-enrichment"><i class="fa fa-check"></i><b>3.2.2</b> metabolite pathway enrichment</a></li>
<li class="chapter" data-level="3.2.3" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#extended-pathway-enrichment"><i class="fa fa-check"></i><b>3.2.3</b> Extended pathway enrichment</a></li>
<li class="chapter" data-level="3.2.4" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#overview-of-pathway-enrichment-analysis"><i class="fa fa-check"></i><b>3.2.4</b> Overview of pathway enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#pathway-differential-abundance-score"><i class="fa fa-check"></i><b>3.3</b> Pathway differential abundance score</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#gene-differential-abundance-score"><i class="fa fa-check"></i><b>3.3.1</b> gene differential abundance score</a></li>
<li class="chapter" data-level="3.3.2" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#metabolite-differential-abundance-score"><i class="fa fa-check"></i><b>3.3.2</b> metabolite differential abundance score</a></li>
<li class="chapter" data-level="3.3.3" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#extended-pathway-differential-abundance-score"><i class="fa fa-check"></i><b>3.3.3</b> Extended pathway differential abundance score</a></li>
<li class="chapter" data-level="3.3.4" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#overview-of-pathway-differential-abundance-score"><i class="fa fa-check"></i><b>3.3.4</b> Overview of pathway differential abundance score</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#esea-metabolite-set-enrichment-analysis"><i class="fa fa-check"></i><b>3.4</b> ESEA Metabolite Set Enrichment Analysis</a></li>
<li class="chapter" data-level="3.5" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#subnetwork"><i class="fa fa-check"></i><b>3.5</b> Subnetwork</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html"><i class="fa fa-check"></i><b>4</b> Example-rountine analyses</a>
<ul>
<li class="chapter" data-level="4.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-and-kegg-name-change"><i class="fa fa-check"></i><b>4.1</b> Metabolite and kegg name change</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-name-changed-to-refmet-name"><i class="fa fa-check"></i><b>4.1.1</b> Metabolite name changed to refmet name</a></li>
<li class="chapter" data-level="4.1.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-name-changed-to-kegg-id"><i class="fa fa-check"></i><b>4.1.2</b> Metabolite name changed to kegg id</a></li>
<li class="chapter" data-level="4.1.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-name-corresponding-to-kegg-pathway"><i class="fa fa-check"></i><b>4.1.3</b> Metabolite name corresponding to kegg pathway</a></li>
<li class="chapter" data-level="4.1.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-keggid-to-kegg-pathway"><i class="fa fa-check"></i><b>4.1.4</b> Metabolite keggid to kegg pathway</a></li>
<li class="chapter" data-level="4.1.5" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#the-pathway-information"><i class="fa fa-check"></i><b>4.1.5</b> The pathway information</a></li>
<li class="chapter" data-level="4.1.6" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathway-name-to-pathway-id"><i class="fa fa-check"></i><b>4.1.6</b> Pathway name to pathway id</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#group-wise-analyses"><i class="fa fa-check"></i><b>4.2</b> Group-wise analyses</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pca-plot"><i class="fa fa-check"></i><b>4.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="4.2.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differnetial-metabolite-analysis"><i class="fa fa-check"></i><b>4.2.2</b> Differnetial metabolite analysis</a></li>
<li class="chapter" data-level="4.2.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differential-metabolites-volcano"><i class="fa fa-check"></i><b>4.2.3</b> Differential metabolites’ volcano</a></li>
<li class="chapter" data-level="4.2.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differential-metabolites-heatmap"><i class="fa fa-check"></i><b>4.2.4</b> Differential metabolites’ heatmap</a></li>
<li class="chapter" data-level="4.2.5" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differential-metabolites-zscore"><i class="fa fa-check"></i><b>4.2.5</b> Differential metabolites’ zscore</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathway-analysis"><i class="fa fa-check"></i><b>4.3</b> Pathway analysis</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathview-only-metabolite"><i class="fa fa-check"></i><b>4.3.1</b> Pathview only metabolite</a></li>
<li class="chapter" data-level="4.3.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathview-metabolite-and-gene"><i class="fa fa-check"></i><b>4.3.2</b> Pathview metabolite and gene</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#clinical-analysis"><i class="fa fa-check"></i><b>4.4</b> Clinical analysis</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#time-series-of-clinical"><i class="fa fa-check"></i><b>4.4.1</b> Time series of clinical</a></li>
<li class="chapter" data-level="4.4.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#survival-analysis"><i class="fa fa-check"></i><b>4.4.2</b> Survival analysis</a></li>
<li class="chapter" data-level="4.4.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolites-survival-plot"><i class="fa fa-check"></i><b>4.4.3</b> Metabolites’ survival plot</a></li>
<li class="chapter" data-level="4.4.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#cox-analysis"><i class="fa fa-check"></i><b>4.4.4</b> Cox analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="shiny.html"><a href="shiny.html"><i class="fa fa-check"></i><b>5</b> Shiny</a>
<ul>
<li class="chapter" data-level="5.1" data-path="shiny.html"><a href="shiny.html#tencent-log"><i class="fa fa-check"></i><b>5.1</b> Tencent log</a></li>
<li class="chapter" data-level="5.2" data-path="shiny.html"><a href="shiny.html#shiny-app-log"><i class="fa fa-check"></i><b>5.2</b> Shiny app log</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="figure1.html"><a href="figure1.html"><i class="fa fa-check"></i><b>6</b> Figure1</a>
<ul>
<li class="chapter" data-level="6.1" data-path="figure1.html"><a href="figure1.html#a"><i class="fa fa-check"></i><b>6.1</b> 1.1 (A)</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="figure1.html"><a href="figure1.html#batch-removal"><i class="fa fa-check"></i><b>6.1.1</b> Batch removal</a></li>
<li class="chapter" data-level="6.1.2" data-path="figure1.html"><a href="figure1.html#metabolite-information"><i class="fa fa-check"></i><b>6.1.2</b> Metabolite information</a></li>
<li class="chapter" data-level="6.1.3" data-path="figure1.html"><a href="figure1.html#metabolic-pathway-information"><i class="fa fa-check"></i><b>6.1.3</b> Metabolic pathway information</a></li>
<li class="chapter" data-level="6.1.4" data-path="figure1.html"><a href="figure1.html#metabolite-and-metabolic-pathway-annotation"><i class="fa fa-check"></i><b>6.1.4</b> Metabolite and metabolic pathway annotation</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="figure1.html"><a href="figure1.html#b"><i class="fa fa-check"></i><b>6.2</b> 1.2 (B)</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="figure1.html"><a href="figure1.html#clinical-and-metabolite-analysis"><i class="fa fa-check"></i><b>6.2.1</b> Clinical and metabolite analysis</a></li>
<li class="chapter" data-level="6.2.2" data-path="figure1.html"><a href="figure1.html#time-series-analysis"><i class="fa fa-check"></i><b>6.2.2</b> Time series analysis</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="figure1.html"><a href="figure1.html#c"><i class="fa fa-check"></i><b>6.3</b> 1.3 (C)</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="figure1.html"><a href="figure1.html#dimension-reduction-visualization"><i class="fa fa-check"></i><b>6.3.1</b> Dimension reduction &amp; Visualization</a></li>
<li class="chapter" data-level="6.3.2" data-path="figure1.html"><a href="figure1.html#differential-metabolites-visualization"><i class="fa fa-check"></i><b>6.3.2</b> Differential metabolites visualization</a></li>
<li class="chapter" data-level="6.3.3" data-path="figure1.html"><a href="figure1.html#pathway-enrichment-analyses"><i class="fa fa-check"></i><b>6.3.3</b> Pathway enrichment analyses</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="figure1.html"><a href="figure1.html#d"><i class="fa fa-check"></i><b>6.4</b> 1.4 (D)</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="figure1.html"><a href="figure1.html#extended-pathway-enrichment-analysis-epea"><i class="fa fa-check"></i><b>6.4.1</b> Extended pathway enrichment analysis (EPEA)</a></li>
<li class="chapter" data-level="6.4.2" data-path="figure1.html"><a href="figure1.html#extended-differential-abundance-eda-score"><i class="fa fa-check"></i><b>6.4.2</b> Extended differential abundance (EDA) score</a></li>
<li class="chapter" data-level="6.4.3" data-path="figure1.html"><a href="figure1.html#extended-data-set-enrichment-analysis-esea"><i class="fa fa-check"></i><b>6.4.3</b> Extended data set enrichment analysis (ESEA)</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="figure1.html"><a href="figure1.html#e"><i class="fa fa-check"></i><b>6.5</b> 1.5 (E)</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="figure1.html"><a href="figure1.html#dbmnet-database"><i class="fa fa-check"></i><b>6.5.1</b> dbMNet database</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="figure4.html"><a href="figure4.html"><i class="fa fa-check"></i><b>7</b> Figure4</a>
<ul>
<li class="chapter" data-level="7.1" data-path="figure4.html"><a href="figure4.html#load-data"><i class="fa fa-check"></i><b>7.1</b> Load data</a></li>
<li class="chapter" data-level="7.2" data-path="figure4.html"><a href="figure4.html#mlimma-analysis"><i class="fa fa-check"></i><b>7.2</b> mlimma analysis</a></li>
<li class="chapter" data-level="7.3" data-path="figure4.html"><a href="figure4.html#epe-analysis"><i class="fa fa-check"></i><b>7.3</b> EPE analysis</a></li>
<li class="chapter" data-level="7.4" data-path="figure4.html"><a href="figure4.html#eda-analysis"><i class="fa fa-check"></i><b>7.4</b> EDA analysis</a></li>
<li class="chapter" data-level="7.5" data-path="figure4.html"><a href="figure4.html#esea-analysis"><i class="fa fa-check"></i><b>7.5</b> ESEA analysis</a></li>
<li class="chapter" data-level="7.6" data-path="figure4.html"><a href="figure4.html#aml-epe-analysis"><i class="fa fa-check"></i><b>7.6</b> AML EPE analysis</a></li>
<li class="chapter" data-level="7.7" data-path="figure4.html"><a href="figure4.html#volcano-plot-1"><i class="fa fa-check"></i><b>7.7</b> Volcano plot</a>
<ul>
<li class="chapter" data-level="7.7.1" data-path="figure4.html"><a href="figure4.html#differential-metabolites-and-genes"><i class="fa fa-check"></i><b>7.7.1</b> Differential metabolites and genes</a></li>
<li class="chapter" data-level="7.7.2" data-path="figure4.html"><a href="figure4.html#all"><i class="fa fa-check"></i><b>7.7.2</b> ALL</a></li>
<li class="chapter" data-level="7.7.3" data-path="figure4.html"><a href="figure4.html#aml"><i class="fa fa-check"></i><b>7.7.3</b> AML</a></li>
</ul></li>
<li class="chapter" data-level="7.8" data-path="figure4.html"><a href="figure4.html#heatmap-1"><i class="fa fa-check"></i><b>7.8</b> Heatmap</a></li>
<li class="chapter" data-level="7.9" data-path="figure4.html"><a href="figure4.html#subnetwork-1"><i class="fa fa-check"></i><b>7.9</b> Subnetwork</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>8</b> Session information</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">MNet</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="example-unique-analyses" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Example-unique analyses<a href="example-unique-analyses.html#example-unique-analyses" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="loading-the-data" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Loading the data<a href="example-unique-analyses.html#loading-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The case study is an integrated metabolic and transcriptional analysis to understand the rapidly proliferating breast cancers (Liao, et al., 2022). In this paper, the authors characterized the metabolomic profile of two types of 31 breast tumors in human to uncovered pyrimidine metabolism and glutaminolysis pathways are elevated in TNBC. As an example of mixed metabolites and genes analyses, here we are going to use both the dataset of metabolite and gene expression data as reported in the paper. The metabolomics data was accessed from the Supplementary Tables S2 in the paper, and the gene expression data is from the author. They found that pyrimidine and purine metabolism, carbon metabolism, and the energy production related pathways were highly enriched in Cluster 2 while many lipid/fatty acid metabolism related pathways were enriched in Cluster 1.
We will use MNet to identify a set of features whose activity changes significantly between the two sample clusters.This result will hopefully hint at some specific biological activities that are pathologically altered in tumoral samples.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="example-unique-analyses.html#cb31-1" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb31-2"><a href="example-unique-analyses.html#cb31-2" tabindex="-1"></a><span class="fu">library</span>(MNet)</span>
<span id="cb31-3"><a href="example-unique-analyses.html#cb31-3" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb31-4"><a href="example-unique-analyses.html#cb31-4" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb31-5"><a href="example-unique-analyses.html#cb31-5" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb31-6"><a href="example-unique-analyses.html#cb31-6" tabindex="-1"></a></span>
<span id="cb31-7"><a href="example-unique-analyses.html#cb31-7" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;result&quot;</span>)</span>
<span id="cb31-8"><a href="example-unique-analyses.html#cb31-8" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;result/unique&quot;</span>)</span>
<span id="cb31-9"><a href="example-unique-analyses.html#cb31-9" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;result/routine&quot;</span>)</span>
<span id="cb31-10"><a href="example-unique-analyses.html#cb31-10" tabindex="-1"></a></span>
<span id="cb31-11"><a href="example-unique-analyses.html#cb31-11" tabindex="-1"></a>group[<span class="fu">which</span>(group<span class="sc">==</span><span class="st">&quot;normal&quot;</span>)] <span class="ot">&lt;-</span> <span class="st">&quot;a&quot;</span></span>
<span id="cb31-12"><a href="example-unique-analyses.html#cb31-12" tabindex="-1"></a>group[<span class="fu">which</span>(group<span class="sc">==</span><span class="st">&quot;tumor&quot;</span>)] <span class="ot">&lt;-</span> <span class="st">&quot;normal&quot;</span></span>
<span id="cb31-13"><a href="example-unique-analyses.html#cb31-13" tabindex="-1"></a>group[<span class="fu">which</span>(group<span class="sc">==</span><span class="st">&quot;a&quot;</span>)] <span class="ot">&lt;-</span> <span class="st">&quot;tumor&quot;</span></span>
<span id="cb31-14"><a href="example-unique-analyses.html#cb31-14" tabindex="-1"></a></span>
<span id="cb31-15"><a href="example-unique-analyses.html#cb31-15" tabindex="-1"></a>all_meta <span class="ot">&lt;-</span> <span class="fu">mlimma</span>(meta_dat,group)</span>
<span id="cb31-16"><a href="example-unique-analyses.html#cb31-16" tabindex="-1"></a>all_gene <span class="ot">&lt;-</span> <span class="fu">mlimma</span>(gene_dat,group)</span>
<span id="cb31-17"><a href="example-unique-analyses.html#cb31-17" tabindex="-1"></a><span class="fu">head</span>(all_meta)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 8
##   logFC AveExpr     t  P.Value    adj.P.Val     B  logP name  
##   &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;        &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt; 
## 1 -2.86    22.6 -9.25 1.52e-10 0.0000000332 14.1   7.48 C02045
## 2 -2.44    26.2 -7.80 6.83e- 9 0.000000748  10.4   6.13 C00267
## 3  1.82    27.1  6.80 1.10e- 7 0.00000622    7.64  5.21 C00073
## 4  3.78    20.9  6.79 1.14e- 7 0.00000622    7.61  5.21 C05674
## 5  2.20    21.4  6.58 2.07e- 7 0.00000907    7.02  5.04 C00255
## 6  2.37    21.6  6.45 2.98e- 7 0.0000109     6.66  4.96 C00242</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="example-unique-analyses.html#cb33-1" tabindex="-1"></a><span class="fu">head</span>(all_gene)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 8
##    logFC AveExpr      t  P.Value    adj.P.Val     B  logP name  
##    &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;        &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt; 
## 1  -6.92   23.1  -11.3  1.15e-12 0.0000000203  17.7  7.69 APH1B 
## 2 -21.4    23.4  -10.6  5.13e-12 0.0000000455  16.5  7.34 GFRA1 
## 3   5.76   24.4   10.1  1.85e-11 0.0000000891  15.4  7.05 RFC4  
## 4 -10.5    14.8  -10.1  2.01e-11 0.0000000891  15.3  7.05 FAM47E
## 5  10.8    20.6    9.79 3.85e-11 0.000000137   14.7  6.86 ORC6  
## 6  -4.02    5.65  -9.69 4.98e-11 0.000000147   14.5  6.83 FAM87B</code></pre>
</div>
<div id="pathway-enrichment-analysis" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Pathway enrichment analysis<a href="example-unique-analyses.html#pathway-enrichment-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="gene-pathway-enrichment" class="section level3 hasAnchor" number="3.2.1">
<h3><span class="header-section-number">3.2.1</span> gene pathway enrichment<a href="example-unique-analyses.html#gene-pathway-enrichment" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>KEGG metabolic pathway-based pathway enrichment (PE) analysis to investigate the dysregulation of metabolic pathways considering genes.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="example-unique-analyses.html#cb35-1" tabindex="-1"></a><span class="do">## gene</span></span>
<span id="cb35-2"><a href="example-unique-analyses.html#cb35-2" tabindex="-1"></a>diff_gene_increase <span class="ot">&lt;-</span> all_gene <span class="sc">%&gt;%</span></span>
<span id="cb35-3"><a href="example-unique-analyses.html#cb35-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC <span class="sc">&gt;</span> <span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-4"><a href="example-unique-analyses.html#cb35-4" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb35-5"><a href="example-unique-analyses.html#cb35-5" tabindex="-1"></a></span>
<span id="cb35-6"><a href="example-unique-analyses.html#cb35-6" tabindex="-1"></a>result_gene_increase <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(diff_gene_increase<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;gene&quot;</span>,<span class="at">p_cutoff=</span><span class="fl">0.05</span>)</span>
<span id="cb35-7"><a href="example-unique-analyses.html#cb35-7" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/01.gene_increase.pdf&quot;</span>,result_gene_increase<span class="sc">$</span>p_barplot,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb35-8"><a href="example-unique-analyses.html#cb35-8" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/01.gene_increase.png&quot;</span>,result_gene_increase<span class="sc">$</span>p_barplot,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb35-9"><a href="example-unique-analyses.html#cb35-9" tabindex="-1"></a><span class="fu">write.table</span>(result_gene_increase<span class="sc">$</span>output,<span class="st">&quot;result/unique/01.gene_increase.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names =</span> F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb35-10"><a href="example-unique-analyses.html#cb35-10" tabindex="-1"></a></span>
<span id="cb35-11"><a href="example-unique-analyses.html#cb35-11" tabindex="-1"></a>diff_gene_decrease <span class="ot">&lt;-</span> all_gene <span class="sc">%&gt;%</span></span>
<span id="cb35-12"><a href="example-unique-analyses.html#cb35-12" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-13"><a href="example-unique-analyses.html#cb35-13" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb35-14"><a href="example-unique-analyses.html#cb35-14" tabindex="-1"></a></span>
<span id="cb35-15"><a href="example-unique-analyses.html#cb35-15" tabindex="-1"></a>result_gene_decrease <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(diff_gene_decrease<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;gene&quot;</span>,<span class="at">p_cutoff=</span><span class="fl">0.05</span>)</span>
<span id="cb35-16"><a href="example-unique-analyses.html#cb35-16" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/01.gene_decrease.pdf&quot;</span>,result_gene_decrease<span class="sc">$</span>p_barplot,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb35-17"><a href="example-unique-analyses.html#cb35-17" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/01.gene_decrease.png&quot;</span>,result_gene_decrease<span class="sc">$</span>p_barplot,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb35-18"><a href="example-unique-analyses.html#cb35-18" tabindex="-1"></a><span class="fu">write.table</span>(result_gene_decrease<span class="sc">$</span>output,<span class="st">&quot;result/unique/01.gene_decrease.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names =</span> F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
<p>Gene_increase and Gene_decrease pathway enrichment
<img src="result/unique/01.gene_increase.png" width="50%" /><img src="result/unique/01.gene_decrease.png" width="50%" /></p>
</div>
<div id="metabolite-pathway-enrichment" class="section level3 hasAnchor" number="3.2.2">
<h3><span class="header-section-number">3.2.2</span> metabolite pathway enrichment<a href="example-unique-analyses.html#metabolite-pathway-enrichment" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>KEGG metabolic pathway-based pathway enrichment (PE) analysis to investigate the dysregulation of metabolic pathways considering metabolites.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="example-unique-analyses.html#cb36-1" tabindex="-1"></a><span class="do">## metabolite</span></span>
<span id="cb36-2"><a href="example-unique-analyses.html#cb36-2" tabindex="-1"></a><span class="do">## filter the increase differential genes and metabolites</span></span>
<span id="cb36-3"><a href="example-unique-analyses.html#cb36-3" tabindex="-1"></a>diff_meta_increase <span class="ot">&lt;-</span> all_meta <span class="sc">%&gt;%</span></span>
<span id="cb36-4"><a href="example-unique-analyses.html#cb36-4" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC <span class="sc">&gt;</span> <span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-5"><a href="example-unique-analyses.html#cb36-5" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb36-6"><a href="example-unique-analyses.html#cb36-6" tabindex="-1"></a></span>
<span id="cb36-7"><a href="example-unique-analyses.html#cb36-7" tabindex="-1"></a>result_meta_increase <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(diff_meta_increase<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;metabolite&quot;</span>,<span class="at">p_cutoff=</span><span class="fl">0.05</span>)</span>
<span id="cb36-8"><a href="example-unique-analyses.html#cb36-8" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/02.meta_increase.pdf&quot;</span>,result_meta_increase<span class="sc">$</span>p_barplot,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb36-9"><a href="example-unique-analyses.html#cb36-9" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/02.meta_increase.png&quot;</span>,result_meta_increase<span class="sc">$</span>p_barplot,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb36-10"><a href="example-unique-analyses.html#cb36-10" tabindex="-1"></a><span class="fu">write.table</span>(result_meta_increase<span class="sc">$</span>output,<span class="st">&quot;result/unique/02.meta_increase.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names =</span> F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb36-11"><a href="example-unique-analyses.html#cb36-11" tabindex="-1"></a></span>
<span id="cb36-12"><a href="example-unique-analyses.html#cb36-12" tabindex="-1"></a>diff_meta_decrease <span class="ot">&lt;-</span> all_meta <span class="sc">%&gt;%</span></span>
<span id="cb36-13"><a href="example-unique-analyses.html#cb36-13" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-14"><a href="example-unique-analyses.html#cb36-14" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb36-15"><a href="example-unique-analyses.html#cb36-15" tabindex="-1"></a></span>
<span id="cb36-16"><a href="example-unique-analyses.html#cb36-16" tabindex="-1"></a>result_meta_decrease <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(diff_meta_decrease<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;metabolite&quot;</span>,<span class="at">p_cutoff=</span><span class="fl">0.05</span>)</span>
<span id="cb36-17"><a href="example-unique-analyses.html#cb36-17" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/02.meta_decrease.pdf&quot;</span>,result_meta_decrease<span class="sc">$</span>p_barplot,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb36-18"><a href="example-unique-analyses.html#cb36-18" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/02.meta_decrease.png&quot;</span>,result_meta_decrease<span class="sc">$</span>p_barplot,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb36-19"><a href="example-unique-analyses.html#cb36-19" tabindex="-1"></a><span class="fu">write.table</span>(result_meta_decrease<span class="sc">$</span>output,<span class="st">&quot;result/unique/02.meta_decrease.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names =</span> F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
<p>Metabolite_increase and Metabolite_decrease pathway enrichment
<img src="result/unique/02.meta_increase.png" width="50%" /><img src="result/unique/02.meta_decrease.png" width="50%" /></p>
</div>
<div id="extended-pathway-enrichment" class="section level3 hasAnchor" number="3.2.3">
<h3><span class="header-section-number">3.2.3</span> Extended pathway enrichment<a href="example-unique-analyses.html#extended-pathway-enrichment" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>KEGG metabolic pathway-based extended pathway enrichment analysis (EPEA) to investigate the dysregulation of metabolic pathways considering both metabolites and genes using the function ‘PathwayAnalysis’.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="example-unique-analyses.html#cb37-1" tabindex="-1"></a><span class="do">## gene and metabolite</span></span>
<span id="cb37-2"><a href="example-unique-analyses.html#cb37-2" tabindex="-1"></a>diff_increase <span class="ot">&lt;-</span> <span class="fu">rbind</span>(diff_gene_increase,diff_meta_increase)</span>
<span id="cb37-3"><a href="example-unique-analyses.html#cb37-3" tabindex="-1"></a>result_all_increase <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(diff_increase<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;Extended&quot;</span>,<span class="at">p_cutoff=</span><span class="fl">0.05</span>)</span>
<span id="cb37-4"><a href="example-unique-analyses.html#cb37-4" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/03.all_increase.pdf&quot;</span>,result_all_increase<span class="sc">$</span>p_barplot,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb37-5"><a href="example-unique-analyses.html#cb37-5" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/03.all_increase.png&quot;</span>,result_all_increase<span class="sc">$</span>p_barplot,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb37-6"><a href="example-unique-analyses.html#cb37-6" tabindex="-1"></a><span class="fu">write.table</span>(result_all_increase<span class="sc">$</span>output,<span class="st">&quot;result/unique/03.all_increase.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names =</span> F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb37-7"><a href="example-unique-analyses.html#cb37-7" tabindex="-1"></a></span>
<span id="cb37-8"><a href="example-unique-analyses.html#cb37-8" tabindex="-1"></a>diff_decrease <span class="ot">&lt;-</span> <span class="fu">rbind</span>(diff_gene_decrease,diff_meta_decrease)</span>
<span id="cb37-9"><a href="example-unique-analyses.html#cb37-9" tabindex="-1"></a>result_all_decrease <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(diff_decrease<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;Extended&quot;</span>,<span class="at">p_cutoff=</span><span class="fl">0.05</span>)</span>
<span id="cb37-10"><a href="example-unique-analyses.html#cb37-10" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/03.all_decrease.pdf&quot;</span>,result_all_decrease<span class="sc">$</span>p_barplot,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb37-11"><a href="example-unique-analyses.html#cb37-11" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/03.all_decrease.png&quot;</span>,result_all_decrease<span class="sc">$</span>p_barplot,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb37-12"><a href="example-unique-analyses.html#cb37-12" tabindex="-1"></a><span class="fu">write.table</span>(result_all_decrease<span class="sc">$</span>output,<span class="st">&quot;result/unique/03.all_decrease.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names =</span> F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
<p>Extended pathway enrichment(increase and decrease)
<img src="result/unique/03.all_increase.png" width="50%" /><img src="result/unique/03.all_decrease.png" width="50%" /></p>
</div>
<div id="overview-of-pathway-enrichment-analysis" class="section level3 hasAnchor" number="3.2.4">
<h3><span class="header-section-number">3.2.4</span> Overview of pathway enrichment analysis<a href="example-unique-analyses.html#overview-of-pathway-enrichment-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="example-unique-analyses.html#cb38-1" tabindex="-1"></a>meta_increase <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/unique/02.meta_increase.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-2"><a href="example-unique-analyses.html#cb38-2" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb38-3"><a href="example-unique-analyses.html#cb38-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(name,pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb38-4"><a href="example-unique-analyses.html#cb38-4" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type1 =</span><span class="st">&quot;meta_Cluster1&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-5"><a href="example-unique-analyses.html#cb38-5" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type2 =</span><span class="st">&quot;metabolite&quot;</span>)</span>
<span id="cb38-6"><a href="example-unique-analyses.html#cb38-6" tabindex="-1"></a>meta_decrease <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/unique/02.meta_decrease.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-7"><a href="example-unique-analyses.html#cb38-7" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb38-8"><a href="example-unique-analyses.html#cb38-8" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(name,pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb38-9"><a href="example-unique-analyses.html#cb38-9" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type1=</span><span class="st">&quot;meta_Cluster2&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-10"><a href="example-unique-analyses.html#cb38-10" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type2 =</span><span class="st">&quot;metabolite&quot;</span>)</span>
<span id="cb38-11"><a href="example-unique-analyses.html#cb38-11" tabindex="-1"></a></span>
<span id="cb38-12"><a href="example-unique-analyses.html#cb38-12" tabindex="-1"></a>gene_increase <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/unique/01.gene_increase.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-13"><a href="example-unique-analyses.html#cb38-13" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb38-14"><a href="example-unique-analyses.html#cb38-14" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(name,pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb38-15"><a href="example-unique-analyses.html#cb38-15" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type1 =</span><span class="st">&quot;gene_Cluster1&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-16"><a href="example-unique-analyses.html#cb38-16" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type2 =</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb38-17"><a href="example-unique-analyses.html#cb38-17" tabindex="-1"></a>gene_decrease <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/unique/01.gene_decrease.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-18"><a href="example-unique-analyses.html#cb38-18" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb38-19"><a href="example-unique-analyses.html#cb38-19" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(name,pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb38-20"><a href="example-unique-analyses.html#cb38-20" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type1=</span><span class="st">&quot;gene_Cluster2&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-21"><a href="example-unique-analyses.html#cb38-21" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type2 =</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb38-22"><a href="example-unique-analyses.html#cb38-22" tabindex="-1"></a></span>
<span id="cb38-23"><a href="example-unique-analyses.html#cb38-23" tabindex="-1"></a>all_increase <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/unique/03.all_increase.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-24"><a href="example-unique-analyses.html#cb38-24" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb38-25"><a href="example-unique-analyses.html#cb38-25" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(name,pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb38-26"><a href="example-unique-analyses.html#cb38-26" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type1 =</span><span class="st">&quot;all_Cluster1&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-27"><a href="example-unique-analyses.html#cb38-27" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type2 =</span><span class="st">&quot;all&quot;</span>)</span>
<span id="cb38-28"><a href="example-unique-analyses.html#cb38-28" tabindex="-1"></a>all_decrease <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/unique/03.all_decrease.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-29"><a href="example-unique-analyses.html#cb38-29" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb38-30"><a href="example-unique-analyses.html#cb38-30" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(name,pvalue) <span class="sc">%&gt;%</span></span>
<span id="cb38-31"><a href="example-unique-analyses.html#cb38-31" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type1=</span><span class="st">&quot;all_Cluster2&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-32"><a href="example-unique-analyses.html#cb38-32" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type2 =</span><span class="st">&quot;all&quot;</span>)</span>
<span id="cb38-33"><a href="example-unique-analyses.html#cb38-33" tabindex="-1"></a></span>
<span id="cb38-34"><a href="example-unique-analyses.html#cb38-34" tabindex="-1"></a>kegg_pathway_uniq <span class="ot">&lt;-</span> PathwayExtendData <span class="sc">%&gt;%</span></span>
<span id="cb38-35"><a href="example-unique-analyses.html#cb38-35" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(kegg_pathwayname,kegg_category) <span class="sc">%&gt;%</span></span>
<span id="cb38-36"><a href="example-unique-analyses.html#cb38-36" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;PATHWAY&quot;</span><span class="ot">=</span><span class="st">&quot;kegg_pathwayname&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-37"><a href="example-unique-analyses.html#cb38-37" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;pathway_type&quot;</span><span class="ot">=</span><span class="st">&quot;kegg_category&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-38"><a href="example-unique-analyses.html#cb38-38" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb38-39"><a href="example-unique-analyses.html#cb38-39" tabindex="-1"></a></span>
<span id="cb38-40"><a href="example-unique-analyses.html#cb38-40" tabindex="-1"></a>result_meta <span class="ot">&lt;-</span> <span class="fu">rbind</span>(meta_increase,meta_decrease) <span class="sc">%&gt;%</span></span>
<span id="cb38-41"><a href="example-unique-analyses.html#cb38-41" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pvalue <span class="sc">&lt;</span> <span class="fl">0.01</span>)</span>
<span id="cb38-42"><a href="example-unique-analyses.html#cb38-42" tabindex="-1"></a></span>
<span id="cb38-43"><a href="example-unique-analyses.html#cb38-43" tabindex="-1"></a>result_gene <span class="ot">&lt;-</span> <span class="fu">rbind</span>(gene_increase,gene_decrease) <span class="sc">%&gt;%</span></span>
<span id="cb38-44"><a href="example-unique-analyses.html#cb38-44" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pvalue <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb38-45"><a href="example-unique-analyses.html#cb38-45" tabindex="-1"></a></span>
<span id="cb38-46"><a href="example-unique-analyses.html#cb38-46" tabindex="-1"></a>result_all <span class="ot">&lt;-</span> <span class="fu">rbind</span>(all_increase,all_decrease) <span class="sc">%&gt;%</span></span>
<span id="cb38-47"><a href="example-unique-analyses.html#cb38-47" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pvalue <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb38-48"><a href="example-unique-analyses.html#cb38-48" tabindex="-1"></a></span>
<span id="cb38-49"><a href="example-unique-analyses.html#cb38-49" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">rbind</span>(result_meta,result_gene,result_all) <span class="sc">%&gt;%</span></span>
<span id="cb38-50"><a href="example-unique-analyses.html#cb38-50" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Sulfur metabolism&quot;</span>,<span class="st">&quot;Fructose and mannose metabolism&quot;</span>,<span class="st">&quot;Ascorbate and aldarate metabolism&quot;</span>,</span>
<span id="cb38-51"><a href="example-unique-analyses.html#cb38-51" tabindex="-1"></a>                             <span class="st">&quot;Galactose metabolism&quot;</span>,<span class="st">&quot;Butanoate metabolism&quot;</span>,<span class="st">&quot;Pentose and glucuronate interconversions&quot;</span>,</span>
<span id="cb38-52"><a href="example-unique-analyses.html#cb38-52" tabindex="-1"></a>                             <span class="st">&quot;Amino sugar and nucleotide sugar metabolism&quot;</span>,<span class="st">&quot;Propanoate metabolism&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb38-53"><a href="example-unique-analyses.html#cb38-53" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(kegg_pathway_uniq,<span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;name&quot;</span><span class="ot">=</span><span class="st">&quot;PATHWAY&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb38-54"><a href="example-unique-analyses.html#cb38-54" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway_type <span class="sc">!=</span> <span class="st">&quot;Glycan biosynthesis and metabolism&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-55"><a href="example-unique-analyses.html#cb38-55" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(pvalue)</span>
<span id="cb38-56"><a href="example-unique-analyses.html#cb38-56" tabindex="-1"></a></span>
<span id="cb38-57"><a href="example-unique-analyses.html#cb38-57" tabindex="-1"></a>pathway_hh <span class="ot">&lt;-</span> <span class="fu">unique</span>(result<span class="sc">$</span>pathway_type)</span>
<span id="cb38-58"><a href="example-unique-analyses.html#cb38-58" tabindex="-1"></a></span>
<span id="cb38-59"><a href="example-unique-analyses.html#cb38-59" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb38-60"><a href="example-unique-analyses.html#cb38-60" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>pathway_type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Metabolism of terpenoids and polyketides&quot;</span>,<span class="st">&quot;Metabolism of cofactors and vitamins&quot;</span>,<span class="st">&quot;Biosynthesis of other secondary metabolites&quot;</span>,<span class="st">&quot;Metabolism of other amino acids&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb38-61"><a href="example-unique-analyses.html#cb38-61" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">match</span>(pathway_type,pathway_hh))</span>
<span id="cb38-62"><a href="example-unique-analyses.html#cb38-62" tabindex="-1"></a></span>
<span id="cb38-63"><a href="example-unique-analyses.html#cb38-63" tabindex="-1"></a>result<span class="sc">$</span>name <span class="ot">&lt;-</span> <span class="fu">factor</span>(result<span class="sc">$</span>name,<span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">unique</span>(result<span class="sc">$</span>name)))</span>
<span id="cb38-64"><a href="example-unique-analyses.html#cb38-64" tabindex="-1"></a>result<span class="sc">$</span>pathway_type <span class="ot">&lt;-</span> <span class="fu">factor</span>(result<span class="sc">$</span>pathway_type,<span class="at">levels=</span><span class="fu">unique</span>(kegg_pathway_uniq<span class="sc">$</span>pathway_type))</span>
<span id="cb38-65"><a href="example-unique-analyses.html#cb38-65" tabindex="-1"></a>result<span class="sc">$</span>type1 <span class="ot">&lt;-</span> <span class="fu">factor</span>(result<span class="sc">$</span>type1,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;meta_Cluster1&quot;</span>,<span class="st">&quot;meta_Cluster2&quot;</span>,</span>
<span id="cb38-66"><a href="example-unique-analyses.html#cb38-66" tabindex="-1"></a>                                             <span class="st">&quot;gene_Cluster1&quot;</span>,<span class="st">&quot;gene_Cluster2&quot;</span>,<span class="st">&quot;all_Cluster1&quot;</span>,<span class="st">&quot;all_Cluster2&quot;</span>))</span>
<span id="cb38-67"><a href="example-unique-analyses.html#cb38-67" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(result,<span class="fu">aes</span>(name,<span class="sc">-</span><span class="fu">log10</span>(pvalue)))<span class="sc">+</span></span>
<span id="cb38-68"><a href="example-unique-analyses.html#cb38-68" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>,<span class="fu">aes</span>(<span class="at">fill=</span>pathway_type))<span class="sc">+</span></span>
<span id="cb38-69"><a href="example-unique-analyses.html#cb38-69" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;Set3&quot;</span>),<span class="at">name=</span><span class="st">&quot;Pathway Category&quot;</span>,</span>
<span id="cb38-70"><a href="example-unique-analyses.html#cb38-70" tabindex="-1"></a>                    <span class="at">breaks=</span><span class="fu">unique</span>(kegg_pathway_uniq<span class="sc">$</span>pathway_type))<span class="sc">+</span></span>
<span id="cb38-71"><a href="example-unique-analyses.html#cb38-71" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>))<span class="sc">+</span></span>
<span id="cb38-72"><a href="example-unique-analyses.html#cb38-72" tabindex="-1"></a>  <span class="fu">coord_flip</span>()<span class="sc">+</span></span>
<span id="cb38-73"><a href="example-unique-analyses.html#cb38-73" tabindex="-1"></a>  <span class="fu">facet_grid</span>(. <span class="sc">~</span> type1,<span class="at">scales=</span><span class="st">&quot;free_x&quot;</span>)<span class="sc">+</span></span>
<span id="cb38-74"><a href="example-unique-analyses.html#cb38-74" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb38-75"><a href="example-unique-analyses.html#cb38-75" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="cn">NULL</span>)</span>
<span id="cb38-76"><a href="example-unique-analyses.html#cb38-76" tabindex="-1"></a><span class="co">#p1</span></span>
<span id="cb38-77"><a href="example-unique-analyses.html#cb38-77" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/04.epe-v0102.pdf&quot;</span>,p1,<span class="at">width=</span><span class="dv">15</span>,<span class="at">height =</span> <span class="dv">7</span>)</span>
<span id="cb38-78"><a href="example-unique-analyses.html#cb38-78" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/04.epe-v0102.png&quot;</span>,p1,<span class="at">width=</span><span class="dv">15</span>,<span class="at">height =</span> <span class="dv">7</span>)</span></code></pre></div>
<p><img src="result/unique/04.epe-v0102.png" width="720" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="pathway-differential-abundance-score" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Pathway differential abundance score<a href="example-unique-analyses.html#pathway-differential-abundance-score" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="gene-differential-abundance-score" class="section level3 hasAnchor" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> gene differential abundance score<a href="example-unique-analyses.html#gene-differential-abundance-score" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>KEGG metabolic pathway-based differential abundance (DA) analysis investigate the dysregulation of metabolic pathways considering only genes.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="example-unique-analyses.html#cb39-1" tabindex="-1"></a>diff_gene_increase <span class="ot">&lt;-</span> all_gene <span class="sc">%&gt;%</span></span>
<span id="cb39-2"><a href="example-unique-analyses.html#cb39-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC <span class="sc">&gt;</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb39-3"><a href="example-unique-analyses.html#cb39-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.01</span>)</span>
<span id="cb39-4"><a href="example-unique-analyses.html#cb39-4" tabindex="-1"></a></span>
<span id="cb39-5"><a href="example-unique-analyses.html#cb39-5" tabindex="-1"></a>diff_gene_decrease <span class="ot">&lt;-</span> all_gene <span class="sc">%&gt;%</span></span>
<span id="cb39-6"><a href="example-unique-analyses.html#cb39-6" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb39-7"><a href="example-unique-analyses.html#cb39-7" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.01</span>)</span>
<span id="cb39-8"><a href="example-unique-analyses.html#cb39-8" tabindex="-1"></a></span>
<span id="cb39-9"><a href="example-unique-analyses.html#cb39-9" tabindex="-1"></a>DAscore_gene <span class="ot">&lt;-</span> <span class="fu">DAscore</span>(diff_gene_increase<span class="sc">$</span>name,diff_gene_decrease<span class="sc">$</span>name,all_gene<span class="sc">$</span>name,</span>
<span id="cb39-10"><a href="example-unique-analyses.html#cb39-10" tabindex="-1"></a>                        <span class="at">min_measured_num =</span> <span class="dv">10</span>,<span class="at">out=</span><span class="st">&quot;gene&quot;</span>,<span class="at">sort_plot=</span><span class="st">&quot;category&quot;</span>)</span>
<span id="cb39-11"><a href="example-unique-analyses.html#cb39-11" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/05.DA_gene.pdf&quot;</span>,DAscore_gene<span class="sc">$</span>p,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">8</span>)</span>
<span id="cb39-12"><a href="example-unique-analyses.html#cb39-12" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/05.DA_gene.png&quot;</span>,DAscore_gene<span class="sc">$</span>p,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">8</span>)</span>
<span id="cb39-13"><a href="example-unique-analyses.html#cb39-13" tabindex="-1"></a><span class="fu">write.table</span>(DAscore_gene<span class="sc">$</span>result,<span class="st">&quot;result/unique/05.DA_gene.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
<p><img src="result/unique/05.DA_gene.png" width="70%" style="display: block; margin: auto;" /></p>
</div>
<div id="metabolite-differential-abundance-score" class="section level3 hasAnchor" number="3.3.2">
<h3><span class="header-section-number">3.3.2</span> metabolite differential abundance score<a href="example-unique-analyses.html#metabolite-differential-abundance-score" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>KEGG metabolic pathway-based differential abundance (DA) analysis investigate the dysregulation of metabolic pathways considering only metabolites.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="example-unique-analyses.html#cb40-1" tabindex="-1"></a><span class="do">## metabolite</span></span>
<span id="cb40-2"><a href="example-unique-analyses.html#cb40-2" tabindex="-1"></a><span class="do">## filter the increase differential genes and metabolites</span></span>
<span id="cb40-3"><a href="example-unique-analyses.html#cb40-3" tabindex="-1"></a>diff_meta_increase <span class="ot">&lt;-</span> all_meta <span class="sc">%&gt;%</span></span>
<span id="cb40-4"><a href="example-unique-analyses.html#cb40-4" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC <span class="sc">&gt;</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-5"><a href="example-unique-analyses.html#cb40-5" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(P.Value <span class="sc">&lt;</span> <span class="fl">0.01</span>)</span>
<span id="cb40-6"><a href="example-unique-analyses.html#cb40-6" tabindex="-1"></a></span>
<span id="cb40-7"><a href="example-unique-analyses.html#cb40-7" tabindex="-1"></a>diff_meta_decrease <span class="ot">&lt;-</span> all_meta <span class="sc">%&gt;%</span></span>
<span id="cb40-8"><a href="example-unique-analyses.html#cb40-8" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC <span class="sc">&lt;</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-9"><a href="example-unique-analyses.html#cb40-9" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(P.Value <span class="sc">&lt;</span> <span class="fl">0.01</span>)</span>
<span id="cb40-10"><a href="example-unique-analyses.html#cb40-10" tabindex="-1"></a></span>
<span id="cb40-11"><a href="example-unique-analyses.html#cb40-11" tabindex="-1"></a>DAscore_meta <span class="ot">&lt;-</span> <span class="fu">DAscore</span>(diff_meta_increase<span class="sc">$</span>name,diff_meta_decrease<span class="sc">$</span>name,all_meta<span class="sc">$</span>name,</span>
<span id="cb40-12"><a href="example-unique-analyses.html#cb40-12" tabindex="-1"></a>                        <span class="at">min_measured_num =</span> <span class="dv">3</span>,<span class="at">out=</span><span class="st">&quot;metabolite&quot;</span>,<span class="at">sort_plot=</span><span class="st">&quot;category&quot;</span>)</span>
<span id="cb40-13"><a href="example-unique-analyses.html#cb40-13" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/06.DA_meta.pdf&quot;</span>,DAscore_meta<span class="sc">$</span>p,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">8</span>)</span>
<span id="cb40-14"><a href="example-unique-analyses.html#cb40-14" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/06.DA_meta.png&quot;</span>,DAscore_meta<span class="sc">$</span>p,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">8</span>)</span>
<span id="cb40-15"><a href="example-unique-analyses.html#cb40-15" tabindex="-1"></a><span class="fu">write.table</span>(DAscore_meta<span class="sc">$</span>result,<span class="st">&quot;result/unique/06.DA_meta.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
<p><img src="result/unique/06.DA_meta.png" width="70%" style="display: block; margin: auto;" /></p>
</div>
<div id="extended-pathway-differential-abundance-score" class="section level3 hasAnchor" number="3.3.3">
<h3><span class="header-section-number">3.3.3</span> Extended pathway differential abundance score<a href="example-unique-analyses.html#extended-pathway-differential-abundance-score" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>KEGG metabolic pathway-based extended differential abundance (EDA) analysis investigate the dysregulation of metabolic pathways considering both metabolites and genes.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="example-unique-analyses.html#cb41-1" tabindex="-1"></a><span class="do">## gene and metabolite</span></span>
<span id="cb41-2"><a href="example-unique-analyses.html#cb41-2" tabindex="-1"></a>DAscore_all <span class="ot">&lt;-</span> <span class="fu">DAscore</span>(<span class="fu">c</span>(diff_gene_increase<span class="sc">$</span>name,diff_meta_increase<span class="sc">$</span>name),</span>
<span id="cb41-3"><a href="example-unique-analyses.html#cb41-3" tabindex="-1"></a>                        <span class="fu">c</span>(diff_gene_decrease<span class="sc">$</span>name,diff_meta_decrease<span class="sc">$</span>name),</span>
<span id="cb41-4"><a href="example-unique-analyses.html#cb41-4" tabindex="-1"></a>                        <span class="fu">c</span>(all_gene<span class="sc">$</span>name,all_meta<span class="sc">$</span>name),<span class="at">min_measured_num =</span> <span class="dv">13</span>,</span>
<span id="cb41-5"><a href="example-unique-analyses.html#cb41-5" tabindex="-1"></a>                        <span class="at">out=</span><span class="st">&quot;Extended&quot;</span>,<span class="at">sort_plot=</span><span class="st">&quot;category&quot;</span>)</span>
<span id="cb41-6"><a href="example-unique-analyses.html#cb41-6" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/07.DA_all.pdf&quot;</span>,DAscore_all<span class="sc">$</span>p,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb41-7"><a href="example-unique-analyses.html#cb41-7" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/07.DA_all.png&quot;</span>,DAscore_all<span class="sc">$</span>p,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb41-8"><a href="example-unique-analyses.html#cb41-8" tabindex="-1"></a><span class="fu">write.table</span>(DAscore_all<span class="sc">$</span>result,<span class="st">&quot;result/unique/07.DA_all.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
<p><img src="result/unique/07.DA_all.png" width="70%" style="display: block; margin: auto;" /></p>
</div>
<div id="overview-of-pathway-differential-abundance-score" class="section level3 hasAnchor" number="3.3.4">
<h3><span class="header-section-number">3.3.4</span> Overview of pathway differential abundance score<a href="example-unique-analyses.html#overview-of-pathway-differential-abundance-score" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="example-unique-analyses.html#cb42-1" tabindex="-1"></a>meta <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/unique/06.DA_meta.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb42-2"><a href="example-unique-analyses.html#cb42-2" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb42-3"><a href="example-unique-analyses.html#cb42-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Measured_members_num <span class="sc">&gt;=</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb42-4"><a href="example-unique-analyses.html#cb42-4" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Type=</span><span class="st">&quot;metabolite&quot;</span>)</span>
<span id="cb42-5"><a href="example-unique-analyses.html#cb42-5" tabindex="-1"></a></span>
<span id="cb42-6"><a href="example-unique-analyses.html#cb42-6" tabindex="-1"></a>gene <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/unique/05.DA_gene.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb42-7"><a href="example-unique-analyses.html#cb42-7" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb42-8"><a href="example-unique-analyses.html#cb42-8" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Measured_members_num <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb42-9"><a href="example-unique-analyses.html#cb42-9" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Type=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb42-10"><a href="example-unique-analyses.html#cb42-10" tabindex="-1"></a></span>
<span id="cb42-11"><a href="example-unique-analyses.html#cb42-11" tabindex="-1"></a>all <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/unique/07.DA_all.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb42-12"><a href="example-unique-analyses.html#cb42-12" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb42-13"><a href="example-unique-analyses.html#cb42-13" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Measured_members_num <span class="sc">&gt;=</span> <span class="dv">13</span>) <span class="sc">%&gt;%</span></span>
<span id="cb42-14"><a href="example-unique-analyses.html#cb42-14" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Type=</span><span class="st">&quot;all&quot;</span>)</span>
<span id="cb42-15"><a href="example-unique-analyses.html#cb42-15" tabindex="-1"></a></span>
<span id="cb42-16"><a href="example-unique-analyses.html#cb42-16" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">rbind</span>(meta,gene,all) <span class="sc">%&gt;%</span></span>
<span id="cb42-17"><a href="example-unique-analyses.html#cb42-17" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>Pathway <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Sulfur metabolism&quot;</span>,<span class="st">&quot;Fructose and mannose metabolism&quot;</span>,<span class="st">&quot;Ascorbate and aldarate metabolism&quot;</span>,</span>
<span id="cb42-18"><a href="example-unique-analyses.html#cb42-18" tabindex="-1"></a>                                <span class="st">&quot;Galactose metabolism&quot;</span>,<span class="st">&quot;Butanoate metabolism&quot;</span>,<span class="st">&quot;Pentose and glucuronate interconversions&quot;</span>,<span class="st">&quot;Starch and sucrose metabolism&quot;</span>,</span>
<span id="cb42-19"><a href="example-unique-analyses.html#cb42-19" tabindex="-1"></a>                                <span class="st">&quot;Amino sugar and nucleotide sugar metabolism&quot;</span>,<span class="st">&quot;Propanoate metabolism&quot;</span>,<span class="st">&quot;Inositol phosphate metabolism&quot;</span>,</span>
<span id="cb42-20"><a href="example-unique-analyses.html#cb42-20" tabindex="-1"></a>                                <span class="st">&quot;Tryptophan metabolism&quot;</span>,</span>
<span id="cb42-21"><a href="example-unique-analyses.html#cb42-21" tabindex="-1"></a>                                <span class="st">&quot;Selenocompound metabolism&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb42-22"><a href="example-unique-analyses.html#cb42-22" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="st">`</span><span class="at">Pathway Category</span><span class="st">`</span> <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Metabolism of terpenoids and polyketides&quot;</span>,<span class="st">&quot;Metabolism of cofactors and vitamins&quot;</span>,</span>
<span id="cb42-23"><a href="example-unique-analyses.html#cb42-23" tabindex="-1"></a>                                           <span class="st">&quot;Biosynthesis of other secondary metabolites&quot;</span>,<span class="st">&quot;Xenobiotics biodegradation and metabolism&quot;</span>,</span>
<span id="cb42-24"><a href="example-unique-analyses.html#cb42-24" tabindex="-1"></a>                                           <span class="st">&quot;Glycan biosynthesis and metabolism&quot;</span>,<span class="st">&quot;Metabolism of other amino acids&quot;</span>))</span>
<span id="cb42-25"><a href="example-unique-analyses.html#cb42-25" tabindex="-1"></a></span>
<span id="cb42-26"><a href="example-unique-analyses.html#cb42-26" tabindex="-1"></a>result_filter <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb42-27"><a href="example-unique-analyses.html#cb42-27" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(DA_score) <span class="sc">%&gt;%</span></span>
<span id="cb42-28"><a href="example-unique-analyses.html#cb42-28" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="st">`</span><span class="at">Pathway Category</span><span class="st">`</span>)</span>
<span id="cb42-29"><a href="example-unique-analyses.html#cb42-29" tabindex="-1"></a>  </span>
<span id="cb42-30"><a href="example-unique-analyses.html#cb42-30" tabindex="-1"></a>result_filter<span class="sc">$</span>Pathway <span class="ot">&lt;-</span> <span class="fu">factor</span>(result_filter<span class="sc">$</span>Pathway,<span class="at">levels=</span><span class="fu">unique</span>(result_filter<span class="sc">$</span>Pathway))</span>
<span id="cb42-31"><a href="example-unique-analyses.html#cb42-31" tabindex="-1"></a></span>
<span id="cb42-32"><a href="example-unique-analyses.html#cb42-32" tabindex="-1"></a>kegg_pathway_uniq <span class="ot">&lt;-</span> PathwayExtendData <span class="sc">%&gt;%</span></span>
<span id="cb42-33"><a href="example-unique-analyses.html#cb42-33" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(kegg_pathwayname,kegg_category) <span class="sc">%&gt;%</span></span>
<span id="cb42-34"><a href="example-unique-analyses.html#cb42-34" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;PATHWAY&quot;</span><span class="ot">=</span><span class="st">&quot;kegg_pathwayname&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb42-35"><a href="example-unique-analyses.html#cb42-35" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;pathway_type&quot;</span><span class="ot">=</span><span class="st">&quot;kegg_category&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb42-36"><a href="example-unique-analyses.html#cb42-36" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb42-37"><a href="example-unique-analyses.html#cb42-37" tabindex="-1"></a></span>
<span id="cb42-38"><a href="example-unique-analyses.html#cb42-38" tabindex="-1"></a>result_filter<span class="sc">$</span>Type <span class="ot">&lt;-</span> <span class="fu">factor</span>(result_filter<span class="sc">$</span>Type,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;metabolite&quot;</span>,</span>
<span id="cb42-39"><a href="example-unique-analyses.html#cb42-39" tabindex="-1"></a>                                                         <span class="st">&quot;gene&quot;</span>,<span class="st">&quot;all&quot;</span>))</span>
<span id="cb42-40"><a href="example-unique-analyses.html#cb42-40" tabindex="-1"></a></span>
<span id="cb42-41"><a href="example-unique-analyses.html#cb42-41" tabindex="-1"></a></span>
<span id="cb42-42"><a href="example-unique-analyses.html#cb42-42" tabindex="-1"></a>p <span class="ot">&lt;-</span> ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(result_filter)<span class="sc">+</span></span>
<span id="cb42-43"><a href="example-unique-analyses.html#cb42-43" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">x=</span>Pathway,<span class="at">y=</span>DA_score,<span class="at">size=</span><span class="fu">log2</span>(Measured_members_num),<span class="at">color=</span><span class="st">`</span><span class="at">Pathway Category</span><span class="st">`</span>))<span class="sc">+</span></span>
<span id="cb42-44"><a href="example-unique-analyses.html#cb42-44" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_pointrange</span>(ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">x=</span>Pathway,<span class="at">y=</span>DA_score,<span class="at">ymin=</span><span class="dv">0</span>,<span class="at">ymax=</span>DA_score,<span class="at">color=</span><span class="st">`</span><span class="at">Pathway Category</span><span class="st">`</span>))<span class="sc">+</span></span>
<span id="cb42-45"><a href="example-unique-analyses.html#cb42-45" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;Set3&quot;</span>),<span class="at">name=</span><span class="st">&quot;Pathway Category&quot;</span>,</span>
<span id="cb42-46"><a href="example-unique-analyses.html#cb42-46" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">unique</span>(kegg_pathway_uniq<span class="sc">$</span>pathway_type))<span class="sc">+</span></span>
<span id="cb42-47"><a href="example-unique-analyses.html#cb42-47" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">coord_flip</span>()<span class="sc">+</span></span>
<span id="cb42-48"><a href="example-unique-analyses.html#cb42-48" tabindex="-1"></a>  <span class="co">#      ggplot2::ylab(&quot;DA score&quot;)+</span></span>
<span id="cb42-49"><a href="example-unique-analyses.html#cb42-49" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">xlab</span>(<span class="cn">NULL</span>)<span class="sc">+</span></span>
<span id="cb42-50"><a href="example-unique-analyses.html#cb42-50" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb42-51"><a href="example-unique-analyses.html#cb42-51" tabindex="-1"></a>  <span class="fu">facet_grid</span>(. <span class="sc">~</span> Type,<span class="at">scales=</span><span class="st">&quot;free_x&quot;</span>)</span>
<span id="cb42-52"><a href="example-unique-analyses.html#cb42-52" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/08.DA-v0102.pdf&quot;</span>,p,<span class="at">width=</span><span class="dv">18</span>,<span class="at">height =</span> <span class="dv">7</span>)</span>
<span id="cb42-53"><a href="example-unique-analyses.html#cb42-53" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/08.DA-v0102.png&quot;</span>,p,<span class="at">width=</span><span class="dv">18</span>,<span class="at">height =</span> <span class="dv">7</span>)</span></code></pre></div>
<p><img src="result/unique/08.DA-v0102.png" width="864" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="esea-metabolite-set-enrichment-analysis" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> ESEA Metabolite Set Enrichment Analysis<a href="example-unique-analyses.html#esea-metabolite-set-enrichment-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The metabolite set enrichment analysis using function “ESEA”.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="example-unique-analyses.html#cb43-1" tabindex="-1"></a>all_result <span class="ot">&lt;-</span> <span class="fu">rbind</span>(all_meta,all_gene)</span>
<span id="cb43-2"><a href="example-unique-analyses.html#cb43-2" tabindex="-1"></a></span>
<span id="cb43-3"><a href="example-unique-analyses.html#cb43-3" tabindex="-1"></a>all <span class="ot">&lt;-</span> all_result<span class="sc">$</span>logFC</span>
<span id="cb43-4"><a href="example-unique-analyses.html#cb43-4" tabindex="-1"></a><span class="fu">names</span>(all) <span class="ot">&lt;-</span> all_result<span class="sc">$</span>name</span>
<span id="cb43-5"><a href="example-unique-analyses.html#cb43-5" tabindex="-1"></a></span>
<span id="cb43-6"><a href="example-unique-analyses.html#cb43-6" tabindex="-1"></a>gene <span class="ot">&lt;-</span> all_gene<span class="sc">$</span>logFC</span>
<span id="cb43-7"><a href="example-unique-analyses.html#cb43-7" tabindex="-1"></a><span class="fu">names</span>(gene) <span class="ot">&lt;-</span> all_gene<span class="sc">$</span>name</span>
<span id="cb43-8"><a href="example-unique-analyses.html#cb43-8" tabindex="-1"></a></span>
<span id="cb43-9"><a href="example-unique-analyses.html#cb43-9" tabindex="-1"></a>meta <span class="ot">&lt;-</span> all_meta<span class="sc">$</span>logFC</span>
<span id="cb43-10"><a href="example-unique-analyses.html#cb43-10" tabindex="-1"></a><span class="fu">names</span>(meta) <span class="ot">&lt;-</span> all_meta<span class="sc">$</span>name</span>
<span id="cb43-11"><a href="example-unique-analyses.html#cb43-11" tabindex="-1"></a></span>
<span id="cb43-12"><a href="example-unique-analyses.html#cb43-12" tabindex="-1"></a>result_meta <span class="ot">&lt;-</span> <span class="fu">ESEA</span>(meta,<span class="at">out=</span><span class="st">&quot;metabolite&quot;</span>,<span class="at">minSize =</span> <span class="dv">3</span>)</span>
<span id="cb43-13"><a href="example-unique-analyses.html#cb43-13" tabindex="-1"></a>result_gene <span class="ot">&lt;-</span> <span class="fu">ESEA</span>(gene,<span class="at">out =</span> <span class="st">&quot;gene&quot;</span>,<span class="at">minSize =</span> <span class="dv">10</span>)</span>
<span id="cb43-14"><a href="example-unique-analyses.html#cb43-14" tabindex="-1"></a>result_all <span class="ot">&lt;-</span> <span class="fu">ESEA</span>(all,<span class="at">out=</span><span class="st">&quot;Extended&quot;</span>,<span class="at">minSize =</span> <span class="dv">13</span>)</span>
<span id="cb43-15"><a href="example-unique-analyses.html#cb43-15" tabindex="-1"></a></span>
<span id="cb43-16"><a href="example-unique-analyses.html#cb43-16" tabindex="-1"></a></span>
<span id="cb43-17"><a href="example-unique-analyses.html#cb43-17" tabindex="-1"></a>kegg_pathway_uniq <span class="ot">&lt;-</span> PathwayExtendData <span class="sc">%&gt;%</span></span>
<span id="cb43-18"><a href="example-unique-analyses.html#cb43-18" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(kegg_pathwayname,kegg_category) <span class="sc">%&gt;%</span></span>
<span id="cb43-19"><a href="example-unique-analyses.html#cb43-19" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;PATHWAY&quot;</span><span class="ot">=</span><span class="st">&quot;kegg_pathwayname&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-20"><a href="example-unique-analyses.html#cb43-20" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;pathway_type&quot;</span><span class="ot">=</span><span class="st">&quot;kegg_category&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-21"><a href="example-unique-analyses.html#cb43-21" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb43-22"><a href="example-unique-analyses.html#cb43-22" tabindex="-1"></a></span>
<span id="cb43-23"><a href="example-unique-analyses.html#cb43-23" tabindex="-1"></a>result_meta_filter <span class="ot">&lt;-</span> result_meta <span class="sc">%&gt;%</span></span>
<span id="cb43-24"><a href="example-unique-analyses.html#cb43-24" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pval <span class="sc">&lt;</span> <span class="fl">0.5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-25"><a href="example-unique-analyses.html#cb43-25" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Type=</span><span class="fu">ifelse</span>(NES<span class="sc">&gt;</span><span class="dv">0</span>,<span class="st">&quot;meta_Cluster1&quot;</span>,<span class="st">&quot;meta_Cluster2&quot;</span>))</span>
<span id="cb43-26"><a href="example-unique-analyses.html#cb43-26" tabindex="-1"></a></span>
<span id="cb43-27"><a href="example-unique-analyses.html#cb43-27" tabindex="-1"></a>result_gene_filter <span class="ot">&lt;-</span> result_gene <span class="sc">%&gt;%</span></span>
<span id="cb43-28"><a href="example-unique-analyses.html#cb43-28" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pval <span class="sc">&lt;</span> <span class="fl">0.2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-29"><a href="example-unique-analyses.html#cb43-29" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Type=</span><span class="fu">ifelse</span>(NES<span class="sc">&gt;</span><span class="dv">0</span>,<span class="st">&quot;gene_Cluster1&quot;</span>,<span class="st">&quot;gene_Cluster2&quot;</span>))</span>
<span id="cb43-30"><a href="example-unique-analyses.html#cb43-30" tabindex="-1"></a></span>
<span id="cb43-31"><a href="example-unique-analyses.html#cb43-31" tabindex="-1"></a>result_all_filter <span class="ot">&lt;-</span> result_all <span class="sc">%&gt;%</span></span>
<span id="cb43-32"><a href="example-unique-analyses.html#cb43-32" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pval <span class="sc">&lt;</span> <span class="fl">0.2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-33"><a href="example-unique-analyses.html#cb43-33" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Type=</span><span class="fu">ifelse</span>(NES<span class="sc">&gt;</span><span class="dv">0</span>,<span class="st">&quot;all_Cluster1&quot;</span>,<span class="st">&quot;all_Cluster2&quot;</span>))</span>
<span id="cb43-34"><a href="example-unique-analyses.html#cb43-34" tabindex="-1"></a></span>
<span id="cb43-35"><a href="example-unique-analyses.html#cb43-35" tabindex="-1"></a></span>
<span id="cb43-36"><a href="example-unique-analyses.html#cb43-36" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">rbind</span>(result_meta_filter,result_gene_filter,result_all_filter)</span>
<span id="cb43-37"><a href="example-unique-analyses.html#cb43-37" tabindex="-1"></a></span>
<span id="cb43-38"><a href="example-unique-analyses.html#cb43-38" tabindex="-1"></a>result_filter <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb43-39"><a href="example-unique-analyses.html#cb43-39" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(kegg_pathway_uniq,<span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;pathway&quot;</span><span class="ot">=</span><span class="st">&quot;PATHWAY&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb43-40"><a href="example-unique-analyses.html#cb43-40" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway_type <span class="sc">!=</span> <span class="st">&quot;Glycan biosynthesis and metabolism&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-41"><a href="example-unique-analyses.html#cb43-41" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>pathway_type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Metabolism of terpenoids and polyketides&quot;</span>,<span class="st">&quot;Metabolism of cofactors and vitamins&quot;</span>,</span>
<span id="cb43-42"><a href="example-unique-analyses.html#cb43-42" tabindex="-1"></a>                                   <span class="st">&quot;Biosynthesis of other secondary metabolites&quot;</span>,<span class="st">&quot;Xenobiotics biodegradation and metabolism&quot;</span>,<span class="st">&quot;Metabolism of other amino acids&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb43-43"><a href="example-unique-analyses.html#cb43-43" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(pval)</span>
<span id="cb43-44"><a href="example-unique-analyses.html#cb43-44" tabindex="-1"></a></span>
<span id="cb43-45"><a href="example-unique-analyses.html#cb43-45" tabindex="-1"></a>pathway_hh <span class="ot">&lt;-</span> <span class="fu">unique</span>(result_filter<span class="sc">$</span>pathway_type)</span>
<span id="cb43-46"><a href="example-unique-analyses.html#cb43-46" tabindex="-1"></a></span>
<span id="cb43-47"><a href="example-unique-analyses.html#cb43-47" tabindex="-1"></a>result_filter <span class="ot">&lt;-</span> result_filter <span class="sc">%&gt;%</span></span>
<span id="cb43-48"><a href="example-unique-analyses.html#cb43-48" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">match</span>(pathway_type,pathway_hh))</span>
<span id="cb43-49"><a href="example-unique-analyses.html#cb43-49" tabindex="-1"></a></span>
<span id="cb43-50"><a href="example-unique-analyses.html#cb43-50" tabindex="-1"></a>result_filter<span class="sc">$</span>pathway <span class="ot">&lt;-</span> <span class="fu">factor</span>(result_filter<span class="sc">$</span>pathway,<span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">unique</span>(result_filter<span class="sc">$</span>pathway)))</span>
<span id="cb43-51"><a href="example-unique-analyses.html#cb43-51" tabindex="-1"></a>result_filter<span class="sc">$</span>pathway_type <span class="ot">&lt;-</span> <span class="fu">factor</span>(result_filter<span class="sc">$</span>pathway_type,<span class="at">levels=</span><span class="fu">unique</span>(kegg_pathway_uniq<span class="sc">$</span>pathway_type))</span>
<span id="cb43-52"><a href="example-unique-analyses.html#cb43-52" tabindex="-1"></a>result_filter<span class="sc">$</span>Type <span class="ot">&lt;-</span> <span class="fu">factor</span>(result_filter<span class="sc">$</span>Type,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;meta_Cluster1&quot;</span>,<span class="st">&quot;meta_Cluster2&quot;</span>,</span>
<span id="cb43-53"><a href="example-unique-analyses.html#cb43-53" tabindex="-1"></a>                                             <span class="st">&quot;gene_Cluster1&quot;</span>,<span class="st">&quot;gene_Cluster2&quot;</span>,<span class="st">&quot;all_Cluster1&quot;</span>,<span class="st">&quot;all_Cluster2&quot;</span>))</span>
<span id="cb43-54"><a href="example-unique-analyses.html#cb43-54" tabindex="-1"></a></span>
<span id="cb43-55"><a href="example-unique-analyses.html#cb43-55" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(result_filter,<span class="fu">aes</span>(pathway,<span class="sc">-</span><span class="fu">log10</span>(pval))) <span class="sc">+</span></span>
<span id="cb43-56"><a href="example-unique-analyses.html#cb43-56" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>,<span class="fu">aes</span>(<span class="at">fill=</span>pathway_type))<span class="sc">+</span></span>
<span id="cb43-57"><a href="example-unique-analyses.html#cb43-57" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;Set3&quot;</span>),<span class="at">name=</span><span class="st">&quot;Pathway Category&quot;</span>,</span>
<span id="cb43-58"><a href="example-unique-analyses.html#cb43-58" tabindex="-1"></a>                    <span class="at">breaks=</span><span class="fu">unique</span>(kegg_pathway_uniq<span class="sc">$</span>pathway_type))<span class="sc">+</span></span>
<span id="cb43-59"><a href="example-unique-analyses.html#cb43-59" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>))<span class="sc">+</span></span>
<span id="cb43-60"><a href="example-unique-analyses.html#cb43-60" tabindex="-1"></a>  <span class="fu">facet_grid</span>(. <span class="sc">~</span> Type,<span class="at">scales=</span><span class="st">&quot;free_x&quot;</span>)<span class="sc">+</span></span>
<span id="cb43-61"><a href="example-unique-analyses.html#cb43-61" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb43-62"><a href="example-unique-analyses.html#cb43-62" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="cn">NULL</span>)<span class="sc">+</span></span>
<span id="cb43-63"><a href="example-unique-analyses.html#cb43-63" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb43-64"><a href="example-unique-analyses.html#cb43-64" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/09.ESEA_facet.pdf&quot;</span>,p,<span class="at">width=</span><span class="dv">15</span>,<span class="at">height =</span> <span class="dv">7</span>)</span>
<span id="cb43-65"><a href="example-unique-analyses.html#cb43-65" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/09.ESEA_facet.png&quot;</span>,p,<span class="at">width=</span><span class="dv">15</span>,<span class="at">height =</span> <span class="dv">7</span>)</span></code></pre></div>
<p><img src="result/unique/09.ESEA_facet.png" width="720" style="display: block; margin: auto;" /></p>
<p>The MSEA plot of interested pathway.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="example-unique-analyses.html#cb44-1" tabindex="-1"></a>all_meta <span class="ot">&lt;-</span> <span class="fu">mlimma</span>(meta_dat,group) <span class="sc">%&gt;%</span></span>
<span id="cb44-2"><a href="example-unique-analyses.html#cb44-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(logFC)</span>
<span id="cb44-3"><a href="example-unique-analyses.html#cb44-3" tabindex="-1"></a>all_gene <span class="ot">&lt;-</span> <span class="fu">mlimma</span>(gene_dat,group) <span class="sc">%&gt;%</span></span>
<span id="cb44-4"><a href="example-unique-analyses.html#cb44-4" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(logFC)</span>
<span id="cb44-5"><a href="example-unique-analyses.html#cb44-5" tabindex="-1"></a>all_result <span class="ot">&lt;-</span> <span class="fu">rbind</span>(all_meta,all_gene) <span class="sc">%&gt;%</span></span>
<span id="cb44-6"><a href="example-unique-analyses.html#cb44-6" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(logFC)</span>
<span id="cb44-7"><a href="example-unique-analyses.html#cb44-7" tabindex="-1"></a></span>
<span id="cb44-8"><a href="example-unique-analyses.html#cb44-8" tabindex="-1"></a>all <span class="ot">&lt;-</span> all_result<span class="sc">$</span>logFC</span>
<span id="cb44-9"><a href="example-unique-analyses.html#cb44-9" tabindex="-1"></a><span class="fu">names</span>(all) <span class="ot">&lt;-</span> all_result<span class="sc">$</span>name</span>
<span id="cb44-10"><a href="example-unique-analyses.html#cb44-10" tabindex="-1"></a></span>
<span id="cb44-11"><a href="example-unique-analyses.html#cb44-11" tabindex="-1"></a>gene <span class="ot">&lt;-</span> all_gene<span class="sc">$</span>logFC</span>
<span id="cb44-12"><a href="example-unique-analyses.html#cb44-12" tabindex="-1"></a><span class="fu">names</span>(gene) <span class="ot">&lt;-</span> all_gene<span class="sc">$</span>name</span>
<span id="cb44-13"><a href="example-unique-analyses.html#cb44-13" tabindex="-1"></a></span>
<span id="cb44-14"><a href="example-unique-analyses.html#cb44-14" tabindex="-1"></a>meta <span class="ot">&lt;-</span> all_meta<span class="sc">$</span>logFC</span>
<span id="cb44-15"><a href="example-unique-analyses.html#cb44-15" tabindex="-1"></a><span class="fu">names</span>(meta) <span class="ot">&lt;-</span> all_meta<span class="sc">$</span>name</span>
<span id="cb44-16"><a href="example-unique-analyses.html#cb44-16" tabindex="-1"></a></span>
<span id="cb44-17"><a href="example-unique-analyses.html#cb44-17" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">pMSEA</span>(<span class="st">&quot;Oxidative phosphorylation&quot;</span>,gene,<span class="at">out=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb44-18"><a href="example-unique-analyses.html#cb44-18" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/10.plot_msea_oxipho_gene.pdf&quot;</span>,p1,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb44-19"><a href="example-unique-analyses.html#cb44-19" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/10.plot_msea_oxipho_gene.png&quot;</span>,p1,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb44-20"><a href="example-unique-analyses.html#cb44-20" tabindex="-1"></a></span>
<span id="cb44-21"><a href="example-unique-analyses.html#cb44-21" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">pMSEA</span>(<span class="st">&quot;Oxidative phosphorylation&quot;</span>,meta,<span class="at">out=</span><span class="st">&quot;metabolite&quot;</span>)</span>
<span id="cb44-22"><a href="example-unique-analyses.html#cb44-22" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/10.plot_msea_oxipho_metabolite.pdf&quot;</span>,p2,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb44-23"><a href="example-unique-analyses.html#cb44-23" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/10.plot_msea_oxipho_metabolite.png&quot;</span>,p2,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb44-24"><a href="example-unique-analyses.html#cb44-24" tabindex="-1"></a></span>
<span id="cb44-25"><a href="example-unique-analyses.html#cb44-25" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">pMSEA</span>(<span class="st">&quot;Oxidative phosphorylation&quot;</span>,all,<span class="at">out=</span><span class="st">&quot;Extended&quot;</span>)</span>
<span id="cb44-26"><a href="example-unique-analyses.html#cb44-26" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/10.plot_msea_oxipho_all.pdf&quot;</span>,p3,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb44-27"><a href="example-unique-analyses.html#cb44-27" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/10.plot_msea_oxipho_all.png&quot;</span>,p3,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb44-28"><a href="example-unique-analyses.html#cb44-28" tabindex="-1"></a></span>
<span id="cb44-29"><a href="example-unique-analyses.html#cb44-29" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">pMSEA</span>(<span class="st">&quot;Fatty acid biosynthesis&quot;</span>,meta,<span class="at">out=</span><span class="st">&quot;metabolite&quot;</span>)</span>
<span id="cb44-30"><a href="example-unique-analyses.html#cb44-30" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/10.plot_msea_Fatty acid biosynthesis_metabolite.pdf&quot;</span>,p,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb44-31"><a href="example-unique-analyses.html#cb44-31" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/10.plot_msea_Fatty acid biosynthesis_metabolite.png&quot;</span>,p,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="dv">5</span>)</span></code></pre></div>
<p><img src="result/unique/10.plot_msea_Fatty%20acid%20biosynthesis_metabolite.png" width="50%" style="display: block; margin: auto;" /></p>
</div>
<div id="subnetwork" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> Subnetwork<a href="example-unique-analyses.html#subnetwork" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="example-unique-analyses.html#cb45-1" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb45-2"><a href="example-unique-analyses.html#cb45-2" tabindex="-1"></a><span class="fu">library</span>(MNet)</span>
<span id="cb45-3"><a href="example-unique-analyses.html#cb45-3" tabindex="-1"></a></span>
<span id="cb45-4"><a href="example-unique-analyses.html#cb45-4" tabindex="-1"></a>diff_meta <span class="ot">&lt;-</span> <span class="fu">mlimma</span>(meta_dat,group)</span>
<span id="cb45-5"><a href="example-unique-analyses.html#cb45-5" tabindex="-1"></a>diff_gene <span class="ot">&lt;-</span> <span class="fu">mlimma</span>(gene_dat,group)</span>
<span id="cb45-6"><a href="example-unique-analyses.html#cb45-6" tabindex="-1"></a><span class="fu">names</span>(diff_meta)[<span class="dv">4</span>]  <span class="ot">&lt;-</span> <span class="st">&quot;p_value&quot;</span></span>
<span id="cb45-7"><a href="example-unique-analyses.html#cb45-7" tabindex="-1"></a><span class="fu">names</span>(diff_gene)[<span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="st">&quot;p_value&quot;</span></span>
<span id="cb45-8"><a href="example-unique-analyses.html#cb45-8" tabindex="-1"></a></span>
<span id="cb45-9"><a href="example-unique-analyses.html#cb45-9" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">&quot;result/unique/11.pdent.pdf&quot;</span>,<span class="at">width=</span><span class="dv">12</span>,<span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb45-10"><a href="example-unique-analyses.html#cb45-10" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">pdnet</span>(diff_meta,diff_gene,<span class="at">nsize=</span><span class="dv">60</span>)</span>
<span id="cb45-11"><a href="example-unique-analyses.html#cb45-11" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb45-12"><a href="example-unique-analyses.html#cb45-12" tabindex="-1"></a></span>
<span id="cb45-13"><a href="example-unique-analyses.html#cb45-13" tabindex="-1"></a><span class="fu">png</span>(<span class="st">&quot;result/unique/11.pdent.png&quot;</span>,<span class="at">width=</span><span class="dv">1000</span>,<span class="at">height =</span> <span class="dv">800</span>)</span>
<span id="cb45-14"><a href="example-unique-analyses.html#cb45-14" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">pdnet</span>(diff_meta,diff_gene,<span class="at">nsize=</span><span class="dv">60</span>)</span>
<span id="cb45-15"><a href="example-unique-analyses.html#cb45-15" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb45-16"><a href="example-unique-analyses.html#cb45-16" tabindex="-1"></a></span>
<span id="cb45-17"><a href="example-unique-analyses.html#cb45-17" tabindex="-1"></a>node <span class="ot">&lt;-</span> a<span class="sc">$</span>node_result</span>
<span id="cb45-18"><a href="example-unique-analyses.html#cb45-18" tabindex="-1"></a><span class="fu">write.table</span>(node,<span class="st">&quot;result/unique/11.pdnet_node.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb45-19"><a href="example-unique-analyses.html#cb45-19" tabindex="-1"></a></span>
<span id="cb45-20"><a href="example-unique-analyses.html#cb45-20" tabindex="-1"></a>edge <span class="ot">&lt;-</span> a<span class="sc">$</span>edge_result</span>
<span id="cb45-21"><a href="example-unique-analyses.html#cb45-21" tabindex="-1"></a><span class="fu">write.table</span>(edge,<span class="st">&quot;result/unique/11.pdnet_edge.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb45-22"><a href="example-unique-analyses.html#cb45-22" tabindex="-1"></a></span>
<span id="cb45-23"><a href="example-unique-analyses.html#cb45-23" tabindex="-1"></a>metabolite <span class="ot">&lt;-</span> node <span class="sc">%&gt;%</span></span>
<span id="cb45-24"><a href="example-unique-analyses.html#cb45-24" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(type<span class="sc">==</span><span class="st">&quot;metabolite&quot;</span>)</span>
<span id="cb45-25"><a href="example-unique-analyses.html#cb45-25" tabindex="-1"></a></span>
<span id="cb45-26"><a href="example-unique-analyses.html#cb45-26" tabindex="-1"></a>gene <span class="ot">&lt;-</span> node <span class="sc">%&gt;%</span></span>
<span id="cb45-27"><a href="example-unique-analyses.html#cb45-27" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(type<span class="sc">==</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb45-28"><a href="example-unique-analyses.html#cb45-28" tabindex="-1"></a></span>
<span id="cb45-29"><a href="example-unique-analyses.html#cb45-29" tabindex="-1"></a>result_metabolite <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(metabolite<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;metabolite&quot;</span>,<span class="at">test=</span><span class="st">&quot;hypergeo&quot;</span>)</span>
<span id="cb45-30"><a href="example-unique-analyses.html#cb45-30" tabindex="-1"></a>result_gene <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(gene<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;gene&quot;</span>,<span class="at">test=</span><span class="st">&quot;hypergeo&quot;</span>)</span>
<span id="cb45-31"><a href="example-unique-analyses.html#cb45-31" tabindex="-1"></a>result_all <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(<span class="fu">c</span>(gene<span class="sc">$</span>name,metabolite<span class="sc">$</span>name),<span class="at">out=</span><span class="st">&quot;Extended&quot;</span>,<span class="at">test=</span><span class="st">&quot;hypergeo&quot;</span>)</span>
<span id="cb45-32"><a href="example-unique-analyses.html#cb45-32" tabindex="-1"></a></span>
<span id="cb45-33"><a href="example-unique-analyses.html#cb45-33" tabindex="-1"></a><span class="fu">write.table</span>(result_metabolite<span class="sc">$</span>output,<span class="st">&quot;result/unique/11.pdnet_metabolite_enrichment.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names =</span> F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb45-34"><a href="example-unique-analyses.html#cb45-34" tabindex="-1"></a><span class="fu">write.table</span>(result_gene<span class="sc">$</span>output,<span class="st">&quot;result/unique/11.pdnet_gene_enrichment.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names =</span> F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb45-35"><a href="example-unique-analyses.html#cb45-35" tabindex="-1"></a><span class="fu">write.table</span>(result_all<span class="sc">$</span>output,<span class="st">&quot;result/unique/11.pdnet_all_enrichment.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names =</span> F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb45-36"><a href="example-unique-analyses.html#cb45-36" tabindex="-1"></a></span>
<span id="cb45-37"><a href="example-unique-analyses.html#cb45-37" tabindex="-1"></a>kegg_pathway_uniq <span class="ot">&lt;-</span> PathwayExtendData <span class="sc">%&gt;%</span></span>
<span id="cb45-38"><a href="example-unique-analyses.html#cb45-38" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(kegg_pathwayname,kegg_category) <span class="sc">%&gt;%</span></span>
<span id="cb45-39"><a href="example-unique-analyses.html#cb45-39" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;PATHWAY&quot;</span><span class="ot">=</span><span class="st">&quot;kegg_pathwayname&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-40"><a href="example-unique-analyses.html#cb45-40" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;pathway_type&quot;</span><span class="ot">=</span><span class="st">&quot;kegg_category&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-41"><a href="example-unique-analyses.html#cb45-41" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb45-42"><a href="example-unique-analyses.html#cb45-42" tabindex="-1"></a></span>
<span id="cb45-43"><a href="example-unique-analyses.html#cb45-43" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">rbind</span>(result_metabolite<span class="sc">$</span>output <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type=</span><span class="st">&quot;metabolite&quot;</span>),</span>
<span id="cb45-44"><a href="example-unique-analyses.html#cb45-44" tabindex="-1"></a>                    result_gene<span class="sc">$</span>output <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type=</span><span class="st">&quot;gene&quot;</span>),</span>
<span id="cb45-45"><a href="example-unique-analyses.html#cb45-45" tabindex="-1"></a>                    result_all<span class="sc">$</span>output <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type=</span><span class="st">&quot;all&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb45-46"><a href="example-unique-analyses.html#cb45-46" tabindex="-1"></a> <span class="co"># dplyr::filter(nOverlap &gt; 1) %&gt;%</span></span>
<span id="cb45-47"><a href="example-unique-analyses.html#cb45-47" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(kegg_pathway_uniq,<span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;name&quot;</span><span class="ot">=</span><span class="st">&quot;PATHWAY&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb45-48"><a href="example-unique-analyses.html#cb45-48" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway_type <span class="sc">!=</span> <span class="st">&quot;Glycan biosynthesis and metabolism&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-49"><a href="example-unique-analyses.html#cb45-49" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Fructose and mannose metabolism&quot;</span>,<span class="st">&quot;Amino sugar and nucleotide sugar metabolism&quot;</span>,</span>
<span id="cb45-50"><a href="example-unique-analyses.html#cb45-50" tabindex="-1"></a>                            <span class="st">&quot;Galactose metabolism&quot;</span>,<span class="st">&quot;Ascorbate and aldarate metabolism&quot;</span>,<span class="st">&quot;Pentose and glucuronate interconversions&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb45-51"><a href="example-unique-analyses.html#cb45-51" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(pvalue)</span>
<span id="cb45-52"><a href="example-unique-analyses.html#cb45-52" tabindex="-1"></a></span>
<span id="cb45-53"><a href="example-unique-analyses.html#cb45-53" tabindex="-1"></a>pathway_hh <span class="ot">&lt;-</span> <span class="fu">unique</span>(result<span class="sc">$</span>pathway_type)</span>
<span id="cb45-54"><a href="example-unique-analyses.html#cb45-54" tabindex="-1"></a></span>
<span id="cb45-55"><a href="example-unique-analyses.html#cb45-55" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb45-56"><a href="example-unique-analyses.html#cb45-56" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>pathway_type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Metabolism of terpenoids and polyketides&quot;</span>,<span class="st">&quot;Metabolism of cofactors and vitamins&quot;</span>,</span>
<span id="cb45-57"><a href="example-unique-analyses.html#cb45-57" tabindex="-1"></a>                                     <span class="st">&quot;Biosynthesis of other secondary metabolites&quot;</span>,<span class="st">&quot;Metabolism of other amino acids&quot;</span>,</span>
<span id="cb45-58"><a href="example-unique-analyses.html#cb45-58" tabindex="-1"></a>                                     <span class="st">&quot;Xenobiotics biodegradation and metabolism&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb45-59"><a href="example-unique-analyses.html#cb45-59" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">match</span>(pathway_type,pathway_hh))</span>
<span id="cb45-60"><a href="example-unique-analyses.html#cb45-60" tabindex="-1"></a></span>
<span id="cb45-61"><a href="example-unique-analyses.html#cb45-61" tabindex="-1"></a>result<span class="sc">$</span>name <span class="ot">&lt;-</span> <span class="fu">factor</span>(result<span class="sc">$</span>name,<span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">unique</span>(result<span class="sc">$</span>name)))</span>
<span id="cb45-62"><a href="example-unique-analyses.html#cb45-62" tabindex="-1"></a>result<span class="sc">$</span>pathway_type <span class="ot">&lt;-</span> <span class="fu">factor</span>(result<span class="sc">$</span>pathway_type,<span class="at">levels=</span><span class="fu">unique</span>(kegg_pathway_uniq<span class="sc">$</span>pathway_type))</span>
<span id="cb45-63"><a href="example-unique-analyses.html#cb45-63" tabindex="-1"></a>result<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="fu">factor</span>(result<span class="sc">$</span>type,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;metabolite&quot;</span>,<span class="st">&quot;gene&quot;</span>,<span class="st">&quot;all&quot;</span>))</span>
<span id="cb45-64"><a href="example-unique-analyses.html#cb45-64" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(result,<span class="fu">aes</span>(name,<span class="sc">-</span><span class="fu">log10</span>(pvalue)))<span class="sc">+</span></span>
<span id="cb45-65"><a href="example-unique-analyses.html#cb45-65" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>,<span class="fu">aes</span>(<span class="at">fill=</span>pathway_type))<span class="sc">+</span></span>
<span id="cb45-66"><a href="example-unique-analyses.html#cb45-66" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;Set3&quot;</span>),<span class="at">name=</span><span class="st">&quot;Pathway Category&quot;</span>,</span>
<span id="cb45-67"><a href="example-unique-analyses.html#cb45-67" tabindex="-1"></a>                    <span class="at">breaks=</span><span class="fu">unique</span>(kegg_pathway_uniq<span class="sc">$</span>pathway_type))<span class="sc">+</span></span>
<span id="cb45-68"><a href="example-unique-analyses.html#cb45-68" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>))<span class="sc">+</span></span>
<span id="cb45-69"><a href="example-unique-analyses.html#cb45-69" tabindex="-1"></a>  <span class="fu">coord_flip</span>()<span class="sc">+</span></span>
<span id="cb45-70"><a href="example-unique-analyses.html#cb45-70" tabindex="-1"></a>  <span class="fu">facet_grid</span>(. <span class="sc">~</span> type,<span class="at">scales=</span><span class="st">&quot;free_x&quot;</span>)<span class="sc">+</span></span>
<span id="cb45-71"><a href="example-unique-analyses.html#cb45-71" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb45-72"><a href="example-unique-analyses.html#cb45-72" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="cn">NULL</span>)</span>
<span id="cb45-73"><a href="example-unique-analyses.html#cb45-73" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/11.pdnet_epe.pdf&quot;</span>,p1,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb45-74"><a href="example-unique-analyses.html#cb45-74" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/unique/11.pdnet_epe.png&quot;</span>,p1,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">5</span>)</span></code></pre></div>
<p><img src="result/unique/11.pdent.png" width="100%" style="display: block; margin: auto;" /></p>
<p><img src="result/unique/11.pdnet_epe.png" width="70%" style="display: block; margin: auto;" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="database-construction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="example-rountine-analyses.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/02-example_unique.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
