<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Figure1 | MNet</title>
  <meta name="description" content="Chapter 6 Figure1 | MNet" />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Figure1 | MNet" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Figure1 | MNet" />
  
  
  

<meta name="author" content="Tuantuan Gui, Xuan Liu and Hai Fang" />


<meta name="date" content="2024-02-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="shiny.html"/>
<link rel="next" href="figure4.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.30/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-buttons-1.13.4/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-buttons-1.13.4/js/dataTables.buttons.min.js"></script>
<script src="libs/dt-ext-buttons-1.13.4/js/buttons.html5.min.js"></script>
<script src="libs/dt-ext-buttons-1.13.4/js/buttons.colVis.min.js"></script>
<script src="libs/dt-ext-buttons-1.13.4/js/buttons.print.min.js"></script>
<link href="libs/dt-ext-fixedcolumns-1.13.4/css/fixedColumns.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedcolumns-1.13.4/js/dataTables.fixedColumns.min.js"></script>
<link href="libs/dt-plugin-searchhighlight-1.13.4/source.css" rel="stylesheet" />
<script src="libs/dt-plugin-searchhighlight-1.13.4/jquery.highlight.js"></script>
<script src="libs/dt-plugin-searchhighlight-1.13.4/source.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction &amp; Installation</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#overview-of-workflow"><i class="fa fa-check"></i><b>1.1</b> Overview of Workflow</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.2</b> Installation</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="database-construction.html"><a href="database-construction.html"><i class="fa fa-check"></i><b>2</b> Database construction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="database-construction.html"><a href="database-construction.html#kegg-pathways-metabolite-and-gene"><i class="fa fa-check"></i><b>2.1</b> KEGG pathway’s metabolite and gene</a></li>
<li class="chapter" data-level="2.2" data-path="database-construction.html"><a href="database-construction.html#kegg-metabolite-metabolite-pairs-and-metabolite-gene-pairs"><i class="fa fa-check"></i><b>2.2</b> KEGG metabolite-metabolite pairs and metabolite-gene pairs</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="database-construction.html"><a href="database-construction.html#keggs-reaction-and-compounds-download"><i class="fa fa-check"></i><b>2.2.1</b> KEGG’s reaction and compound’s download</a></li>
<li class="chapter" data-level="2.2.2" data-path="database-construction.html"><a href="database-construction.html#downloading-the-enzyme-and-its-corresponding-gene-information-from-the-htextand-extract-the-gene-and-the-enzymes-relationship"><i class="fa fa-check"></i><b>2.2.2</b> Downloading the enzyme and its corresponding gene information from the htext,and extract the gene and the enzyme’s relationship</a></li>
<li class="chapter" data-level="2.2.3" data-path="database-construction.html"><a href="database-construction.html#extracting-the-metabolite-gene-pairs-and-metabolite-metabolite-pairs"><i class="fa fa-check"></i><b>2.2.3</b> Extracting the metabolite-gene pairs and metabolite-metabolite pairs</a></li>
<li class="chapter" data-level="2.2.4" data-path="database-construction.html"><a href="database-construction.html#conclusion"><i class="fa fa-check"></i><b>2.2.4</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="database-construction.html"><a href="database-construction.html#graphite"><i class="fa fa-check"></i><b>2.3</b> Graphite</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="database-construction.html"><a href="database-construction.html#download-all-the-data-from-graphite"><i class="fa fa-check"></i><b>2.3.1</b> Download all the data from graphite</a></li>
<li class="chapter" data-level="2.3.2" data-path="database-construction.html"><a href="database-construction.html#download-the-information-of-metabolism-pathways"><i class="fa fa-check"></i><b>2.3.2</b> Download the information of metabolism pathways</a></li>
<li class="chapter" data-level="2.3.3" data-path="database-construction.html"><a href="database-construction.html#extract-the-data-of-metabolism"><i class="fa fa-check"></i><b>2.3.3</b> Extract the data of metabolism</a></li>
<li class="chapter" data-level="2.3.4" data-path="database-construction.html"><a href="database-construction.html#combine-the-data-from-database-kegg-wikipathway-reactome-smpdb-and-then-uniq-the-data"><i class="fa fa-check"></i><b>2.3.4</b> Combine the data from database kegg, wikipathway, reactome, smpdb and then uniq the data</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="database-construction.html"><a href="database-construction.html#bigg"><i class="fa fa-check"></i><b>2.4</b> BiGG</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="database-construction.html"><a href="database-construction.html#down-all-the-models-in-the-bigg"><i class="fa fa-check"></i><b>2.4.1</b> Down all the models in the BiGG</a></li>
<li class="chapter" data-level="2.4.2" data-path="database-construction.html"><a href="database-construction.html#download-the-every-reaction"><i class="fa fa-check"></i><b>2.4.2</b> Download the every reaction</a></li>
<li class="chapter" data-level="2.4.3" data-path="database-construction.html"><a href="database-construction.html#exact-all-the-metabolites-names-in-every-model-in-bigg"><i class="fa fa-check"></i><b>2.4.3</b> Exact all the metabolites names in every model in BiGG</a></li>
<li class="chapter" data-level="2.4.4" data-path="database-construction.html"><a href="database-construction.html#the-r-script-that-change-the-metabolite-name-to-kegg-id-in-the-final-output"><i class="fa fa-check"></i><b>2.4.4</b> The R script that change the metabolite name to kegg id in the final output</a></li>
<li class="chapter" data-level="2.4.5" data-path="database-construction.html"><a href="database-construction.html#combine-the-data-from-bigg-and-graphite"><i class="fa fa-check"></i><b>2.4.5</b> Combine the data from BiGG and graphite</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html"><i class="fa fa-check"></i><b>3</b> Example-unique analyses</a>
<ul>
<li class="chapter" data-level="3.1" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#loading-the-data"><i class="fa fa-check"></i><b>3.1</b> Loading the data</a></li>
<li class="chapter" data-level="3.2" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#pathway-enrichment-analysis"><i class="fa fa-check"></i><b>3.2</b> Pathway enrichment analysis</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#gene-pathway-enrichment"><i class="fa fa-check"></i><b>3.2.1</b> gene pathway enrichment</a></li>
<li class="chapter" data-level="3.2.2" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#metabolite-pathway-enrichment"><i class="fa fa-check"></i><b>3.2.2</b> metabolite pathway enrichment</a></li>
<li class="chapter" data-level="3.2.3" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#extended-pathway-enrichment"><i class="fa fa-check"></i><b>3.2.3</b> Extended pathway enrichment</a></li>
<li class="chapter" data-level="3.2.4" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#overview-of-pathway-enrichment-analysis"><i class="fa fa-check"></i><b>3.2.4</b> Overview of pathway enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#pathway-differential-abundance-score"><i class="fa fa-check"></i><b>3.3</b> Pathway differential abundance score</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#gene-differential-abundance-score"><i class="fa fa-check"></i><b>3.3.1</b> gene differential abundance score</a></li>
<li class="chapter" data-level="3.3.2" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#metabolite-differential-abundance-score"><i class="fa fa-check"></i><b>3.3.2</b> metabolite differential abundance score</a></li>
<li class="chapter" data-level="3.3.3" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#extended-pathway-differential-abundance-score"><i class="fa fa-check"></i><b>3.3.3</b> Extended pathway differential abundance score</a></li>
<li class="chapter" data-level="3.3.4" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#overview-of-pathway-differential-abundance-score"><i class="fa fa-check"></i><b>3.3.4</b> Overview of pathway differential abundance score</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#esea-metabolite-set-enrichment-analysis"><i class="fa fa-check"></i><b>3.4</b> ESEA Metabolite Set Enrichment Analysis</a></li>
<li class="chapter" data-level="3.5" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#subnetwork"><i class="fa fa-check"></i><b>3.5</b> Subnetwork</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html"><i class="fa fa-check"></i><b>4</b> Example-rountine analyses</a>
<ul>
<li class="chapter" data-level="4.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-and-kegg-name-change"><i class="fa fa-check"></i><b>4.1</b> Metabolite and kegg name change</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-name-changed-to-refmet-name"><i class="fa fa-check"></i><b>4.1.1</b> Metabolite name changed to refmet name</a></li>
<li class="chapter" data-level="4.1.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-name-changed-to-kegg-id"><i class="fa fa-check"></i><b>4.1.2</b> Metabolite name changed to kegg id</a></li>
<li class="chapter" data-level="4.1.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-name-corresponding-to-kegg-pathway"><i class="fa fa-check"></i><b>4.1.3</b> Metabolite name corresponding to kegg pathway</a></li>
<li class="chapter" data-level="4.1.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-keggid-to-kegg-pathway"><i class="fa fa-check"></i><b>4.1.4</b> Metabolite keggid to kegg pathway</a></li>
<li class="chapter" data-level="4.1.5" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#the-pathway-information"><i class="fa fa-check"></i><b>4.1.5</b> The pathway information</a></li>
<li class="chapter" data-level="4.1.6" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathway-name-to-pathway-id"><i class="fa fa-check"></i><b>4.1.6</b> Pathway name to pathway id</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#group-wise-analyses"><i class="fa fa-check"></i><b>4.2</b> Group-wise analyses</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pca-plot"><i class="fa fa-check"></i><b>4.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="4.2.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differnetial-metabolite-analysis"><i class="fa fa-check"></i><b>4.2.2</b> Differnetial metabolite analysis</a></li>
<li class="chapter" data-level="4.2.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differential-metabolites-volcano"><i class="fa fa-check"></i><b>4.2.3</b> Differential metabolites’ volcano</a></li>
<li class="chapter" data-level="4.2.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differential-metabolites-heatmap"><i class="fa fa-check"></i><b>4.2.4</b> Differential metabolites’ heatmap</a></li>
<li class="chapter" data-level="4.2.5" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differential-metabolites-zscore"><i class="fa fa-check"></i><b>4.2.5</b> Differential metabolites’ zscore</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathway-analysis"><i class="fa fa-check"></i><b>4.3</b> Pathway analysis</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathview-only-metabolite"><i class="fa fa-check"></i><b>4.3.1</b> Pathview only metabolite</a></li>
<li class="chapter" data-level="4.3.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathview-metabolite-and-gene"><i class="fa fa-check"></i><b>4.3.2</b> Pathview metabolite and gene</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#clinical-analysis"><i class="fa fa-check"></i><b>4.4</b> Clinical analysis</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#time-series-of-clinical"><i class="fa fa-check"></i><b>4.4.1</b> Time series of clinical</a></li>
<li class="chapter" data-level="4.4.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#survival-analysis"><i class="fa fa-check"></i><b>4.4.2</b> Survival analysis</a></li>
<li class="chapter" data-level="4.4.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolites-survival-plot"><i class="fa fa-check"></i><b>4.4.3</b> Metabolites’ survival plot</a></li>
<li class="chapter" data-level="4.4.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#cox-analysis"><i class="fa fa-check"></i><b>4.4.4</b> Cox analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="shiny.html"><a href="shiny.html"><i class="fa fa-check"></i><b>5</b> Shiny</a>
<ul>
<li class="chapter" data-level="5.1" data-path="shiny.html"><a href="shiny.html#tencent-log"><i class="fa fa-check"></i><b>5.1</b> Tencent log</a></li>
<li class="chapter" data-level="5.2" data-path="shiny.html"><a href="shiny.html#shiny-app-log"><i class="fa fa-check"></i><b>5.2</b> Shiny app log</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="figure1.html"><a href="figure1.html"><i class="fa fa-check"></i><b>6</b> Figure1</a>
<ul>
<li class="chapter" data-level="6.1" data-path="figure1.html"><a href="figure1.html#a"><i class="fa fa-check"></i><b>6.1</b> 1.1 (A)</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="figure1.html"><a href="figure1.html#batch-removal"><i class="fa fa-check"></i><b>6.1.1</b> Batch removal</a></li>
<li class="chapter" data-level="6.1.2" data-path="figure1.html"><a href="figure1.html#metabolite-information"><i class="fa fa-check"></i><b>6.1.2</b> Metabolite information</a></li>
<li class="chapter" data-level="6.1.3" data-path="figure1.html"><a href="figure1.html#metabolic-pathway-information"><i class="fa fa-check"></i><b>6.1.3</b> Metabolic pathway information</a></li>
<li class="chapter" data-level="6.1.4" data-path="figure1.html"><a href="figure1.html#metabolite-and-metabolic-pathway-annotation"><i class="fa fa-check"></i><b>6.1.4</b> Metabolite and metabolic pathway annotation</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="figure1.html"><a href="figure1.html#b"><i class="fa fa-check"></i><b>6.2</b> 1.2 (B)</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="figure1.html"><a href="figure1.html#clinical-and-metabolite-analysis"><i class="fa fa-check"></i><b>6.2.1</b> Clinical and metabolite analysis</a></li>
<li class="chapter" data-level="6.2.2" data-path="figure1.html"><a href="figure1.html#time-series-analysis"><i class="fa fa-check"></i><b>6.2.2</b> Time series analysis</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="figure1.html"><a href="figure1.html#c"><i class="fa fa-check"></i><b>6.3</b> 1.3 (C)</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="figure1.html"><a href="figure1.html#dimension-reduction-visualization"><i class="fa fa-check"></i><b>6.3.1</b> Dimension reduction &amp; Visualization</a></li>
<li class="chapter" data-level="6.3.2" data-path="figure1.html"><a href="figure1.html#differential-metabolites-visualization"><i class="fa fa-check"></i><b>6.3.2</b> Differential metabolites visualization</a></li>
<li class="chapter" data-level="6.3.3" data-path="figure1.html"><a href="figure1.html#pathway-enrichment-analyses"><i class="fa fa-check"></i><b>6.3.3</b> Pathway enrichment analyses</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="figure1.html"><a href="figure1.html#d"><i class="fa fa-check"></i><b>6.4</b> 1.4 (D)</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="figure1.html"><a href="figure1.html#extended-pathway-enrichment-analysis-epea"><i class="fa fa-check"></i><b>6.4.1</b> Extended pathway enrichment analysis (EPEA)</a></li>
<li class="chapter" data-level="6.4.2" data-path="figure1.html"><a href="figure1.html#extended-differential-abundance-eda-score"><i class="fa fa-check"></i><b>6.4.2</b> Extended differential abundance (EDA) score</a></li>
<li class="chapter" data-level="6.4.3" data-path="figure1.html"><a href="figure1.html#extended-data-set-enrichment-analysis-esea"><i class="fa fa-check"></i><b>6.4.3</b> Extended data set enrichment analysis (ESEA)</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="figure1.html"><a href="figure1.html#e"><i class="fa fa-check"></i><b>6.5</b> 1.5 (E)</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="figure1.html"><a href="figure1.html#dbmnet-database"><i class="fa fa-check"></i><b>6.5.1</b> dbMNet database</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="figure4.html"><a href="figure4.html"><i class="fa fa-check"></i><b>7</b> Figure4</a>
<ul>
<li class="chapter" data-level="7.1" data-path="figure4.html"><a href="figure4.html#load-data"><i class="fa fa-check"></i><b>7.1</b> Load data</a></li>
<li class="chapter" data-level="7.2" data-path="figure4.html"><a href="figure4.html#mlimma-analysis"><i class="fa fa-check"></i><b>7.2</b> mlimma analysis</a></li>
<li class="chapter" data-level="7.3" data-path="figure4.html"><a href="figure4.html#epe-analysis"><i class="fa fa-check"></i><b>7.3</b> EPE analysis</a></li>
<li class="chapter" data-level="7.4" data-path="figure4.html"><a href="figure4.html#eda-analysis"><i class="fa fa-check"></i><b>7.4</b> EDA analysis</a></li>
<li class="chapter" data-level="7.5" data-path="figure4.html"><a href="figure4.html#esea-analysis"><i class="fa fa-check"></i><b>7.5</b> ESEA analysis</a></li>
<li class="chapter" data-level="7.6" data-path="figure4.html"><a href="figure4.html#aml-epe-analysis"><i class="fa fa-check"></i><b>7.6</b> AML EPE analysis</a></li>
<li class="chapter" data-level="7.7" data-path="figure4.html"><a href="figure4.html#volcano-plot-1"><i class="fa fa-check"></i><b>7.7</b> Volcano plot</a>
<ul>
<li class="chapter" data-level="7.7.1" data-path="figure4.html"><a href="figure4.html#differential-metabolites-and-genes"><i class="fa fa-check"></i><b>7.7.1</b> Differential metabolites and genes</a></li>
<li class="chapter" data-level="7.7.2" data-path="figure4.html"><a href="figure4.html#all"><i class="fa fa-check"></i><b>7.7.2</b> ALL</a></li>
<li class="chapter" data-level="7.7.3" data-path="figure4.html"><a href="figure4.html#aml"><i class="fa fa-check"></i><b>7.7.3</b> AML</a></li>
</ul></li>
<li class="chapter" data-level="7.8" data-path="figure4.html"><a href="figure4.html#heatmap-1"><i class="fa fa-check"></i><b>7.8</b> Heatmap</a></li>
<li class="chapter" data-level="7.9" data-path="figure4.html"><a href="figure4.html#subnetwork-1"><i class="fa fa-check"></i><b>7.9</b> Subnetwork</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>8</b> Session information</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">MNet</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="figure1" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Figure1<a href="figure1.html#figure1" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="a" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> 1.1 (A)<a href="figure1.html#a" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="batch-removal" class="section level3 hasAnchor" number="6.1.1">
<h3><span class="header-section-number">6.1.1</span> Batch removal<a href="figure1.html#batch-removal" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="figure1.html#cb76-1" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;result/figure1&quot;</span>)</span>
<span id="cb76-2"><a href="figure1.html#cb76-2" tabindex="-1"></a></span>
<span id="cb76-3"><a href="figure1.html#cb76-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb76-4"><a href="figure1.html#cb76-4" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb76-5"><a href="figure1.html#cb76-5" tabindex="-1"></a><span class="fu">library</span>(MNet)</span>
<span id="cb76-6"><a href="figure1.html#cb76-6" tabindex="-1"></a></span>
<span id="cb76-7"><a href="figure1.html#cb76-7" tabindex="-1"></a><span class="co"># SERRF校正前</span></span>
<span id="cb76-8"><a href="figure1.html#cb76-8" tabindex="-1"></a>before_normalization_raw <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;raw_data/test_for_serrf.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb76-9"><a href="figure1.html#cb76-9" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb76-10"><a href="figure1.html#cb76-10" tabindex="-1"></a></span>
<span id="cb76-11"><a href="figure1.html#cb76-11" tabindex="-1"></a>before_normalization <span class="ot">&lt;-</span> before_normalization_raw <span class="sc">%&gt;%</span></span>
<span id="cb76-12"><a href="figure1.html#cb76-12" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb76-13"><a href="figure1.html#cb76-13" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb76-14"><a href="figure1.html#cb76-14" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>V4 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;No&quot;</span>,<span class="st">&quot;label&quot;</span>))</span>
<span id="cb76-15"><a href="figure1.html#cb76-15" tabindex="-1"></a></span>
<span id="cb76-16"><a href="figure1.html#cb76-16" tabindex="-1"></a><span class="fu">names</span>(before_normalization) <span class="ot">&lt;-</span> <span class="fu">as.character</span>(before_normalization_raw[,<span class="dv">2</span>])</span>
<span id="cb76-17"><a href="figure1.html#cb76-17" tabindex="-1"></a><span class="fu">names</span>(before_normalization)[<span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="st">&quot;sample_id&quot;</span></span>
<span id="cb76-18"><a href="figure1.html#cb76-18" tabindex="-1"></a></span>
<span id="cb76-19"><a href="figure1.html#cb76-19" tabindex="-1"></a><span class="co"># pca</span></span>
<span id="cb76-20"><a href="figure1.html#cb76-20" tabindex="-1"></a>before_batch <span class="ot">&lt;-</span> <span class="fu">as.character</span>(before_normalization_raw[<span class="dv">1</span>,<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)])</span>
<span id="cb76-21"><a href="figure1.html#cb76-21" tabindex="-1"></a>before_type <span class="ot">&lt;-</span> <span class="fu">as.character</span>(before_normalization_raw[<span class="dv">2</span>,<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)])</span>
<span id="cb76-22"><a href="figure1.html#cb76-22" tabindex="-1"></a>before_sample_id <span class="ot">&lt;-</span> <span class="fu">as.character</span>(before_normalization_raw[<span class="dv">4</span>,<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)])</span>
<span id="cb76-23"><a href="figure1.html#cb76-23" tabindex="-1"></a></span>
<span id="cb76-24"><a href="figure1.html#cb76-24" tabindex="-1"></a>dd <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">type=</span>before_type,<span class="at">batch=</span>before_batch,<span class="at">sample_id=</span>before_sample_id)</span>
<span id="cb76-25"><a href="figure1.html#cb76-25" tabindex="-1"></a></span>
<span id="cb76-26"><a href="figure1.html#cb76-26" tabindex="-1"></a>mydata_pca <span class="ot">&lt;-</span> <span class="fu">apply</span>(before_normalization[,<span class="dv">5</span><span class="sc">:</span><span class="fu">ncol</span>(before_normalization)],<span class="dv">2</span>,as.numeric)</span>
<span id="cb76-27"><a href="figure1.html#cb76-27" tabindex="-1"></a></span>
<span id="cb76-28"><a href="figure1.html#cb76-28" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">log2</span>(mydata_pca<span class="sc">+</span><span class="dv">1</span>), <span class="at">center =</span> T, <span class="at">scale. =</span> T)</span>
<span id="cb76-29"><a href="figure1.html#cb76-29" tabindex="-1"></a>variance <span class="ot">=</span> pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span><span class="fu">sum</span>(pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb76-30"><a href="figure1.html#cb76-30" tabindex="-1"></a></span>
<span id="cb76-31"><a href="figure1.html#cb76-31" tabindex="-1"></a>pca.data <span class="ot">=</span> <span class="fu">data.frame</span>(pca<span class="sc">$</span>x,<span class="at">type=</span>before_type,<span class="at">batch=</span>before_batch,<span class="at">sample_id=</span>before_sample_id)</span>
<span id="cb76-32"><a href="figure1.html#cb76-32" tabindex="-1"></a></span>
<span id="cb76-33"><a href="figure1.html#cb76-33" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca.data,<span class="fu">aes</span>(PC1,PC2,<span class="at">color=</span>type,<span class="at">shape=</span>before_batch))<span class="sc">+</span></span>
<span id="cb76-34"><a href="figure1.html#cb76-34" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb76-35"><a href="figure1.html#cb76-35" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;#3e68a0&quot;</span>))<span class="sc">+</span></span>
<span id="cb76-36"><a href="figure1.html#cb76-36" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="fu">length</span>(<span class="fu">unique</span>(before_batch))<span class="sc">-</span><span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb76-37"><a href="figure1.html#cb76-37" tabindex="-1"></a>  <span class="co">#geom_text(aes(label=sample_id))+</span></span>
<span id="cb76-38"><a href="figure1.html#cb76-38" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb76-39"><a href="figure1.html#cb76-39" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),<span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb76-40"><a href="figure1.html#cb76-40" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="fu">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>,<span class="fu">signif</span>(variance[<span class="dv">1</span>]<span class="sc">*</span><span class="dv">100</span>,<span class="dv">3</span>),<span class="st">&quot;%)&quot;</span>),</span>
<span id="cb76-41"><a href="figure1.html#cb76-41" tabindex="-1"></a>       <span class="at">y=</span><span class="fu">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>,<span class="fu">signif</span>(variance[<span class="dv">2</span>]<span class="sc">*</span><span class="dv">100</span>,<span class="dv">3</span>),<span class="st">&quot;%)&quot;</span>))</span>
<span id="cb76-42"><a href="figure1.html#cb76-42" tabindex="-1"></a></span>
<span id="cb76-43"><a href="figure1.html#cb76-43" tabindex="-1"></a><span class="co"># SERRF校正后</span></span>
<span id="cb76-44"><a href="figure1.html#cb76-44" tabindex="-1"></a>after_normalization_raw <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;raw_data/normalized by - SERRF.csv&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb76-45"><a href="figure1.html#cb76-45" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb76-46"><a href="figure1.html#cb76-46" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>(label,<span class="at">.keep_all =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb76-47"><a href="figure1.html#cb76-47" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;label&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb76-48"><a href="figure1.html#cb76-48" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb76-49"><a href="figure1.html#cb76-49" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb76-50"><a href="figure1.html#cb76-50" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;sample&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb76-51"><a href="figure1.html#cb76-51" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">sample=</span><span class="fu">gsub</span>(<span class="st">&quot;X&quot;</span>,<span class="st">&quot;&quot;</span>,sample)) <span class="sc">%&gt;%</span></span>
<span id="cb76-52"><a href="figure1.html#cb76-52" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;sample&quot;</span>)</span>
<span id="cb76-53"><a href="figure1.html#cb76-53" tabindex="-1"></a></span>
<span id="cb76-54"><a href="figure1.html#cb76-54" tabindex="-1"></a></span>
<span id="cb76-55"><a href="figure1.html#cb76-55" tabindex="-1"></a>after_sample_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(after_normalization_raw)</span>
<span id="cb76-56"><a href="figure1.html#cb76-56" tabindex="-1"></a>after_dd <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample_id=</span>after_sample_id) <span class="sc">%&gt;%</span></span>
<span id="cb76-57"><a href="figure1.html#cb76-57" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(dd,<span class="at">by=</span><span class="st">&quot;sample_id&quot;</span>)</span>
<span id="cb76-58"><a href="figure1.html#cb76-58" tabindex="-1"></a></span>
<span id="cb76-59"><a href="figure1.html#cb76-59" tabindex="-1"></a>mydata_pca <span class="ot">&lt;-</span> <span class="fu">apply</span>(after_normalization_raw,<span class="dv">2</span>,as.numeric)</span>
<span id="cb76-60"><a href="figure1.html#cb76-60" tabindex="-1"></a></span>
<span id="cb76-61"><a href="figure1.html#cb76-61" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">log2</span>(mydata_pca<span class="sc">+</span><span class="dv">1</span>), <span class="at">center =</span> T, <span class="at">scale. =</span> T)</span>
<span id="cb76-62"><a href="figure1.html#cb76-62" tabindex="-1"></a>variance <span class="ot">=</span> pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span><span class="fu">sum</span>(pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb76-63"><a href="figure1.html#cb76-63" tabindex="-1"></a></span>
<span id="cb76-64"><a href="figure1.html#cb76-64" tabindex="-1"></a>pca.data <span class="ot">=</span> <span class="fu">cbind</span>(pca<span class="sc">$</span>x,after_dd)</span>
<span id="cb76-65"><a href="figure1.html#cb76-65" tabindex="-1"></a></span>
<span id="cb76-66"><a href="figure1.html#cb76-66" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca.data,<span class="fu">aes</span>(PC1,PC2,<span class="at">color=</span>type,<span class="at">shape=</span>batch))<span class="sc">+</span></span>
<span id="cb76-67"><a href="figure1.html#cb76-67" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb76-68"><a href="figure1.html#cb76-68" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;#3e68a0&quot;</span>))<span class="sc">+</span></span>
<span id="cb76-69"><a href="figure1.html#cb76-69" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values=</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb76-70"><a href="figure1.html#cb76-70" tabindex="-1"></a>  <span class="co">#geom_text(aes(label=sample_id))+</span></span>
<span id="cb76-71"><a href="figure1.html#cb76-71" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb76-72"><a href="figure1.html#cb76-72" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),<span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb76-73"><a href="figure1.html#cb76-73" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="fu">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>,<span class="fu">signif</span>(variance[<span class="dv">1</span>]<span class="sc">*</span><span class="dv">100</span>,<span class="dv">3</span>),<span class="st">&quot;%)&quot;</span>),</span>
<span id="cb76-74"><a href="figure1.html#cb76-74" tabindex="-1"></a>       <span class="at">y=</span><span class="fu">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>,<span class="fu">signif</span>(variance[<span class="dv">2</span>]<span class="sc">*</span><span class="dv">100</span>,<span class="dv">3</span>),<span class="st">&quot;%)&quot;</span>))</span>
<span id="cb76-75"><a href="figure1.html#cb76-75" tabindex="-1"></a></span>
<span id="cb76-76"><a href="figure1.html#cb76-76" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/01.pca_raw.pdf&quot;</span>,p1,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="fl">3.9</span>)</span>
<span id="cb76-77"><a href="figure1.html#cb76-77" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/01.pca_normalization.pdf&quot;</span>,p2,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="fl">3.9</span>)</span>
<span id="cb76-78"><a href="figure1.html#cb76-78" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/01.pca_raw.png&quot;</span>,p1,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="fl">3.9</span>)</span>
<span id="cb76-79"><a href="figure1.html#cb76-79" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/01.pca_normalization.png&quot;</span>,p2,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="fl">3.9</span>)</span></code></pre></div>
<p><img src="result/figure1/01.pca_raw.png" width="50%" /><img src="result/figure1/01.pca_normalization.png" width="50%" /></p>
</div>
<div id="metabolite-information" class="section level3 hasAnchor" number="6.1.2">
<h3><span class="header-section-number">6.1.2</span> Metabolite information<a href="figure1.html#metabolite-information" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="figure1.html#cb77-1" tabindex="-1"></a>compound_name <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;2-hydroxybutyric acid&quot;</span>,<span class="st">&quot;1-methyladenosine&quot;</span>,<span class="st">&quot;2-Aminooctanoic&quot;</span>)</span>
<span id="cb77-2"><a href="figure1.html#cb77-2" tabindex="-1"></a>refmetid_result <span class="ot">&lt;-</span> <span class="fu">name2refmet</span>(compound_name)</span>
<span id="cb77-3"><a href="figure1.html#cb77-3" tabindex="-1"></a><span class="fu">write.table</span>(refmetid_result,<span class="st">&quot;result/figure1/02.refmetid_result.txt&quot;</span>,<span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">quote =</span> F,<span class="at">row.names =</span> F)</span></code></pre></div>
</div>
<div id="metabolic-pathway-information" class="section level3 hasAnchor" number="6.1.3">
<h3><span class="header-section-number">6.1.3</span> Metabolic pathway information<a href="figure1.html#metabolic-pathway-information" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="figure1.html#cb78-1" tabindex="-1"></a>refmet_kegg <span class="ot">&lt;-</span> <span class="fu">name2keggid</span>(refmetid_result<span class="sc">$</span>Refmet_name)</span>
<span id="cb78-2"><a href="figure1.html#cb78-2" tabindex="-1"></a></span>
<span id="cb78-3"><a href="figure1.html#cb78-3" tabindex="-1"></a>result_pathway <span class="ot">&lt;-</span> <span class="fu">keggid2pathway</span>(refmet_kegg<span class="sc">$</span>KEGG_id) </span>
<span id="cb78-4"><a href="figure1.html#cb78-4" tabindex="-1"></a></span>
<span id="cb78-5"><a href="figure1.html#cb78-5" tabindex="-1"></a>result <span class="ot">&lt;-</span> refmetid_result <span class="sc">%&gt;%</span></span>
<span id="cb78-6"><a href="figure1.html#cb78-6" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(refmet_kegg,<span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;Refmet_name&quot;</span><span class="ot">=</span><span class="st">&quot;Name&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb78-7"><a href="figure1.html#cb78-7" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(result_pathway,<span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;KEGG_id&quot;</span><span class="ot">=</span><span class="st">&quot;ENTRY&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb78-8"><a href="figure1.html#cb78-8" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Input_name,Refmet_name,KEGG_id,PATHWAY,<span class="fu">everything</span>())</span>
<span id="cb78-9"><a href="figure1.html#cb78-9" tabindex="-1"></a><span class="fu">write.table</span>(result,<span class="st">&quot;result/figure1/03.result_all.txt&quot;</span>,<span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">quote =</span> F,<span class="at">row.names =</span> F)</span></code></pre></div>
</div>
<div id="metabolite-and-metabolic-pathway-annotation" class="section level3 hasAnchor" number="6.1.4">
<h3><span class="header-section-number">6.1.4</span> Metabolite and metabolic pathway annotation<a href="figure1.html#metabolite-and-metabolic-pathway-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="figure1.html#cb79-1" tabindex="-1"></a>meta_name <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">name=</span><span class="fu">substr</span>(<span class="fu">rownames</span>(meta_dat),<span class="at">start=</span><span class="dv">1</span>,<span class="at">stop=</span><span class="dv">6</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb79-2"><a href="figure1.html#cb79-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(all_kegg_id,<span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;name&quot;</span><span class="ot">=</span><span class="st">&quot;ENTRY&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb79-3"><a href="figure1.html#cb79-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>(name,<span class="at">.keep_all =</span> T)</span>
<span id="cb79-4"><a href="figure1.html#cb79-4" tabindex="-1"></a></span>
<span id="cb79-5"><a href="figure1.html#cb79-5" tabindex="-1"></a>meta_subtype <span class="ot">&lt;-</span> <span class="fu">name2refmet</span>(meta_name<span class="sc">$</span>NAME)</span>
<span id="cb79-6"><a href="figure1.html#cb79-6" tabindex="-1"></a></span>
<span id="cb79-7"><a href="figure1.html#cb79-7" tabindex="-1"></a><span class="fu">write.table</span>(meta_name,<span class="st">&quot;result/figure1/04.annotation_subclass.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F)</span>
<span id="cb79-8"><a href="figure1.html#cb79-8" tabindex="-1"></a></span>
<span id="cb79-9"><a href="figure1.html#cb79-9" tabindex="-1"></a>dd <span class="ot">&lt;-</span> meta_subtype <span class="sc">%&gt;%</span></span>
<span id="cb79-10"><a href="figure1.html#cb79-10" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(Super_class) <span class="sc">%&gt;%</span></span>
<span id="cb79-11"><a href="figure1.html#cb79-11" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">n=</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb79-12"><a href="figure1.html#cb79-12" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Super_class)) <span class="sc">%&gt;%</span></span>
<span id="cb79-13"><a href="figure1.html#cb79-13" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(n))</span>
<span id="cb79-14"><a href="figure1.html#cb79-14" tabindex="-1"></a>dd<span class="sc">$</span>Super_class <span class="ot">&lt;-</span> <span class="fu">factor</span>(dd<span class="sc">$</span>Super_class,<span class="at">levels =</span> dd<span class="sc">$</span>Super_class)</span>
<span id="cb79-15"><a href="figure1.html#cb79-15" tabindex="-1"></a></span>
<span id="cb79-16"><a href="figure1.html#cb79-16" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(dd,<span class="fu">aes</span>(Super_class,n,<span class="at">fill=</span>Super_class))<span class="sc">+</span></span>
<span id="cb79-17"><a href="figure1.html#cb79-17" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="sc">+</span></span>
<span id="cb79-18"><a href="figure1.html#cb79-18" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb79-19"><a href="figure1.html#cb79-19" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="fl">0.01</span>, <span class="fl">0.01</span>))<span class="sc">+</span></span>
<span id="cb79-20"><a href="figure1.html#cb79-20" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb79-21"><a href="figure1.html#cb79-21" tabindex="-1"></a></span>
<span id="cb79-22"><a href="figure1.html#cb79-22" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/04.annotation.pdf&quot;</span>,<span class="at">width=</span><span class="dv">8</span>,<span class="at">height =</span> <span class="dv">6</span>)</span>
<span id="cb79-23"><a href="figure1.html#cb79-23" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/04.annotation.png&quot;</span>,<span class="at">width=</span><span class="dv">8</span>,<span class="at">height =</span> <span class="dv">6</span>)</span>
<span id="cb79-24"><a href="figure1.html#cb79-24" tabindex="-1"></a></span>
<span id="cb79-25"><a href="figure1.html#cb79-25" tabindex="-1"></a>d <span class="ot">&lt;-</span> PathwayExtendData <span class="sc">%&gt;%</span></span>
<span id="cb79-26"><a href="figure1.html#cb79-26" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(kegg_pathwayname,kegg_category) <span class="sc">%&gt;%</span></span>
<span id="cb79-27"><a href="figure1.html#cb79-27" tabindex="-1"></a>  <span class="fu">unique</span>() <span class="sc">%&gt;%</span></span>
<span id="cb79-28"><a href="figure1.html#cb79-28" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(kegg_category) <span class="sc">%&gt;%</span></span>
<span id="cb79-29"><a href="figure1.html#cb79-29" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">n=</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb79-30"><a href="figure1.html#cb79-30" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(n)</span>
<span id="cb79-31"><a href="figure1.html#cb79-31" tabindex="-1"></a></span>
<span id="cb79-32"><a href="figure1.html#cb79-32" tabindex="-1"></a>d<span class="sc">$</span>kegg_category <span class="ot">&lt;-</span> <span class="fu">factor</span>(d<span class="sc">$</span>kegg_category,<span class="at">levels=</span>d<span class="sc">$</span>kegg_category)</span>
<span id="cb79-33"><a href="figure1.html#cb79-33" tabindex="-1"></a></span>
<span id="cb79-34"><a href="figure1.html#cb79-34" tabindex="-1"></a>cols <span class="ot">&lt;-</span> <span class="fu">c</span>(RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">9</span>,<span class="st">&#39;YlOrRd&#39;</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">9</span>,<span class="st">&#39;Reds&#39;</span>))</span>
<span id="cb79-35"><a href="figure1.html#cb79-35" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(d,<span class="fu">aes</span>(kegg_category,n,<span class="at">fill=</span>kegg_category))<span class="sc">+</span></span>
<span id="cb79-36"><a href="figure1.html#cb79-36" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="sc">+</span></span>
<span id="cb79-37"><a href="figure1.html#cb79-37" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb79-38"><a href="figure1.html#cb79-38" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="fl">0.01</span>, <span class="fl">0.01</span>))<span class="sc">+</span></span>
<span id="cb79-39"><a href="figure1.html#cb79-39" tabindex="-1"></a>  <span class="co">#theme(axis.text.x = element_text(angle = 45, hjust = 1))+</span></span>
<span id="cb79-40"><a href="figure1.html#cb79-40" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>cols)<span class="sc">+</span></span>
<span id="cb79-41"><a href="figure1.html#cb79-41" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb79-42"><a href="figure1.html#cb79-42" tabindex="-1"></a>p1</span>
<span id="cb79-43"><a href="figure1.html#cb79-43" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/04.annotation_pathways.pdf&quot;</span>,p1,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">6</span>)</span>
<span id="cb79-44"><a href="figure1.html#cb79-44" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/04.annotation_pathways.png&quot;</span>,p1,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">6</span>)</span></code></pre></div>
<p><img src="result/figure1/04.annotation.png" width="50%" /><img src="result/figure1/04.annotation_pathways.png" width="50%" /></p>
</div>
</div>
<div id="b" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> 1.2 (B)<a href="figure1.html#b" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="clinical-and-metabolite-analysis" class="section level3 hasAnchor" number="6.2.1">
<h3><span class="header-section-number">6.2.1</span> Clinical and metabolite analysis<a href="figure1.html#clinical-and-metabolite-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="figure1.html#cb80-1" tabindex="-1"></a>metabolites <span class="ot">&lt;-</span> <span class="fu">names</span>(dat_surv)[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)]</span>
<span id="cb80-2"><a href="figure1.html#cb80-2" tabindex="-1"></a><span class="fu">survMet</span>(dat_surv[<span class="sc">-</span><span class="dv">1</span>,],metabolites,<span class="at">cluster_method=</span><span class="st">&quot;mean&quot;</span>,<span class="at">out_dir=</span><span class="st">&quot;result/figure1/survival/&quot;</span>)</span></code></pre></div>
</div>
<div id="time-series-analysis" class="section level3 hasAnchor" number="6.2.2">
<h3><span class="header-section-number">6.2.2</span> Time series analysis<a href="figure1.html#time-series-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="figure1.html#cb81-1" tabindex="-1"></a><span class="fu">names</span>(clinical_index)[<span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="st">&quot;1-methyladenosine&quot;</span></span>
<span id="cb81-2"><a href="figure1.html#cb81-2" tabindex="-1"></a>clinical_index <span class="ot">&lt;-</span> clinical_index <span class="sc">%&gt;%</span></span>
<span id="cb81-3"><a href="figure1.html#cb81-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">group=</span><span class="fu">ifelse</span>(group<span class="sc">==</span><span class="st">&quot;A&quot;</span>,<span class="st">&quot;tumor&quot;</span>,<span class="st">&quot;normal&quot;</span>))</span>
<span id="cb81-4"><a href="figure1.html#cb81-4" tabindex="-1"></a>time_series_ALT <span class="ot">&lt;-</span> <span class="fu">pCliTS</span>(clinical_index,<span class="st">&quot;1-methyladenosine&quot;</span>)</span>
<span id="cb81-5"><a href="figure1.html#cb81-5" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/05.clinical_time_series.pdf&quot;</span>,time_series_ALT,<span class="at">width=</span><span class="dv">7</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb81-6"><a href="figure1.html#cb81-6" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/05.clinical_time_series.png&quot;</span>,time_series_ALT,<span class="at">width=</span><span class="dv">7</span>,<span class="at">height =</span> <span class="dv">5</span>)</span></code></pre></div>
<p><img src="result/figure1/survival/C00022.survival.png" width="50%" /><img src="result/figure1/05.clinical_time_series.png" width="50%" /></p>
</div>
</div>
<div id="c" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> 1.3 (C)<a href="figure1.html#c" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="dimension-reduction-visualization" class="section level3 hasAnchor" number="6.3.1">
<h3><span class="header-section-number">6.3.1</span> Dimension reduction &amp; Visualization<a href="figure1.html#dimension-reduction-visualization" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="figure1.html#cb82-1" tabindex="-1"></a><span class="co">#------------------------------------------------------------------------------</span></span>
<span id="cb82-2"><a href="figure1.html#cb82-2" tabindex="-1"></a><span class="co"># limma analysis</span></span>
<span id="cb82-3"><a href="figure1.html#cb82-3" tabindex="-1"></a><span class="co">#------------------------------------------------------------------------------</span></span>
<span id="cb82-4"><a href="figure1.html#cb82-4" tabindex="-1"></a>group <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;tumor&quot;</span>,<span class="dv">24</span>),<span class="fu">rep</span>(<span class="st">&quot;normal&quot;</span>,<span class="dv">7</span>))</span>
<span id="cb82-5"><a href="figure1.html#cb82-5" tabindex="-1"></a>result_mlimma_all <span class="ot">&lt;-</span> <span class="fu">mlimma</span>(meta_dat,group)</span>
<span id="cb82-6"><a href="figure1.html#cb82-6" tabindex="-1"></a><span class="fu">write.table</span>(result_mlimma_all,<span class="st">&quot;result/figure1/result_mlimma_all.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb82-7"><a href="figure1.html#cb82-7" tabindex="-1"></a></span>
<span id="cb82-8"><a href="figure1.html#cb82-8" tabindex="-1"></a><span class="co">#------------------------------------------------------------------------------</span></span>
<span id="cb82-9"><a href="figure1.html#cb82-9" tabindex="-1"></a><span class="co"># PCA</span></span>
<span id="cb82-10"><a href="figure1.html#cb82-10" tabindex="-1"></a><span class="co">#------------------------------------------------------------------------------</span></span>
<span id="cb82-11"><a href="figure1.html#cb82-11" tabindex="-1"></a>p_out <span class="ot">&lt;-</span> <span class="fu">pPCA</span>(meta_dat,group)</span>
<span id="cb82-12"><a href="figure1.html#cb82-12" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/06.PCA_p2.pdf&quot;</span>,p_out<span class="sc">$</span>p2,<span class="at">width=</span><span class="dv">8</span>,<span class="at">height =</span> <span class="dv">6</span>)</span>
<span id="cb82-13"><a href="figure1.html#cb82-13" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/06.PCA_p2.png&quot;</span>,p_out<span class="sc">$</span>p2,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="fl">3.9</span>)</span>
<span id="cb82-14"><a href="figure1.html#cb82-14" tabindex="-1"></a></span>
<span id="cb82-15"><a href="figure1.html#cb82-15" tabindex="-1"></a><span class="co">#------------------------------------------------------------------------------</span></span>
<span id="cb82-16"><a href="figure1.html#cb82-16" tabindex="-1"></a><span class="co"># OPLS-PCA</span></span>
<span id="cb82-17"><a href="figure1.html#cb82-17" tabindex="-1"></a><span class="co">#------------------------------------------------------------------------------</span></span>
<span id="cb82-18"><a href="figure1.html#cb82-18" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">&quot;result/figure1/06.OPLS-DA.pdf&quot;</span>)</span>
<span id="cb82-19"><a href="figure1.html#cb82-19" tabindex="-1"></a>result_DM_all <span class="ot">&lt;-</span> <span class="fu">DM</span>(<span class="dv">2</span><span class="sc">**</span>meta_dat,group)</span>
<span id="cb82-20"><a href="figure1.html#cb82-20" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb82-21"><a href="figure1.html#cb82-21" tabindex="-1"></a></span>
<span id="cb82-22"><a href="figure1.html#cb82-22" tabindex="-1"></a>oplsda <span class="ot">&lt;-</span> ropls<span class="sc">::</span><span class="fu">opls</span>(<span class="fu">t</span>(<span class="fu">myscale</span>(meta_dat,<span class="at">method=</span><span class="st">&quot;raw_zscore&quot;</span>)), group, <span class="at">predI =</span> <span class="dv">1</span>, <span class="at">orthoI =</span> <span class="dv">1</span>)</span>
<span id="cb82-23"><a href="figure1.html#cb82-23" tabindex="-1"></a></span>
<span id="cb82-24"><a href="figure1.html#cb82-24" tabindex="-1"></a>sample.score <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(oplsda<span class="sc">@</span>scoreMN,<span class="at">group=</span>group,<span class="at">o1=</span>oplsda<span class="sc">@</span>orthoScoreMN[,<span class="dv">1</span>],</span>
<span id="cb82-25"><a href="figure1.html#cb82-25" tabindex="-1"></a>                           <span class="at">label=</span><span class="fu">rownames</span>(oplsda<span class="sc">@</span>scoreMN))</span>
<span id="cb82-26"><a href="figure1.html#cb82-26" tabindex="-1"></a></span>
<span id="cb82-27"><a href="figure1.html#cb82-27" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(sample.score,<span class="fu">aes</span>(p1,o1,<span class="at">color=</span>group))<span class="sc">+</span></span>
<span id="cb82-28"><a href="figure1.html#cb82-28" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">0</span>,<span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>,<span class="at">size=</span><span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb82-29"><a href="figure1.html#cb82-29" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">0</span>,<span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>,<span class="at">size=</span><span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb82-30"><a href="figure1.html#cb82-30" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="fl">1.9</span>)<span class="sc">+</span></span>
<span id="cb82-31"><a href="figure1.html#cb82-31" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_text</span>(ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">label=</span>label),<span class="at">size=</span><span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb82-32"><a href="figure1.html#cb82-32" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>()<span class="sc">+</span></span>
<span id="cb82-33"><a href="figure1.html#cb82-33" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span><span class="fu">c</span>(<span class="st">&quot;#00599F&quot;</span>,<span class="st">&quot;#D01910&quot;</span>))<span class="sc">+</span></span>
<span id="cb82-34"><a href="figure1.html#cb82-34" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb82-35"><a href="figure1.html#cb82-35" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;p1 (20%)&quot;</span>)</span>
<span id="cb82-36"><a href="figure1.html#cb82-36" tabindex="-1"></a><span class="co">#p</span></span>
<span id="cb82-37"><a href="figure1.html#cb82-37" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/06.p.opls-da.pdf&quot;</span>,p,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="fl">4.5</span>)</span>
<span id="cb82-38"><a href="figure1.html#cb82-38" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/06.p.opls-da.png&quot;</span>,p,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="fl">4.5</span>)</span></code></pre></div>
<p><img src="result/figure1/06.PCA_p2.png" width="30%" /><img src="result/figure1/06.opls-sd.png" width="30%" /><img src="result/figure1/06.p.opls-da.png" width="30%" /></p>
</div>
<div id="differential-metabolites-visualization" class="section level3 hasAnchor" number="6.3.2">
<h3><span class="header-section-number">6.3.2</span> Differential metabolites visualization<a href="figure1.html#differential-metabolites-visualization" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="volcano-plot" class="section level4 hasAnchor" number="6.3.2.1">
<h4><span class="header-section-number">6.3.2.1</span> Volcano plot<a href="figure1.html#volcano-plot" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="figure1.html#cb83-1" tabindex="-1"></a>result_DM_all <span class="ot">&lt;-</span> <span class="fu">DM</span>(<span class="dv">2</span><span class="sc">**</span>meta_dat,group)</span>
<span id="cb83-2"><a href="figure1.html#cb83-2" tabindex="-1"></a></span>
<span id="cb83-3"><a href="figure1.html#cb83-3" tabindex="-1"></a>p_Volcano <span class="ot">&lt;-</span> <span class="fu">pVolcano</span>(result_DM_all,<span class="at">foldchange_threshold =</span> <span class="fl">1.5</span>,<span class="at">p_threshold =</span> <span class="fl">0.01</span>)</span>
<span id="cb83-4"><a href="figure1.html#cb83-4" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/07.p_valocano.pdf&quot;</span>,p_Volcano,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="dv">4</span>)</span>
<span id="cb83-5"><a href="figure1.html#cb83-5" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/07.p_valocano.png&quot;</span>,p_Volcano,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="dv">4</span>)</span></code></pre></div>
</div>
<div id="heatmap" class="section level4 hasAnchor" number="6.3.2.2">
<h4><span class="header-section-number">6.3.2.2</span> Heatmap<a href="figure1.html#heatmap" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="figure1.html#cb84-1" tabindex="-1"></a>result_mlimma_all <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/figure1/result_mlimma_all.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb84-2"><a href="figure1.html#cb84-2" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb84-3"><a href="figure1.html#cb84-3" tabindex="-1"></a>result_mlimma_filter <span class="ot">&lt;-</span> result_mlimma_all <span class="sc">%&gt;%</span></span>
<span id="cb84-4"><a href="figure1.html#cb84-4" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">abs</span>(logFC) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb84-5"><a href="figure1.html#cb84-5" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="st">`</span><span class="at">adj.P.Val</span><span class="st">`</span> <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb84-6"><a href="figure1.html#cb84-6" tabindex="-1"></a></span>
<span id="cb84-7"><a href="figure1.html#cb84-7" tabindex="-1"></a>dat_filter <span class="ot">&lt;-</span> meta_dat <span class="sc">%&gt;%</span></span>
<span id="cb84-8"><a href="figure1.html#cb84-8" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;label&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb84-9"><a href="figure1.html#cb84-9" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(label <span class="sc">%in%</span> result_mlimma_filter<span class="sc">$</span>name) <span class="sc">%&gt;%</span></span>
<span id="cb84-10"><a href="figure1.html#cb84-10" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;label&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb84-11"><a href="figure1.html#cb84-11" tabindex="-1"></a>  <span class="fu">head</span>(<span class="at">n=</span><span class="dv">10</span>)</span>
<span id="cb84-12"><a href="figure1.html#cb84-12" tabindex="-1"></a></span>
<span id="cb84-13"><a href="figure1.html#cb84-13" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">&quot;result/figure1/08.p_heatmap.pdf&quot;</span>,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb84-14"><a href="figure1.html#cb84-14" tabindex="-1"></a>p_heatmap <span class="ot">&lt;-</span> MNet<span class="sc">::</span><span class="fu">pHeatmap</span>(dat_filter,group,<span class="at">clustering_distance_cols =</span><span class="st">&quot;manhattan&quot;</span>,</span>
<span id="cb84-15"><a href="figure1.html#cb84-15" tabindex="-1"></a>                            <span class="at">clustering_method=</span><span class="st">&quot;ward.D&quot;</span>,<span class="at">fontsize_row=</span><span class="dv">3</span>)</span>
<span id="cb84-16"><a href="figure1.html#cb84-16" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb84-17"><a href="figure1.html#cb84-17" tabindex="-1"></a></span>
<span id="cb84-18"><a href="figure1.html#cb84-18" tabindex="-1"></a><span class="fu">png</span>(<span class="st">&quot;result/figure1/08.p_heatmap.png&quot;</span>, <span class="at">width =</span> <span class="dv">9</span>, <span class="at">height =</span> <span class="dv">8</span>, <span class="at">units =</span> <span class="st">&#39;in&#39;</span>, <span class="at">res =</span> <span class="dv">200</span> )</span>
<span id="cb84-19"><a href="figure1.html#cb84-19" tabindex="-1"></a>p_heatmap <span class="ot">&lt;-</span> MNet<span class="sc">::</span><span class="fu">pHeatmap</span>(dat_filter,group,<span class="at">clustering_distance_cols =</span><span class="st">&quot;manhattan&quot;</span>,</span>
<span id="cb84-20"><a href="figure1.html#cb84-20" tabindex="-1"></a>                            <span class="at">clustering_method=</span><span class="st">&quot;ward.D&quot;</span>,<span class="at">fontsize_row=</span><span class="dv">3</span>)</span>
<span id="cb84-21"><a href="figure1.html#cb84-21" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
</div>
<div id="z-score-plot" class="section level4 hasAnchor" number="6.3.2.3">
<h4><span class="header-section-number">6.3.2.3</span> Z-score plot<a href="figure1.html#z-score-plot" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="figure1.html#cb85-1" tabindex="-1"></a>p_Zscore <span class="ot">&lt;-</span> <span class="fu">pZscore</span>(dat_filter,group)</span>
<span id="cb85-2"><a href="figure1.html#cb85-2" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/09.p_Zscore.pdf&quot;</span>,p_Zscore,<span class="at">width=</span><span class="dv">3</span>,<span class="at">height =</span> <span class="dv">2</span>)</span>
<span id="cb85-3"><a href="figure1.html#cb85-3" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/09.p_Zscore.png&quot;</span>,p_Zscore,<span class="at">width=</span><span class="dv">3</span>,<span class="at">height =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="result/figure1/07.p_valocano.png" width="30%" /><img src="result/figure1/09.p_Zscore.png" width="30%" /><img src="result/figure1/08.p_heatmap.png" width="30%" /></p>
</div>
</div>
<div id="pathway-enrichment-analyses" class="section level3 hasAnchor" number="6.3.3">
<h3><span class="header-section-number">6.3.3</span> Pathway enrichment analyses<a href="figure1.html#pathway-enrichment-analyses" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="pea" class="section level4 hasAnchor" number="6.3.3.1">
<h4><span class="header-section-number">6.3.3.1</span> PEA<a href="figure1.html#pea" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="figure1.html#cb86-1" tabindex="-1"></a>result_mlimma_all <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/figure1/result_mlimma_all.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb86-2"><a href="figure1.html#cb86-2" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb86-3"><a href="figure1.html#cb86-3" tabindex="-1"></a></span>
<span id="cb86-4"><a href="figure1.html#cb86-4" tabindex="-1"></a>dat_increase <span class="ot">&lt;-</span> result_mlimma_all <span class="sc">%&gt;%</span></span>
<span id="cb86-5"><a href="figure1.html#cb86-5" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC <span class="sc">&gt;</span> <span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb86-6"><a href="figure1.html#cb86-6" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(P.Value <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb86-7"><a href="figure1.html#cb86-7" tabindex="-1"></a></span>
<span id="cb86-8"><a href="figure1.html#cb86-8" tabindex="-1"></a>dat_decrease <span class="ot">&lt;-</span> result_mlimma_all <span class="sc">%&gt;%</span></span>
<span id="cb86-9"><a href="figure1.html#cb86-9" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb86-10"><a href="figure1.html#cb86-10" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(P.Value <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb86-11"><a href="figure1.html#cb86-11" tabindex="-1"></a></span>
<span id="cb86-12"><a href="figure1.html#cb86-12" tabindex="-1"></a>kegg_all <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(dat_increase<span class="sc">$</span>name,dat_decrease<span class="sc">$</span>name))</span>
<span id="cb86-13"><a href="figure1.html#cb86-13" tabindex="-1"></a></span>
<span id="cb86-14"><a href="figure1.html#cb86-14" tabindex="-1"></a>pathway_result <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(kegg_all,<span class="at">out=</span><span class="st">&quot;metabolite&quot;</span>)</span>
<span id="cb86-15"><a href="figure1.html#cb86-15" tabindex="-1"></a></span>
<span id="cb86-16"><a href="figure1.html#cb86-16" tabindex="-1"></a>result <span class="ot">&lt;-</span> pathway_result</span>
<span id="cb86-17"><a href="figure1.html#cb86-17" tabindex="-1"></a></span>
<span id="cb86-18"><a href="figure1.html#cb86-18" tabindex="-1"></a>kegg_pathway_uniq <span class="ot">&lt;-</span> PathwayExtendData <span class="sc">%&gt;%</span></span>
<span id="cb86-19"><a href="figure1.html#cb86-19" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(kegg_pathwayname,kegg_category) <span class="sc">%&gt;%</span></span>
<span id="cb86-20"><a href="figure1.html#cb86-20" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;PATHWAY&quot;</span><span class="ot">=</span><span class="st">&quot;kegg_pathwayname&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb86-21"><a href="figure1.html#cb86-21" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;pathway_type&quot;</span><span class="ot">=</span><span class="st">&quot;kegg_category&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb86-22"><a href="figure1.html#cb86-22" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb86-23"><a href="figure1.html#cb86-23" tabindex="-1"></a></span>
<span id="cb86-24"><a href="figure1.html#cb86-24" tabindex="-1"></a>p_cutoff <span class="ot">&lt;-</span> <span class="fl">0.0005</span></span>
<span id="cb86-25"><a href="figure1.html#cb86-25" tabindex="-1"></a>result_1 <span class="ot">&lt;-</span> result<span class="sc">$</span>output <span class="sc">%&gt;%</span></span>
<span id="cb86-26"><a href="figure1.html#cb86-26" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pvalue <span class="sc">&lt;</span> p_cutoff) <span class="sc">%&gt;%</span></span>
<span id="cb86-27"><a href="figure1.html#cb86-27" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(kegg_pathway_uniq,<span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;name&quot;</span><span class="ot">=</span><span class="st">&quot;PATHWAY&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb86-28"><a href="figure1.html#cb86-28" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(pvalue)</span>
<span id="cb86-29"><a href="figure1.html#cb86-29" tabindex="-1"></a></span>
<span id="cb86-30"><a href="figure1.html#cb86-30" tabindex="-1"></a>pathway_hh <span class="ot">&lt;-</span> <span class="fu">unique</span>(result_1<span class="sc">$</span>pathway_type)</span>
<span id="cb86-31"><a href="figure1.html#cb86-31" tabindex="-1"></a></span>
<span id="cb86-32"><a href="figure1.html#cb86-32" tabindex="-1"></a>result_1 <span class="ot">&lt;-</span> result_1 <span class="sc">%&gt;%</span></span>
<span id="cb86-33"><a href="figure1.html#cb86-33" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">match</span>(pathway_type,pathway_hh))</span>
<span id="cb86-34"><a href="figure1.html#cb86-34" tabindex="-1"></a></span>
<span id="cb86-35"><a href="figure1.html#cb86-35" tabindex="-1"></a>result_1<span class="sc">$</span>name <span class="ot">&lt;-</span> <span class="fu">factor</span>(result_1<span class="sc">$</span>name,<span class="at">levels =</span> <span class="fu">rev</span>(result_1<span class="sc">$</span>name))</span>
<span id="cb86-36"><a href="figure1.html#cb86-36" tabindex="-1"></a>result_1<span class="sc">$</span>pathway_type <span class="ot">&lt;-</span> <span class="fu">factor</span>(result_1<span class="sc">$</span>pathway_type,<span class="at">levels=</span><span class="fu">unique</span>(kegg_pathway_uniq<span class="sc">$</span>pathway_type))</span>
<span id="cb86-37"><a href="figure1.html#cb86-37" tabindex="-1"></a></span>
<span id="cb86-38"><a href="figure1.html#cb86-38" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(result_1,<span class="fu">aes</span>(name,<span class="sc">-</span><span class="fu">log10</span>(pvalue)))<span class="sc">+</span></span>
<span id="cb86-39"><a href="figure1.html#cb86-39" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>,<span class="fu">aes</span>(<span class="at">fill=</span>pathway_type))<span class="sc">+</span></span>
<span id="cb86-40"><a href="figure1.html#cb86-40" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;Set3&quot;</span>),<span class="at">name=</span><span class="st">&quot;Pathway Category&quot;</span>,</span>
<span id="cb86-41"><a href="figure1.html#cb86-41" tabindex="-1"></a>                    <span class="at">breaks=</span><span class="fu">unique</span>(kegg_pathway_uniq<span class="sc">$</span>pathway_type))<span class="sc">+</span></span>
<span id="cb86-42"><a href="figure1.html#cb86-42" tabindex="-1"></a>  <span class="fu">coord_flip</span>()<span class="sc">+</span></span>
<span id="cb86-43"><a href="figure1.html#cb86-43" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb86-44"><a href="figure1.html#cb86-44" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="cn">NULL</span>)</span>
<span id="cb86-45"><a href="figure1.html#cb86-45" tabindex="-1"></a>p1</span>
<span id="cb86-46"><a href="figure1.html#cb86-46" tabindex="-1"></a></span>
<span id="cb86-47"><a href="figure1.html#cb86-47" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/10.pathway_enrichment.pdf&quot;</span>,p1,<span class="at">width =</span> <span class="dv">8</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb86-48"><a href="figure1.html#cb86-48" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/10.pathway_enrichment.png&quot;</span>,p1,<span class="at">width =</span> <span class="dv">8</span>,<span class="at">height =</span> <span class="dv">5</span>)</span></code></pre></div>
</div>
<div id="da-score" class="section level4 hasAnchor" number="6.3.3.2">
<h4><span class="header-section-number">6.3.3.2</span> DA score<a href="figure1.html#da-score" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="figure1.html#cb87-1" tabindex="-1"></a>result_da <span class="ot">&lt;-</span> <span class="fu">DAscore</span>(dat_increase<span class="sc">$</span>name,dat_decrease<span class="sc">$</span>name,result_mlimma_all<span class="sc">$</span>name,<span class="at">sort_plot =</span> <span class="st">&quot;category&quot;</span>,<span class="at">out=</span><span class="st">&quot;metabolite&quot;</span> )</span>
<span id="cb87-2"><a href="figure1.html#cb87-2" tabindex="-1"></a></span>
<span id="cb87-3"><a href="figure1.html#cb87-3" tabindex="-1"></a>result <span class="ot">&lt;-</span> result_da<span class="sc">$</span>result</span>
<span id="cb87-4"><a href="figure1.html#cb87-4" tabindex="-1"></a>min_measured_num <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb87-5"><a href="figure1.html#cb87-5" tabindex="-1"></a>result_filter <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb87-6"><a href="figure1.html#cb87-6" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Measured_members_num<span class="sc">&gt;=</span>min_measured_num) <span class="sc">%&gt;%</span></span>
<span id="cb87-7"><a href="figure1.html#cb87-7" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Pathway <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Histidine metabolism&quot;</span>,<span class="st">&quot;Tyrosine metabolism&quot;</span>,</span>
<span id="cb87-8"><a href="figure1.html#cb87-8" tabindex="-1"></a>                               <span class="st">&quot;Citrate cycle (TCA cycle)&quot;</span>,<span class="st">&quot;Pyruvate metabolism&quot;</span>,</span>
<span id="cb87-9"><a href="figure1.html#cb87-9" tabindex="-1"></a>                               <span class="st">&quot;Oxidative phosphorylation&quot;</span>,<span class="st">&quot;Sulfur metabolism&quot;</span>,</span>
<span id="cb87-10"><a href="figure1.html#cb87-10" tabindex="-1"></a>                               <span class="st">&quot;Glycerophospholipid metabolism&quot;</span>,<span class="st">&quot;Glycerolipid metabolism&quot;</span>,</span>
<span id="cb87-11"><a href="figure1.html#cb87-11" tabindex="-1"></a>                               <span class="st">&quot;Biosynthesis of unsaturated fatty acids&quot;</span>,<span class="st">&quot;Fatty acid biosynthesis&quot;</span>,</span>
<span id="cb87-12"><a href="figure1.html#cb87-12" tabindex="-1"></a>                               <span class="st">&quot;Pyrimidine metabolism&quot;</span>,<span class="st">&quot;Purine metabolism&quot;</span>))</span>
<span id="cb87-13"><a href="figure1.html#cb87-13" tabindex="-1"></a></span>
<span id="cb87-14"><a href="figure1.html#cb87-14" tabindex="-1"></a>result_filter <span class="ot">&lt;-</span> result_filter <span class="sc">%&gt;%</span></span>
<span id="cb87-15"><a href="figure1.html#cb87-15" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(DA_score) <span class="sc">%&gt;%</span></span>
<span id="cb87-16"><a href="figure1.html#cb87-16" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="st">`</span><span class="at">Pathway Category</span><span class="st">`</span>)</span>
<span id="cb87-17"><a href="figure1.html#cb87-17" tabindex="-1"></a>  </span>
<span id="cb87-18"><a href="figure1.html#cb87-18" tabindex="-1"></a>result_filter<span class="sc">$</span>Pathway <span class="ot">&lt;-</span> <span class="fu">factor</span>(result_filter<span class="sc">$</span>Pathway,<span class="at">levels=</span>result_filter<span class="sc">$</span>Pathway)</span>
<span id="cb87-19"><a href="figure1.html#cb87-19" tabindex="-1"></a>pathway_data <span class="ot">&lt;-</span> PathwayExtendData <span class="sc">%&gt;%</span></span>
<span id="cb87-20"><a href="figure1.html#cb87-20" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(type<span class="sc">==</span><span class="st">&quot;metabolite&quot;</span>)</span>
<span id="cb87-21"><a href="figure1.html#cb87-21" tabindex="-1"></a></span>
<span id="cb87-22"><a href="figure1.html#cb87-22" tabindex="-1"></a>p <span class="ot">&lt;-</span> ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(result_filter)<span class="sc">+</span></span>
<span id="cb87-23"><a href="figure1.html#cb87-23" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">x=</span>Pathway,<span class="at">y=</span>DA_score,<span class="at">size=</span><span class="fu">log2</span>(Measured_members_num),<span class="at">color=</span><span class="st">`</span><span class="at">Pathway Category</span><span class="st">`</span>))<span class="sc">+</span></span>
<span id="cb87-24"><a href="figure1.html#cb87-24" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_pointrange</span>(ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">x=</span>Pathway,<span class="at">y=</span>DA_score,<span class="at">ymin=</span><span class="dv">0</span>,<span class="at">ymax=</span>DA_score,<span class="at">color=</span><span class="st">`</span><span class="at">Pathway Category</span><span class="st">`</span>))<span class="sc">+</span></span>
<span id="cb87-25"><a href="figure1.html#cb87-25" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;Set3&quot;</span>),<span class="at">name=</span><span class="st">&quot;Pathway Category&quot;</span>,</span>
<span id="cb87-26"><a href="figure1.html#cb87-26" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">unique</span>(pathway_data<span class="sc">$</span>kegg_category))<span class="sc">+</span></span>
<span id="cb87-27"><a href="figure1.html#cb87-27" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">coord_flip</span>()<span class="sc">+</span></span>
<span id="cb87-28"><a href="figure1.html#cb87-28" tabindex="-1"></a>  <span class="co">#      ggplot2::ylab(&quot;DA score&quot;)+</span></span>
<span id="cb87-29"><a href="figure1.html#cb87-29" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">xlab</span>(<span class="cn">NULL</span>)<span class="sc">+</span></span>
<span id="cb87-30"><a href="figure1.html#cb87-30" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb87-31"><a href="figure1.html#cb87-31" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/11.DA_score.pdf&quot;</span>,p,<span class="at">width=</span><span class="dv">7</span>,<span class="at">height =</span> <span class="dv">4</span>)</span>
<span id="cb87-32"><a href="figure1.html#cb87-32" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/11.DA_score.png&quot;</span>,p,<span class="at">width=</span><span class="dv">7</span>,<span class="at">height =</span> <span class="dv">4</span>)</span></code></pre></div>
</div>
<div id="msea" class="section level4 hasAnchor" number="6.3.3.3">
<h4><span class="header-section-number">6.3.3.3</span> MSEA<a href="figure1.html#msea" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="figure1.html#cb88-1" tabindex="-1"></a>dd_arrange <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/figure1/result_mlimma_all.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb88-2"><a href="figure1.html#cb88-2" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb88-3"><a href="figure1.html#cb88-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(logFC)</span>
<span id="cb88-4"><a href="figure1.html#cb88-4" tabindex="-1"></a>d_logFC <span class="ot">&lt;-</span> dd_arrange<span class="sc">$</span>logFC</span>
<span id="cb88-5"><a href="figure1.html#cb88-5" tabindex="-1"></a><span class="fu">names</span>(d_logFC) <span class="ot">&lt;-</span> dd_arrange<span class="sc">$</span>name</span>
<span id="cb88-6"><a href="figure1.html#cb88-6" tabindex="-1"></a>result_msea <span class="ot">&lt;-</span> <span class="fu">MSEA</span>(d_logFC)</span>
<span id="cb88-7"><a href="figure1.html#cb88-7" tabindex="-1"></a></span>
<span id="cb88-8"><a href="figure1.html#cb88-8" tabindex="-1"></a>result_msea <span class="ot">&lt;-</span> result_msea <span class="sc">%&gt;%</span></span>
<span id="cb88-9"><a href="figure1.html#cb88-9" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">leadingEdge=</span><span class="fu">as.character</span>(leadingEdge))</span>
<span id="cb88-10"><a href="figure1.html#cb88-10" tabindex="-1"></a></span>
<span id="cb88-11"><a href="figure1.html#cb88-11" tabindex="-1"></a><span class="fu">write.table</span>(result_msea,<span class="st">&quot;result/figure1/12.MSEA.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb88-12"><a href="figure1.html#cb88-12" tabindex="-1"></a></span>
<span id="cb88-13"><a href="figure1.html#cb88-13" tabindex="-1"></a>p_MSEA <span class="ot">&lt;-</span> <span class="fu">pMSEA</span>(<span class="st">&quot;Pyrimidine metabolism&quot;</span>,d_logFC)</span>
<span id="cb88-14"><a href="figure1.html#cb88-14" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/12.p_MSEA_pyrimidine.pdf&quot;</span>,p_MSEA,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="dv">4</span>)</span>
<span id="cb88-15"><a href="figure1.html#cb88-15" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/12.p_MSEA_pyrimidine.png&quot;</span>,p_MSEA,<span class="at">width=</span><span class="dv">5</span>,<span class="at">height =</span> <span class="dv">4</span>)</span></code></pre></div>
<p><img src="result/figure1/10.pathway_enrichment.png" width="30%" /><img src="result/figure1/11.DA_score.png" width="30%" /><img src="result/figure1/12.p_MSEA_pyrimidine.png" width="30%" /></p>
</div>
</div>
</div>
<div id="d" class="section level2 hasAnchor" number="6.4">
<h2><span class="header-section-number">6.4</span> 1.4 (D)<a href="figure1.html#d" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="extended-pathway-enrichment-analysis-epea" class="section level3 hasAnchor" number="6.4.1">
<h3><span class="header-section-number">6.4.1</span> Extended pathway enrichment analysis (EPEA)<a href="figure1.html#extended-pathway-enrichment-analysis-epea" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="figure1.html#cb89-1" tabindex="-1"></a><span class="fu">library</span>(MNet)</span>
<span id="cb89-2"><a href="figure1.html#cb89-2" tabindex="-1"></a></span>
<span id="cb89-3"><a href="figure1.html#cb89-3" tabindex="-1"></a>group[<span class="fu">which</span>(group<span class="sc">==</span><span class="st">&quot;normal&quot;</span>)] <span class="ot">&lt;-</span> <span class="st">&quot;a&quot;</span></span>
<span id="cb89-4"><a href="figure1.html#cb89-4" tabindex="-1"></a>group[<span class="fu">which</span>(group<span class="sc">==</span><span class="st">&quot;tumor&quot;</span>)] <span class="ot">&lt;-</span> <span class="st">&quot;normal&quot;</span></span>
<span id="cb89-5"><a href="figure1.html#cb89-5" tabindex="-1"></a>group[<span class="fu">which</span>(group<span class="sc">==</span><span class="st">&quot;a&quot;</span>)] <span class="ot">&lt;-</span> <span class="st">&quot;tumor&quot;</span></span>
<span id="cb89-6"><a href="figure1.html#cb89-6" tabindex="-1"></a></span>
<span id="cb89-7"><a href="figure1.html#cb89-7" tabindex="-1"></a>all_meta <span class="ot">&lt;-</span> <span class="fu">mlimma</span>(meta_dat,group)</span>
<span id="cb89-8"><a href="figure1.html#cb89-8" tabindex="-1"></a>all_gene <span class="ot">&lt;-</span> <span class="fu">mlimma</span>(gene_dat,group)</span>
<span id="cb89-9"><a href="figure1.html#cb89-9" tabindex="-1"></a></span>
<span id="cb89-10"><a href="figure1.html#cb89-10" tabindex="-1"></a>diff_gene <span class="ot">&lt;-</span> all_gene <span class="sc">%&gt;%</span></span>
<span id="cb89-11"><a href="figure1.html#cb89-11" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">abs</span>(logFC) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb89-12"><a href="figure1.html#cb89-12" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb89-13"><a href="figure1.html#cb89-13" tabindex="-1"></a></span>
<span id="cb89-14"><a href="figure1.html#cb89-14" tabindex="-1"></a>diff_meta <span class="ot">&lt;-</span> all_meta <span class="sc">%&gt;%</span></span>
<span id="cb89-15"><a href="figure1.html#cb89-15" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">abs</span>(logFC) <span class="sc">&gt;</span> <span class="fl">0.58</span>) <span class="sc">%&gt;%</span></span>
<span id="cb89-16"><a href="figure1.html#cb89-16" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb89-17"><a href="figure1.html#cb89-17" tabindex="-1"></a></span>
<span id="cb89-18"><a href="figure1.html#cb89-18" tabindex="-1"></a>diff_all <span class="ot">&lt;-</span> <span class="fu">rbind</span>(diff_gene,diff_meta)</span>
<span id="cb89-19"><a href="figure1.html#cb89-19" tabindex="-1"></a></span>
<span id="cb89-20"><a href="figure1.html#cb89-20" tabindex="-1"></a>result_gene <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(diff_gene<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;gene&quot;</span>,<span class="at">p_cutoff=</span><span class="fl">1.5</span>)</span>
<span id="cb89-21"><a href="figure1.html#cb89-21" tabindex="-1"></a>result_meta <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(diff_meta<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;metabolite&quot;</span>,<span class="at">p_cutoff=</span><span class="fl">1.5</span>)</span>
<span id="cb89-22"><a href="figure1.html#cb89-22" tabindex="-1"></a>result_all <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(diff_all<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;Extended&quot;</span>,<span class="at">p_cutoff=</span><span class="fl">1.5</span>)</span>
<span id="cb89-23"><a href="figure1.html#cb89-23" tabindex="-1"></a></span>
<span id="cb89-24"><a href="figure1.html#cb89-24" tabindex="-1"></a>pathway_all <span class="ot">&lt;-</span> <span class="fu">rbind</span>(result_gene<span class="sc">$</span>output,result_meta<span class="sc">$</span>output,result_all<span class="sc">$</span>output) <span class="sc">%&gt;%</span></span>
<span id="cb89-25"><a href="figure1.html#cb89-25" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(adjp <span class="sc">&lt;</span> <span class="fl">0.001</span>) <span class="sc">%&gt;%</span></span>
<span id="cb89-26"><a href="figure1.html#cb89-26" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(name) <span class="sc">%&gt;%</span></span>
<span id="cb89-27"><a href="figure1.html#cb89-27" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb89-28"><a href="figure1.html#cb89-28" tabindex="-1"></a></span>
<span id="cb89-29"><a href="figure1.html#cb89-29" tabindex="-1"></a>result_gene_filter <span class="ot">&lt;-</span> result_gene<span class="sc">$</span>output <span class="sc">%&gt;%</span></span>
<span id="cb89-30"><a href="figure1.html#cb89-30" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb89-31"><a href="figure1.html#cb89-31" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(name,nOverlap,adjp) <span class="sc">%&gt;%</span></span>
<span id="cb89-32"><a href="figure1.html#cb89-32" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(name <span class="sc">%in%</span> pathway_all) <span class="sc">%&gt;%</span></span>
<span id="cb89-33"><a href="figure1.html#cb89-33" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb89-34"><a href="figure1.html#cb89-34" tabindex="-1"></a></span>
<span id="cb89-35"><a href="figure1.html#cb89-35" tabindex="-1"></a>result_meta_filter <span class="ot">&lt;-</span> result_meta<span class="sc">$</span>output <span class="sc">%&gt;%</span></span>
<span id="cb89-36"><a href="figure1.html#cb89-36" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb89-37"><a href="figure1.html#cb89-37" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(name,nOverlap,adjp) <span class="sc">%&gt;%</span></span>
<span id="cb89-38"><a href="figure1.html#cb89-38" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(name <span class="sc">%in%</span> pathway_all) <span class="sc">%&gt;%</span></span>
<span id="cb89-39"><a href="figure1.html#cb89-39" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type=</span><span class="st">&quot;metabolite&quot;</span>)</span>
<span id="cb89-40"><a href="figure1.html#cb89-40" tabindex="-1"></a></span>
<span id="cb89-41"><a href="figure1.html#cb89-41" tabindex="-1"></a>result_all_filter <span class="ot">&lt;-</span> result_all<span class="sc">$</span>output <span class="sc">%&gt;%</span></span>
<span id="cb89-42"><a href="figure1.html#cb89-42" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb89-43"><a href="figure1.html#cb89-43" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(name,nOverlap,adjp) <span class="sc">%&gt;%</span></span>
<span id="cb89-44"><a href="figure1.html#cb89-44" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(name <span class="sc">%in%</span> pathway_all) <span class="sc">%&gt;%</span></span>
<span id="cb89-45"><a href="figure1.html#cb89-45" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type=</span><span class="st">&quot;all&quot;</span>)</span>
<span id="cb89-46"><a href="figure1.html#cb89-46" tabindex="-1"></a></span>
<span id="cb89-47"><a href="figure1.html#cb89-47" tabindex="-1"></a>all <span class="ot">&lt;-</span> <span class="fu">rbind</span>(result_gene_filter,result_meta_filter,result_all_filter) <span class="sc">%&gt;%</span></span>
<span id="cb89-48"><a href="figure1.html#cb89-48" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">adjp=</span><span class="fu">ifelse</span>(adjp <span class="sc">&lt;</span> <span class="fl">1.3e-04</span> ,<span class="fl">1.3e-04</span>,adjp)) <span class="sc">%&gt;%</span></span>
<span id="cb89-49"><a href="figure1.html#cb89-49" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type=</span><span class="fu">factor</span>(type,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;metabolite&quot;</span>,<span class="st">&quot;gene&quot;</span>,<span class="st">&quot;all&quot;</span>)))</span>
<span id="cb89-50"><a href="figure1.html#cb89-50" tabindex="-1"></a></span>
<span id="cb89-51"><a href="figure1.html#cb89-51" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(all,<span class="fu">aes</span>(type,name))<span class="sc">+</span></span>
<span id="cb89-52"><a href="figure1.html#cb89-52" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">size=</span>nOverlap,<span class="at">color=</span><span class="sc">-</span><span class="fu">log10</span>(adjp)))<span class="sc">+</span></span>
<span id="cb89-53"><a href="figure1.html#cb89-53" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb89-54"><a href="figure1.html#cb89-54" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.border =</span> <span class="fu">element_blank</span>(),<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb89-55"><a href="figure1.html#cb89-55" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),<span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>))<span class="sc">+</span></span>
<span id="cb89-56"><a href="figure1.html#cb89-56" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low=</span><span class="st">&quot;white&quot;</span>,<span class="at">high=</span><span class="st">&quot;red&quot;</span>)<span class="sc">+</span></span>
<span id="cb89-57"><a href="figure1.html#cb89-57" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="cn">NULL</span>,<span class="at">y=</span><span class="st">&quot;Pathway name&quot;</span>)</span>
<span id="cb89-58"><a href="figure1.html#cb89-58" tabindex="-1"></a></span>
<span id="cb89-59"><a href="figure1.html#cb89-59" tabindex="-1"></a></span>
<span id="cb89-60"><a href="figure1.html#cb89-60" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/13.EPEA.pdf&quot;</span>,p,<span class="at">width=</span><span class="dv">6</span>,<span class="at">height =</span> <span class="dv">4</span>)</span>
<span id="cb89-61"><a href="figure1.html#cb89-61" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/13.EPEA.png&quot;</span>,p,<span class="at">width=</span><span class="dv">6</span>,<span class="at">height =</span> <span class="dv">4</span>)</span></code></pre></div>
<p><img src="result/figure1/13.EPEA.png" width="70%" style="display: block; margin: auto;" /></p>
</div>
<div id="extended-differential-abundance-eda-score" class="section level3 hasAnchor" number="6.4.2">
<h3><span class="header-section-number">6.4.2</span> Extended differential abundance (EDA) score<a href="figure1.html#extended-differential-abundance-eda-score" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="figure1.html#cb90-1" tabindex="-1"></a>meta <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/unique/06.DA_meta.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb90-2"><a href="figure1.html#cb90-2" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb90-3"><a href="figure1.html#cb90-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Measured_members_num <span class="sc">&gt;=</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb90-4"><a href="figure1.html#cb90-4" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Type=</span><span class="st">&quot;metabolite&quot;</span>)</span>
<span id="cb90-5"><a href="figure1.html#cb90-5" tabindex="-1"></a></span>
<span id="cb90-6"><a href="figure1.html#cb90-6" tabindex="-1"></a>gene <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/unique/05.DA_gene.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb90-7"><a href="figure1.html#cb90-7" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb90-8"><a href="figure1.html#cb90-8" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Measured_members_num <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb90-9"><a href="figure1.html#cb90-9" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Type=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb90-10"><a href="figure1.html#cb90-10" tabindex="-1"></a></span>
<span id="cb90-11"><a href="figure1.html#cb90-11" tabindex="-1"></a>all <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/unique/07.DA_all.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb90-12"><a href="figure1.html#cb90-12" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb90-13"><a href="figure1.html#cb90-13" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Measured_members_num <span class="sc">&gt;=</span> <span class="dv">13</span>) <span class="sc">%&gt;%</span></span>
<span id="cb90-14"><a href="figure1.html#cb90-14" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Type=</span><span class="st">&quot;metabolite+gene&quot;</span>)</span>
<span id="cb90-15"><a href="figure1.html#cb90-15" tabindex="-1"></a></span>
<span id="cb90-16"><a href="figure1.html#cb90-16" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">rbind</span>(meta,gene,all) <span class="sc">%&gt;%</span></span>
<span id="cb90-17"><a href="figure1.html#cb90-17" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>Pathway <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Sulfur metabolism&quot;</span>,<span class="st">&quot;Fructose and mannose metabolism&quot;</span>,<span class="st">&quot;Ascorbate and aldarate metabolism&quot;</span>,</span>
<span id="cb90-18"><a href="figure1.html#cb90-18" tabindex="-1"></a>                                <span class="st">&quot;Galactose metabolism&quot;</span>,<span class="st">&quot;Butanoate metabolism&quot;</span>,<span class="st">&quot;Pentose and glucuronate interconversions&quot;</span>,<span class="st">&quot;Starch and sucrose metabolism&quot;</span>,</span>
<span id="cb90-19"><a href="figure1.html#cb90-19" tabindex="-1"></a>                                <span class="st">&quot;Amino sugar and nucleotide sugar metabolism&quot;</span>,<span class="st">&quot;Propanoate metabolism&quot;</span>,<span class="st">&quot;Inositol phosphate metabolism&quot;</span>,</span>
<span id="cb90-20"><a href="figure1.html#cb90-20" tabindex="-1"></a>                                <span class="st">&quot;Tryptophan metabolism&quot;</span>,</span>
<span id="cb90-21"><a href="figure1.html#cb90-21" tabindex="-1"></a>                                <span class="st">&quot;Selenocompound metabolism&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb90-22"><a href="figure1.html#cb90-22" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="st">`</span><span class="at">Pathway Category</span><span class="st">`</span> <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Metabolism of terpenoids and polyketides&quot;</span>,<span class="st">&quot;Metabolism of cofactors and vitamins&quot;</span>,</span>
<span id="cb90-23"><a href="figure1.html#cb90-23" tabindex="-1"></a>                                           <span class="st">&quot;Biosynthesis of other secondary metabolites&quot;</span>,<span class="st">&quot;Xenobiotics biodegradation and metabolism&quot;</span>,</span>
<span id="cb90-24"><a href="figure1.html#cb90-24" tabindex="-1"></a>                                           <span class="st">&quot;Glycan biosynthesis and metabolism&quot;</span>,<span class="st">&quot;Metabolism of other amino acids&quot;</span>))</span>
<span id="cb90-25"><a href="figure1.html#cb90-25" tabindex="-1"></a></span>
<span id="cb90-26"><a href="figure1.html#cb90-26" tabindex="-1"></a>result_filter <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb90-27"><a href="figure1.html#cb90-27" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Pathway <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Sphingolipid metabolism&quot;</span>,<span class="st">&quot;Glycerophospholipid metabolism&quot;</span>,</span>
<span id="cb90-28"><a href="figure1.html#cb90-28" tabindex="-1"></a>                               <span class="st">&quot;Fatty acid biosynthesis&quot;</span>,<span class="st">&quot;Glycerolipid metabolism&quot;</span>,                            </span>
<span id="cb90-29"><a href="figure1.html#cb90-29" tabindex="-1"></a>                               <span class="st">&quot;Biosynthesis of unsaturated fatty acids&quot;</span>,<span class="st">&quot;Fatty acid elongation&quot;</span>,                            </span>
<span id="cb90-30"><a href="figure1.html#cb90-30" tabindex="-1"></a>                               <span class="st">&quot;Steroid biosynthesis&quot;</span>,<span class="st">&quot;Purine metabolism&quot;</span>,                                </span>
<span id="cb90-31"><a href="figure1.html#cb90-31" tabindex="-1"></a>                               <span class="st">&quot;Pyrimidine metabolism&quot;</span>,<span class="st">&quot;Histidine metabolism&quot;</span>,<span class="st">&quot;Tyrosine metabolism&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb90-32"><a href="figure1.html#cb90-32" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">DA_score=</span><span class="fu">ifelse</span>(Pathway<span class="sc">==</span><span class="st">&quot;Histidine metabolism&quot;</span> <span class="sc">&amp;</span> Type<span class="sc">==</span><span class="st">&quot;metabolite&quot;</span>,<span class="sc">-</span><span class="fl">0.2</span>,</span>
<span id="cb90-33"><a href="figure1.html#cb90-33" tabindex="-1"></a>                                <span class="fu">ifelse</span>(Pathway<span class="sc">==</span><span class="st">&quot;Sphingolipid metabolism&quot;</span> <span class="sc">&amp;</span> Type<span class="sc">==</span><span class="st">&quot;metabolite&quot;</span>,<span class="sc">-</span><span class="fl">0.2</span>,DA_score))) <span class="sc">%&gt;%</span></span>
<span id="cb90-34"><a href="figure1.html#cb90-34" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(DA_score) <span class="sc">%&gt;%</span></span>
<span id="cb90-35"><a href="figure1.html#cb90-35" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="st">`</span><span class="at">Pathway Category</span><span class="st">`</span>)</span>
<span id="cb90-36"><a href="figure1.html#cb90-36" tabindex="-1"></a></span>
<span id="cb90-37"><a href="figure1.html#cb90-37" tabindex="-1"></a>result_filter<span class="sc">$</span>Pathway <span class="ot">&lt;-</span> <span class="fu">factor</span>(result_filter<span class="sc">$</span>Pathway,<span class="at">levels=</span><span class="fu">unique</span>(result_filter<span class="sc">$</span>Pathway))</span>
<span id="cb90-38"><a href="figure1.html#cb90-38" tabindex="-1"></a></span>
<span id="cb90-39"><a href="figure1.html#cb90-39" tabindex="-1"></a>kegg_pathway_uniq <span class="ot">&lt;-</span> PathwayExtendData <span class="sc">%&gt;%</span></span>
<span id="cb90-40"><a href="figure1.html#cb90-40" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(kegg_pathwayname,kegg_category) <span class="sc">%&gt;%</span></span>
<span id="cb90-41"><a href="figure1.html#cb90-41" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;PATHWAY&quot;</span><span class="ot">=</span><span class="st">&quot;kegg_pathwayname&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb90-42"><a href="figure1.html#cb90-42" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;pathway_type&quot;</span><span class="ot">=</span><span class="st">&quot;kegg_category&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb90-43"><a href="figure1.html#cb90-43" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb90-44"><a href="figure1.html#cb90-44" tabindex="-1"></a></span>
<span id="cb90-45"><a href="figure1.html#cb90-45" tabindex="-1"></a>result_filter<span class="sc">$</span>Type <span class="ot">&lt;-</span> <span class="fu">factor</span>(result_filter<span class="sc">$</span>Type,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;metabolite&quot;</span>,<span class="st">&quot;gene&quot;</span>,<span class="st">&quot;metabolite+gene&quot;</span>))</span>
<span id="cb90-46"><a href="figure1.html#cb90-46" tabindex="-1"></a></span>
<span id="cb90-47"><a href="figure1.html#cb90-47" tabindex="-1"></a></span>
<span id="cb90-48"><a href="figure1.html#cb90-48" tabindex="-1"></a>p <span class="ot">&lt;-</span> ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(result_filter)<span class="sc">+</span></span>
<span id="cb90-49"><a href="figure1.html#cb90-49" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">x=</span>Pathway,<span class="at">y=</span>DA_score,<span class="at">size=</span><span class="fu">log2</span>(Measured_members_num),<span class="at">color=</span><span class="st">`</span><span class="at">Pathway Category</span><span class="st">`</span>))<span class="sc">+</span></span>
<span id="cb90-50"><a href="figure1.html#cb90-50" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">geom_pointrange</span>(ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">x=</span>Pathway,<span class="at">y=</span>DA_score,<span class="at">ymin=</span><span class="dv">0</span>,<span class="at">ymax=</span>DA_score,<span class="at">color=</span><span class="st">`</span><span class="at">Pathway Category</span><span class="st">`</span>))<span class="sc">+</span></span>
<span id="cb90-51"><a href="figure1.html#cb90-51" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;Set3&quot;</span>),<span class="at">name=</span><span class="st">&quot;Pathway Category&quot;</span>,</span>
<span id="cb90-52"><a href="figure1.html#cb90-52" tabindex="-1"></a>                     <span class="at">breaks=</span><span class="fu">unique</span>(kegg_pathway_uniq<span class="sc">$</span>pathway_type))<span class="sc">+</span></span>
<span id="cb90-53"><a href="figure1.html#cb90-53" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb90-54"><a href="figure1.html#cb90-54" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb90-55"><a href="figure1.html#cb90-55" tabindex="-1"></a>        <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour =</span> <span class="st">&quot;black&quot;</span>))<span class="sc">+</span></span>
<span id="cb90-56"><a href="figure1.html#cb90-56" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="fu">c</span>(<span class="dv">0</span>))<span class="sc">+</span></span>
<span id="cb90-57"><a href="figure1.html#cb90-57" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">coord_flip</span>()<span class="sc">+</span></span>
<span id="cb90-58"><a href="figure1.html#cb90-58" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">xlab</span>(<span class="cn">NULL</span>)<span class="sc">+</span></span>
<span id="cb90-59"><a href="figure1.html#cb90-59" tabindex="-1"></a>  <span class="fu">facet_grid</span>(. <span class="sc">~</span> Type,<span class="at">scales=</span><span class="st">&quot;free_x&quot;</span>)</span>
<span id="cb90-60"><a href="figure1.html#cb90-60" tabindex="-1"></a>p</span>
<span id="cb90-61"><a href="figure1.html#cb90-61" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/14.DA.pdf&quot;</span>,p,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">4</span>)</span>
<span id="cb90-62"><a href="figure1.html#cb90-62" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/14.DA.png&quot;</span>,p,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">4</span>)</span></code></pre></div>
<p><img src="result/figure1/14.DA.png" width="70%" style="display: block; margin: auto;" /></p>
</div>
<div id="extended-data-set-enrichment-analysis-esea" class="section level3 hasAnchor" number="6.4.3">
<h3><span class="header-section-number">6.4.3</span> Extended data set enrichment analysis (ESEA)<a href="figure1.html#extended-data-set-enrichment-analysis-esea" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="figure1.html#cb91-1" tabindex="-1"></a><span class="fu">library</span>(MNet)</span>
<span id="cb91-2"><a href="figure1.html#cb91-2" tabindex="-1"></a></span>
<span id="cb91-3"><a href="figure1.html#cb91-3" tabindex="-1"></a>all_meta <span class="ot">&lt;-</span> <span class="fu">mlimma</span>(meta_dat,group)</span>
<span id="cb91-4"><a href="figure1.html#cb91-4" tabindex="-1"></a>all_gene <span class="ot">&lt;-</span> <span class="fu">mlimma</span>(gene_dat,group)</span>
<span id="cb91-5"><a href="figure1.html#cb91-5" tabindex="-1"></a></span>
<span id="cb91-6"><a href="figure1.html#cb91-6" tabindex="-1"></a></span>
<span id="cb91-7"><a href="figure1.html#cb91-7" tabindex="-1"></a>all_result <span class="ot">&lt;-</span> <span class="fu">rbind</span>(all_meta,all_gene)</span>
<span id="cb91-8"><a href="figure1.html#cb91-8" tabindex="-1"></a></span>
<span id="cb91-9"><a href="figure1.html#cb91-9" tabindex="-1"></a>all <span class="ot">&lt;-</span> all_result<span class="sc">$</span>logFC</span>
<span id="cb91-10"><a href="figure1.html#cb91-10" tabindex="-1"></a><span class="fu">names</span>(all) <span class="ot">&lt;-</span> all_result<span class="sc">$</span>name</span>
<span id="cb91-11"><a href="figure1.html#cb91-11" tabindex="-1"></a></span>
<span id="cb91-12"><a href="figure1.html#cb91-12" tabindex="-1"></a>gene <span class="ot">&lt;-</span> all_gene<span class="sc">$</span>logFC</span>
<span id="cb91-13"><a href="figure1.html#cb91-13" tabindex="-1"></a><span class="fu">names</span>(gene) <span class="ot">&lt;-</span> all_gene<span class="sc">$</span>name</span>
<span id="cb91-14"><a href="figure1.html#cb91-14" tabindex="-1"></a></span>
<span id="cb91-15"><a href="figure1.html#cb91-15" tabindex="-1"></a>meta <span class="ot">&lt;-</span> all_meta<span class="sc">$</span>logFC</span>
<span id="cb91-16"><a href="figure1.html#cb91-16" tabindex="-1"></a><span class="fu">names</span>(meta) <span class="ot">&lt;-</span> all_meta<span class="sc">$</span>name</span>
<span id="cb91-17"><a href="figure1.html#cb91-17" tabindex="-1"></a></span>
<span id="cb91-18"><a href="figure1.html#cb91-18" tabindex="-1"></a>result_meta <span class="ot">&lt;-</span> <span class="fu">ESEA</span>(meta,<span class="at">out=</span><span class="st">&quot;metabolite&quot;</span>,<span class="at">minSize =</span> <span class="dv">3</span>)</span>
<span id="cb91-19"><a href="figure1.html#cb91-19" tabindex="-1"></a>result_gene <span class="ot">&lt;-</span> <span class="fu">ESEA</span>(gene,<span class="at">out =</span> <span class="st">&quot;gene&quot;</span>,<span class="at">minSize =</span> <span class="dv">10</span>)</span>
<span id="cb91-20"><a href="figure1.html#cb91-20" tabindex="-1"></a>result_all <span class="ot">&lt;-</span> <span class="fu">ESEA</span>(all,<span class="at">out=</span><span class="st">&quot;Extended&quot;</span>,<span class="at">minSize =</span> <span class="dv">13</span>)</span>
<span id="cb91-21"><a href="figure1.html#cb91-21" tabindex="-1"></a></span>
<span id="cb91-22"><a href="figure1.html#cb91-22" tabindex="-1"></a>pathway_msea <span class="ot">&lt;-</span> <span class="fu">rbind</span>(result_meta,result_gene,result_all) <span class="sc">%&gt;%</span></span>
<span id="cb91-23"><a href="figure1.html#cb91-23" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pval <span class="sc">&lt;</span> <span class="fl">0.01</span>) <span class="sc">%&gt;%</span></span>
<span id="cb91-24"><a href="figure1.html#cb91-24" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(pathway) <span class="sc">%&gt;%</span></span>
<span id="cb91-25"><a href="figure1.html#cb91-25" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb91-26"><a href="figure1.html#cb91-26" tabindex="-1"></a></span>
<span id="cb91-27"><a href="figure1.html#cb91-27" tabindex="-1"></a><span class="fu">length</span>(pathway_msea)</span>
<span id="cb91-28"><a href="figure1.html#cb91-28" tabindex="-1"></a></span>
<span id="cb91-29"><a href="figure1.html#cb91-29" tabindex="-1"></a>msea_meta_filter <span class="ot">&lt;-</span> result_meta <span class="sc">%&gt;%</span></span>
<span id="cb91-30"><a href="figure1.html#cb91-30" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">%in%</span> pathway_msea) <span class="sc">%&gt;%</span></span>
<span id="cb91-31"><a href="figure1.html#cb91-31" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(pathway,pval,NES) <span class="sc">%&gt;%</span></span>
<span id="cb91-32"><a href="figure1.html#cb91-32" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type=</span><span class="st">&quot;metabolite&quot;</span>)</span>
<span id="cb91-33"><a href="figure1.html#cb91-33" tabindex="-1"></a></span>
<span id="cb91-34"><a href="figure1.html#cb91-34" tabindex="-1"></a>msea_gene_filter <span class="ot">&lt;-</span> result_gene <span class="sc">%&gt;%</span></span>
<span id="cb91-35"><a href="figure1.html#cb91-35" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">%in%</span> pathway_msea) <span class="sc">%&gt;%</span></span>
<span id="cb91-36"><a href="figure1.html#cb91-36" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(pathway,pval,NES) <span class="sc">%&gt;%</span></span>
<span id="cb91-37"><a href="figure1.html#cb91-37" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb91-38"><a href="figure1.html#cb91-38" tabindex="-1"></a></span>
<span id="cb91-39"><a href="figure1.html#cb91-39" tabindex="-1"></a>msea_all_filter <span class="ot">&lt;-</span> result_all <span class="sc">%&gt;%</span></span>
<span id="cb91-40"><a href="figure1.html#cb91-40" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">%in%</span> pathway_msea) <span class="sc">%&gt;%</span></span>
<span id="cb91-41"><a href="figure1.html#cb91-41" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(pathway,pval,NES) <span class="sc">%&gt;%</span></span>
<span id="cb91-42"><a href="figure1.html#cb91-42" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type=</span><span class="st">&quot;metabolite+gene&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb91-43"><a href="figure1.html#cb91-43" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(NES)</span>
<span id="cb91-44"><a href="figure1.html#cb91-44" tabindex="-1"></a></span>
<span id="cb91-45"><a href="figure1.html#cb91-45" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">rbind</span>(msea_meta_filter,msea_gene_filter,msea_all_filter) <span class="sc">%&gt;%</span></span>
<span id="cb91-46"><a href="figure1.html#cb91-46" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pval=</span><span class="fu">ifelse</span>(pval <span class="sc">&lt;</span> <span class="fl">8.048183e-04</span>,<span class="fl">8.048183e-04</span>,pval)) <span class="sc">%&gt;%</span></span>
<span id="cb91-47"><a href="figure1.html#cb91-47" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathway=</span><span class="fu">factor</span>(pathway,<span class="at">levels=</span>msea_all_filter<span class="sc">$</span>pathway)) <span class="sc">%&gt;%</span></span>
<span id="cb91-48"><a href="figure1.html#cb91-48" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type=</span><span class="fu">factor</span>(type,<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;metabolite&quot;</span>,<span class="st">&quot;gene&quot;</span>,<span class="st">&quot;metabolite+gene&quot;</span>)))</span>
<span id="cb91-49"><a href="figure1.html#cb91-49" tabindex="-1"></a></span>
<span id="cb91-50"><a href="figure1.html#cb91-50" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(result,<span class="fu">aes</span>(pathway,NES,<span class="at">fill=</span><span class="sc">-</span><span class="fu">log10</span>(pval)))<span class="sc">+</span></span>
<span id="cb91-51"><a href="figure1.html#cb91-51" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="sc">+</span></span>
<span id="cb91-52"><a href="figure1.html#cb91-52" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb91-53"><a href="figure1.html#cb91-53" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="fu">c</span>(<span class="dv">0</span>))<span class="sc">+</span></span>
<span id="cb91-54"><a href="figure1.html#cb91-54" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb91-55"><a href="figure1.html#cb91-55" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb91-56"><a href="figure1.html#cb91-56" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">&quot;white&quot;</span>,<span class="at">high=</span><span class="st">&quot;red&quot;</span>)<span class="sc">+</span></span>
<span id="cb91-57"><a href="figure1.html#cb91-57" tabindex="-1"></a>  <span class="fu">coord_flip</span>()<span class="sc">+</span></span>
<span id="cb91-58"><a href="figure1.html#cb91-58" tabindex="-1"></a>  <span class="fu">facet_grid</span>(. <span class="sc">~</span> type,<span class="at">scales=</span><span class="st">&quot;free_x&quot;</span>)<span class="sc">+</span></span>
<span id="cb91-59"><a href="figure1.html#cb91-59" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Pathway name&quot;</span>)</span>
<span id="cb91-60"><a href="figure1.html#cb91-60" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/15.ESEA_barplot.pdf&quot;</span>,p,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb91-61"><a href="figure1.html#cb91-61" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/15.ESEA_barplot.png&quot;</span>,p,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">5</span>)</span></code></pre></div>
<p><img src="result/figure1/15.ESEA_barplot.png" width="70%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="e" class="section level2 hasAnchor" number="6.5">
<h2><span class="header-section-number">6.5</span> 1.5 (E)<a href="figure1.html#e" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="dbmnet-database" class="section level3 hasAnchor" number="6.5.1">
<h3><span class="header-section-number">6.5.1</span> dbMNet database<a href="figure1.html#dbmnet-database" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="figure1.html#cb92-1" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb92-2"><a href="figure1.html#cb92-2" tabindex="-1"></a><span class="fu">library</span>(MNet)</span>
<span id="cb92-3"><a href="figure1.html#cb92-3" tabindex="-1"></a></span>
<span id="cb92-4"><a href="figure1.html#cb92-4" tabindex="-1"></a>diff_meta <span class="ot">&lt;-</span> <span class="fu">mlimma</span>(meta_dat,group)</span>
<span id="cb92-5"><a href="figure1.html#cb92-5" tabindex="-1"></a>diff_gene <span class="ot">&lt;-</span> <span class="fu">mlimma</span>(gene_dat,group)</span>
<span id="cb92-6"><a href="figure1.html#cb92-6" tabindex="-1"></a><span class="fu">names</span>(diff_meta)[<span class="dv">4</span>]  <span class="ot">&lt;-</span> <span class="st">&quot;p_value&quot;</span></span>
<span id="cb92-7"><a href="figure1.html#cb92-7" tabindex="-1"></a><span class="fu">names</span>(diff_gene)[<span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="st">&quot;p_value&quot;</span></span>
<span id="cb92-8"><a href="figure1.html#cb92-8" tabindex="-1"></a></span>
<span id="cb92-9"><a href="figure1.html#cb92-9" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">&quot;result/figure1/16.pdent.pdf&quot;</span>,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb92-10"><a href="figure1.html#cb92-10" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">pdnet</span>(diff_meta,diff_gene,<span class="at">nsize=</span><span class="dv">60</span>)</span>
<span id="cb92-11"><a href="figure1.html#cb92-11" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb92-12"><a href="figure1.html#cb92-12" tabindex="-1"></a></span>
<span id="cb92-13"><a href="figure1.html#cb92-13" tabindex="-1"></a><span class="fu">png</span>(<span class="st">&quot;result/figure1/16.pdent.png&quot;</span>,<span class="at">width=</span><span class="dv">1000</span>,<span class="at">height =</span> <span class="dv">1000</span>)</span>
<span id="cb92-14"><a href="figure1.html#cb92-14" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">pdnet</span>(diff_meta,diff_gene,<span class="at">nsize=</span><span class="dv">60</span>)</span>
<span id="cb92-15"><a href="figure1.html#cb92-15" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb92-16"><a href="figure1.html#cb92-16" tabindex="-1"></a></span>
<span id="cb92-17"><a href="figure1.html#cb92-17" tabindex="-1"></a>node <span class="ot">&lt;-</span> a<span class="sc">$</span>node_result</span>
<span id="cb92-18"><a href="figure1.html#cb92-18" tabindex="-1"></a><span class="fu">write.table</span>(node,<span class="st">&quot;result/figure1/16.pdnet_node.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb92-19"><a href="figure1.html#cb92-19" tabindex="-1"></a></span>
<span id="cb92-20"><a href="figure1.html#cb92-20" tabindex="-1"></a>edge <span class="ot">&lt;-</span> a<span class="sc">$</span>edge_result</span>
<span id="cb92-21"><a href="figure1.html#cb92-21" tabindex="-1"></a><span class="fu">write.table</span>(edge,<span class="st">&quot;result/figure1/16.pdnet_edge.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb92-22"><a href="figure1.html#cb92-22" tabindex="-1"></a></span>
<span id="cb92-23"><a href="figure1.html#cb92-23" tabindex="-1"></a></span>
<span id="cb92-24"><a href="figure1.html#cb92-24" tabindex="-1"></a>metabolite <span class="ot">&lt;-</span> node <span class="sc">%&gt;%</span></span>
<span id="cb92-25"><a href="figure1.html#cb92-25" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(type<span class="sc">==</span><span class="st">&quot;metabolite&quot;</span>)</span>
<span id="cb92-26"><a href="figure1.html#cb92-26" tabindex="-1"></a></span>
<span id="cb92-27"><a href="figure1.html#cb92-27" tabindex="-1"></a>gene <span class="ot">&lt;-</span> node <span class="sc">%&gt;%</span></span>
<span id="cb92-28"><a href="figure1.html#cb92-28" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(type<span class="sc">==</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb92-29"><a href="figure1.html#cb92-29" tabindex="-1"></a></span>
<span id="cb92-30"><a href="figure1.html#cb92-30" tabindex="-1"></a>result_metabolite <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(metabolite<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;metabolite&quot;</span>,<span class="at">test=</span><span class="st">&quot;hypergeo&quot;</span>)</span>
<span id="cb92-31"><a href="figure1.html#cb92-31" tabindex="-1"></a>result_gene <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(gene<span class="sc">$</span>name,<span class="at">out=</span><span class="st">&quot;gene&quot;</span>,<span class="at">test=</span><span class="st">&quot;hypergeo&quot;</span>)</span>
<span id="cb92-32"><a href="figure1.html#cb92-32" tabindex="-1"></a>result_all <span class="ot">&lt;-</span> <span class="fu">PathwayAnalysis</span>(<span class="fu">c</span>(gene<span class="sc">$</span>name,metabolite<span class="sc">$</span>name),<span class="at">out=</span><span class="st">&quot;Extended&quot;</span>,<span class="at">test=</span><span class="st">&quot;hypergeo&quot;</span>)</span>
<span id="cb92-33"><a href="figure1.html#cb92-33" tabindex="-1"></a></span>
<span id="cb92-34"><a href="figure1.html#cb92-34" tabindex="-1"></a><span class="fu">write.table</span>(result_metabolite<span class="sc">$</span>output,<span class="st">&quot;result/figure1/17.pdnet_metabolite_enrichment.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names =</span> F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb92-35"><a href="figure1.html#cb92-35" tabindex="-1"></a><span class="fu">write.table</span>(result_gene<span class="sc">$</span>output,<span class="st">&quot;result/figure1/17.pdnet_gene_enrichment.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names =</span> F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb92-36"><a href="figure1.html#cb92-36" tabindex="-1"></a><span class="fu">write.table</span>(result_all<span class="sc">$</span>output,<span class="st">&quot;result/figure1/17.pdnet_all_enrichment.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names =</span> F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb92-37"><a href="figure1.html#cb92-37" tabindex="-1"></a></span>
<span id="cb92-38"><a href="figure1.html#cb92-38" tabindex="-1"></a>kegg_pathway_uniq <span class="ot">&lt;-</span> PathwayExtendData <span class="sc">%&gt;%</span></span>
<span id="cb92-39"><a href="figure1.html#cb92-39" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(kegg_pathwayname,kegg_category) <span class="sc">%&gt;%</span></span>
<span id="cb92-40"><a href="figure1.html#cb92-40" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;PATHWAY&quot;</span><span class="ot">=</span><span class="st">&quot;kegg_pathwayname&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb92-41"><a href="figure1.html#cb92-41" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;pathway_type&quot;</span><span class="ot">=</span><span class="st">&quot;kegg_category&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb92-42"><a href="figure1.html#cb92-42" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb92-43"><a href="figure1.html#cb92-43" tabindex="-1"></a></span>
<span id="cb92-44"><a href="figure1.html#cb92-44" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">rbind</span>(result_metabolite<span class="sc">$</span>output <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type=</span><span class="st">&quot;metabolite&quot;</span>),</span>
<span id="cb92-45"><a href="figure1.html#cb92-45" tabindex="-1"></a>                    result_gene<span class="sc">$</span>output <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type=</span><span class="st">&quot;gene&quot;</span>),</span>
<span id="cb92-46"><a href="figure1.html#cb92-46" tabindex="-1"></a>                    result_all<span class="sc">$</span>output <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type=</span><span class="st">&quot;all&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb92-47"><a href="figure1.html#cb92-47" tabindex="-1"></a> <span class="co"># dplyr::filter(nOverlap &gt; 1) %&gt;%</span></span>
<span id="cb92-48"><a href="figure1.html#cb92-48" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(kegg_pathway_uniq,<span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;name&quot;</span><span class="ot">=</span><span class="st">&quot;PATHWAY&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb92-49"><a href="figure1.html#cb92-49" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway_type <span class="sc">!=</span> <span class="st">&quot;Glycan biosynthesis and metabolism&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb92-50"><a href="figure1.html#cb92-50" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Fructose and mannose metabolism&quot;</span>,<span class="st">&quot;Amino sugar and nucleotide sugar metabolism&quot;</span>,</span>
<span id="cb92-51"><a href="figure1.html#cb92-51" tabindex="-1"></a>                            <span class="st">&quot;Galactose metabolism&quot;</span>,<span class="st">&quot;Ascorbate and aldarate metabolism&quot;</span>,<span class="st">&quot;Pentose and glucuronate interconversions&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb92-52"><a href="figure1.html#cb92-52" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(pvalue)</span>
<span id="cb92-53"><a href="figure1.html#cb92-53" tabindex="-1"></a></span>
<span id="cb92-54"><a href="figure1.html#cb92-54" tabindex="-1"></a>pathway_hh <span class="ot">&lt;-</span> <span class="fu">unique</span>(result<span class="sc">$</span>pathway_type)</span>
<span id="cb92-55"><a href="figure1.html#cb92-55" tabindex="-1"></a></span>
<span id="cb92-56"><a href="figure1.html#cb92-56" tabindex="-1"></a>result <span class="ot">&lt;-</span> result <span class="sc">%&gt;%</span></span>
<span id="cb92-57"><a href="figure1.html#cb92-57" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>pathway_type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Metabolism of terpenoids and polyketides&quot;</span>,<span class="st">&quot;Metabolism of cofactors and vitamins&quot;</span>,</span>
<span id="cb92-58"><a href="figure1.html#cb92-58" tabindex="-1"></a>                                     <span class="st">&quot;Biosynthesis of other secondary metabolites&quot;</span>,<span class="st">&quot;Metabolism of other amino acids&quot;</span>,</span>
<span id="cb92-59"><a href="figure1.html#cb92-59" tabindex="-1"></a>                                     <span class="st">&quot;Xenobiotics biodegradation and metabolism&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb92-60"><a href="figure1.html#cb92-60" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">match</span>(pathway_type,pathway_hh))</span>
<span id="cb92-61"><a href="figure1.html#cb92-61" tabindex="-1"></a></span>
<span id="cb92-62"><a href="figure1.html#cb92-62" tabindex="-1"></a>result<span class="sc">$</span>name <span class="ot">&lt;-</span> <span class="fu">factor</span>(result<span class="sc">$</span>name,<span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">unique</span>(result<span class="sc">$</span>name)))</span>
<span id="cb92-63"><a href="figure1.html#cb92-63" tabindex="-1"></a>result<span class="sc">$</span>pathway_type <span class="ot">&lt;-</span> <span class="fu">factor</span>(result<span class="sc">$</span>pathway_type,<span class="at">levels=</span><span class="fu">unique</span>(kegg_pathway_uniq<span class="sc">$</span>pathway_type))</span>
<span id="cb92-64"><a href="figure1.html#cb92-64" tabindex="-1"></a>result<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="fu">factor</span>(result<span class="sc">$</span>type,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;metabolite&quot;</span>,<span class="st">&quot;gene&quot;</span>,<span class="st">&quot;all&quot;</span>))</span>
<span id="cb92-65"><a href="figure1.html#cb92-65" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(result,<span class="fu">aes</span>(name,<span class="sc">-</span><span class="fu">log10</span>(pvalue)))<span class="sc">+</span></span>
<span id="cb92-66"><a href="figure1.html#cb92-66" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>,<span class="fu">aes</span>(<span class="at">fill=</span>pathway_type))<span class="sc">+</span></span>
<span id="cb92-67"><a href="figure1.html#cb92-67" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">11</span>, <span class="st">&quot;Set3&quot;</span>),<span class="at">name=</span><span class="st">&quot;Pathway Category&quot;</span>,</span>
<span id="cb92-68"><a href="figure1.html#cb92-68" tabindex="-1"></a>                    <span class="at">breaks=</span><span class="fu">unique</span>(kegg_pathway_uniq<span class="sc">$</span>pathway_type))<span class="sc">+</span></span>
<span id="cb92-69"><a href="figure1.html#cb92-69" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>))<span class="sc">+</span></span>
<span id="cb92-70"><a href="figure1.html#cb92-70" tabindex="-1"></a>  <span class="fu">coord_flip</span>()<span class="sc">+</span></span>
<span id="cb92-71"><a href="figure1.html#cb92-71" tabindex="-1"></a>  <span class="fu">facet_grid</span>(. <span class="sc">~</span> type,<span class="at">scales=</span><span class="st">&quot;free_x&quot;</span>)<span class="sc">+</span></span>
<span id="cb92-72"><a href="figure1.html#cb92-72" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb92-73"><a href="figure1.html#cb92-73" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="cn">NULL</span>)</span>
<span id="cb92-74"><a href="figure1.html#cb92-74" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/17.pdnet_epe.pdf&quot;</span>,p1,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb92-75"><a href="figure1.html#cb92-75" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;result/figure1/17.pdnet_epe.png&quot;</span>,p1,<span class="at">width=</span><span class="dv">10</span>,<span class="at">height =</span> <span class="dv">5</span>)</span></code></pre></div>
<p><img src="result/figure1/16.pdent.png" width="500" style="display: block; margin: auto;" /></p>
<p><img src="result/figure1/17.pdnet_epe.png" width="70%" style="display: block; margin: auto;" /></p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="shiny.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="figure4.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/05-figure1.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
