<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>MNet</title>
  <meta name="description" content="MNet" />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="MNet" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="MNet" />
  
  
  

<meta name="author" content="Tuantuan Gui, Xuan Liu and Hai Fang" />


<meta name="date" content="2024-02-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="database-construction.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.30/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-buttons-1.13.4/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-buttons-1.13.4/js/dataTables.buttons.min.js"></script>
<script src="libs/dt-ext-buttons-1.13.4/js/buttons.html5.min.js"></script>
<script src="libs/dt-ext-buttons-1.13.4/js/buttons.colVis.min.js"></script>
<script src="libs/dt-ext-buttons-1.13.4/js/buttons.print.min.js"></script>
<link href="libs/dt-ext-fixedcolumns-1.13.4/css/fixedColumns.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedcolumns-1.13.4/js/dataTables.fixedColumns.min.js"></script>
<link href="libs/dt-plugin-searchhighlight-1.13.4/source.css" rel="stylesheet" />
<script src="libs/dt-plugin-searchhighlight-1.13.4/jquery.highlight.js"></script>
<script src="libs/dt-plugin-searchhighlight-1.13.4/source.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction &amp; Installation</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#overview-of-workflow"><i class="fa fa-check"></i><b>1.1</b> Overview of Workflow</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.2</b> Installation</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="database-construction.html"><a href="database-construction.html"><i class="fa fa-check"></i><b>2</b> Database construction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="database-construction.html"><a href="database-construction.html#kegg-pathways-metabolite-and-gene"><i class="fa fa-check"></i><b>2.1</b> KEGG pathway’s metabolite and gene</a></li>
<li class="chapter" data-level="2.2" data-path="database-construction.html"><a href="database-construction.html#kegg-metabolite-metabolite-pairs-and-metabolite-gene-pairs"><i class="fa fa-check"></i><b>2.2</b> KEGG metabolite-metabolite pairs and metabolite-gene pairs</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="database-construction.html"><a href="database-construction.html#keggs-reaction-and-compounds-download"><i class="fa fa-check"></i><b>2.2.1</b> KEGG’s reaction and compound’s download</a></li>
<li class="chapter" data-level="2.2.2" data-path="database-construction.html"><a href="database-construction.html#downloading-the-enzyme-and-its-corresponding-gene-information-from-the-htextand-extract-the-gene-and-the-enzymes-relationship"><i class="fa fa-check"></i><b>2.2.2</b> Downloading the enzyme and its corresponding gene information from the htext,and extract the gene and the enzyme’s relationship</a></li>
<li class="chapter" data-level="2.2.3" data-path="database-construction.html"><a href="database-construction.html#extracting-the-metabolite-gene-pairs-and-metabolite-metabolite-pairs"><i class="fa fa-check"></i><b>2.2.3</b> Extracting the metabolite-gene pairs and metabolite-metabolite pairs</a></li>
<li class="chapter" data-level="2.2.4" data-path="database-construction.html"><a href="database-construction.html#conclusion"><i class="fa fa-check"></i><b>2.2.4</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="database-construction.html"><a href="database-construction.html#graphite"><i class="fa fa-check"></i><b>2.3</b> Graphite</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="database-construction.html"><a href="database-construction.html#download-all-the-data-from-graphite"><i class="fa fa-check"></i><b>2.3.1</b> Download all the data from graphite</a></li>
<li class="chapter" data-level="2.3.2" data-path="database-construction.html"><a href="database-construction.html#download-the-information-of-metabolism-pathways"><i class="fa fa-check"></i><b>2.3.2</b> Download the information of metabolism pathways</a></li>
<li class="chapter" data-level="2.3.3" data-path="database-construction.html"><a href="database-construction.html#extract-the-data-of-metabolism"><i class="fa fa-check"></i><b>2.3.3</b> Extract the data of metabolism</a></li>
<li class="chapter" data-level="2.3.4" data-path="database-construction.html"><a href="database-construction.html#combine-the-data-from-database-kegg-wikipathway-reactome-smpdb-and-then-uniq-the-data"><i class="fa fa-check"></i><b>2.3.4</b> Combine the data from database kegg, wikipathway, reactome, smpdb and then uniq the data</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="database-construction.html"><a href="database-construction.html#bigg"><i class="fa fa-check"></i><b>2.4</b> BiGG</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="database-construction.html"><a href="database-construction.html#down-all-the-models-in-the-bigg"><i class="fa fa-check"></i><b>2.4.1</b> Down all the models in the BiGG</a></li>
<li class="chapter" data-level="2.4.2" data-path="database-construction.html"><a href="database-construction.html#download-the-every-reaction"><i class="fa fa-check"></i><b>2.4.2</b> Download the every reaction</a></li>
<li class="chapter" data-level="2.4.3" data-path="database-construction.html"><a href="database-construction.html#exact-all-the-metabolites-names-in-every-model-in-bigg"><i class="fa fa-check"></i><b>2.4.3</b> Exact all the metabolites names in every model in BiGG</a></li>
<li class="chapter" data-level="2.4.4" data-path="database-construction.html"><a href="database-construction.html#the-r-script-that-change-the-metabolite-name-to-kegg-id-in-the-final-output"><i class="fa fa-check"></i><b>2.4.4</b> The R script that change the metabolite name to kegg id in the final output</a></li>
<li class="chapter" data-level="2.4.5" data-path="database-construction.html"><a href="database-construction.html#combine-the-data-from-bigg-and-graphite"><i class="fa fa-check"></i><b>2.4.5</b> Combine the data from BiGG and graphite</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html"><i class="fa fa-check"></i><b>3</b> Example-unique analyses</a>
<ul>
<li class="chapter" data-level="3.1" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#loading-the-data"><i class="fa fa-check"></i><b>3.1</b> Loading the data</a></li>
<li class="chapter" data-level="3.2" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#pathway-enrichment-analysis"><i class="fa fa-check"></i><b>3.2</b> Pathway enrichment analysis</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#gene-pathway-enrichment"><i class="fa fa-check"></i><b>3.2.1</b> gene pathway enrichment</a></li>
<li class="chapter" data-level="3.2.2" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#metabolite-pathway-enrichment"><i class="fa fa-check"></i><b>3.2.2</b> metabolite pathway enrichment</a></li>
<li class="chapter" data-level="3.2.3" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#extended-pathway-enrichment"><i class="fa fa-check"></i><b>3.2.3</b> Extended pathway enrichment</a></li>
<li class="chapter" data-level="3.2.4" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#overview-of-pathway-enrichment-analysis"><i class="fa fa-check"></i><b>3.2.4</b> Overview of pathway enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#pathway-differential-abundance-score"><i class="fa fa-check"></i><b>3.3</b> Pathway differential abundance score</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#gene-differential-abundance-score"><i class="fa fa-check"></i><b>3.3.1</b> gene differential abundance score</a></li>
<li class="chapter" data-level="3.3.2" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#metabolite-differential-abundance-score"><i class="fa fa-check"></i><b>3.3.2</b> metabolite differential abundance score</a></li>
<li class="chapter" data-level="3.3.3" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#extended-pathway-differential-abundance-score"><i class="fa fa-check"></i><b>3.3.3</b> Extended pathway differential abundance score</a></li>
<li class="chapter" data-level="3.3.4" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#overview-of-pathway-differential-abundance-score"><i class="fa fa-check"></i><b>3.3.4</b> Overview of pathway differential abundance score</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#esea-metabolite-set-enrichment-analysis"><i class="fa fa-check"></i><b>3.4</b> ESEA Metabolite Set Enrichment Analysis</a></li>
<li class="chapter" data-level="3.5" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#subnetwork"><i class="fa fa-check"></i><b>3.5</b> Subnetwork</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html"><i class="fa fa-check"></i><b>4</b> Example-rountine analyses</a>
<ul>
<li class="chapter" data-level="4.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-and-kegg-name-change"><i class="fa fa-check"></i><b>4.1</b> Metabolite and kegg name change</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-name-changed-to-refmet-name"><i class="fa fa-check"></i><b>4.1.1</b> Metabolite name changed to refmet name</a></li>
<li class="chapter" data-level="4.1.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-name-changed-to-kegg-id"><i class="fa fa-check"></i><b>4.1.2</b> Metabolite name changed to kegg id</a></li>
<li class="chapter" data-level="4.1.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-name-corresponding-to-kegg-pathway"><i class="fa fa-check"></i><b>4.1.3</b> Metabolite name corresponding to kegg pathway</a></li>
<li class="chapter" data-level="4.1.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-keggid-to-kegg-pathway"><i class="fa fa-check"></i><b>4.1.4</b> Metabolite keggid to kegg pathway</a></li>
<li class="chapter" data-level="4.1.5" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#the-pathway-information"><i class="fa fa-check"></i><b>4.1.5</b> The pathway information</a></li>
<li class="chapter" data-level="4.1.6" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathway-name-to-pathway-id"><i class="fa fa-check"></i><b>4.1.6</b> Pathway name to pathway id</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#group-wise-analyses"><i class="fa fa-check"></i><b>4.2</b> Group-wise analyses</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pca-plot"><i class="fa fa-check"></i><b>4.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="4.2.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differnetial-metabolite-analysis"><i class="fa fa-check"></i><b>4.2.2</b> Differnetial metabolite analysis</a></li>
<li class="chapter" data-level="4.2.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differential-metabolites-volcano"><i class="fa fa-check"></i><b>4.2.3</b> Differential metabolites’ volcano</a></li>
<li class="chapter" data-level="4.2.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differential-metabolites-heatmap"><i class="fa fa-check"></i><b>4.2.4</b> Differential metabolites’ heatmap</a></li>
<li class="chapter" data-level="4.2.5" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differential-metabolites-zscore"><i class="fa fa-check"></i><b>4.2.5</b> Differential metabolites’ zscore</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathway-analysis"><i class="fa fa-check"></i><b>4.3</b> Pathway analysis</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathview-only-metabolite"><i class="fa fa-check"></i><b>4.3.1</b> Pathview only metabolite</a></li>
<li class="chapter" data-level="4.3.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathview-metabolite-and-gene"><i class="fa fa-check"></i><b>4.3.2</b> Pathview metabolite and gene</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#clinical-analysis"><i class="fa fa-check"></i><b>4.4</b> Clinical analysis</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#time-series-of-clinical"><i class="fa fa-check"></i><b>4.4.1</b> Time series of clinical</a></li>
<li class="chapter" data-level="4.4.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#survival-analysis"><i class="fa fa-check"></i><b>4.4.2</b> Survival analysis</a></li>
<li class="chapter" data-level="4.4.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolites-survival-plot"><i class="fa fa-check"></i><b>4.4.3</b> Metabolites’ survival plot</a></li>
<li class="chapter" data-level="4.4.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#cox-analysis"><i class="fa fa-check"></i><b>4.4.4</b> Cox analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="shiny.html"><a href="shiny.html"><i class="fa fa-check"></i><b>5</b> Shiny</a>
<ul>
<li class="chapter" data-level="5.1" data-path="shiny.html"><a href="shiny.html#tencent-log"><i class="fa fa-check"></i><b>5.1</b> Tencent log</a></li>
<li class="chapter" data-level="5.2" data-path="shiny.html"><a href="shiny.html#shiny-app-log"><i class="fa fa-check"></i><b>5.2</b> Shiny app log</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="figure1.html"><a href="figure1.html"><i class="fa fa-check"></i><b>6</b> Figure1</a>
<ul>
<li class="chapter" data-level="6.1" data-path="figure1.html"><a href="figure1.html#a"><i class="fa fa-check"></i><b>6.1</b> 1.1 (A)</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="figure1.html"><a href="figure1.html#batch-removal"><i class="fa fa-check"></i><b>6.1.1</b> Batch removal</a></li>
<li class="chapter" data-level="6.1.2" data-path="figure1.html"><a href="figure1.html#metabolite-information"><i class="fa fa-check"></i><b>6.1.2</b> Metabolite information</a></li>
<li class="chapter" data-level="6.1.3" data-path="figure1.html"><a href="figure1.html#metabolic-pathway-information"><i class="fa fa-check"></i><b>6.1.3</b> Metabolic pathway information</a></li>
<li class="chapter" data-level="6.1.4" data-path="figure1.html"><a href="figure1.html#metabolite-and-metabolic-pathway-annotation"><i class="fa fa-check"></i><b>6.1.4</b> Metabolite and metabolic pathway annotation</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="figure1.html"><a href="figure1.html#b"><i class="fa fa-check"></i><b>6.2</b> 1.2 (B)</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="figure1.html"><a href="figure1.html#clinical-and-metabolite-analysis"><i class="fa fa-check"></i><b>6.2.1</b> Clinical and metabolite analysis</a></li>
<li class="chapter" data-level="6.2.2" data-path="figure1.html"><a href="figure1.html#time-series-analysis"><i class="fa fa-check"></i><b>6.2.2</b> Time series analysis</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="figure1.html"><a href="figure1.html#c"><i class="fa fa-check"></i><b>6.3</b> 1.3 (C)</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="figure1.html"><a href="figure1.html#dimension-reduction-visualization"><i class="fa fa-check"></i><b>6.3.1</b> Dimension reduction &amp; Visualization</a></li>
<li class="chapter" data-level="6.3.2" data-path="figure1.html"><a href="figure1.html#differential-metabolites-visualization"><i class="fa fa-check"></i><b>6.3.2</b> Differential metabolites visualization</a></li>
<li class="chapter" data-level="6.3.3" data-path="figure1.html"><a href="figure1.html#pathway-enrichment-analyses"><i class="fa fa-check"></i><b>6.3.3</b> Pathway enrichment analyses</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="figure1.html"><a href="figure1.html#d"><i class="fa fa-check"></i><b>6.4</b> 1.4 (D)</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="figure1.html"><a href="figure1.html#extended-pathway-enrichment-analysis-epea"><i class="fa fa-check"></i><b>6.4.1</b> Extended pathway enrichment analysis (EPEA)</a></li>
<li class="chapter" data-level="6.4.2" data-path="figure1.html"><a href="figure1.html#extended-differential-abundance-eda-score"><i class="fa fa-check"></i><b>6.4.2</b> Extended differential abundance (EDA) score</a></li>
<li class="chapter" data-level="6.4.3" data-path="figure1.html"><a href="figure1.html#extended-data-set-enrichment-analysis-esea"><i class="fa fa-check"></i><b>6.4.3</b> Extended data set enrichment analysis (ESEA)</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="figure1.html"><a href="figure1.html#e"><i class="fa fa-check"></i><b>6.5</b> 1.5 (E)</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="figure1.html"><a href="figure1.html#dbmnet-database"><i class="fa fa-check"></i><b>6.5.1</b> dbMNet database</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="figure4.html"><a href="figure4.html"><i class="fa fa-check"></i><b>7</b> Figure4</a>
<ul>
<li class="chapter" data-level="7.1" data-path="figure4.html"><a href="figure4.html#load-data"><i class="fa fa-check"></i><b>7.1</b> Load data</a></li>
<li class="chapter" data-level="7.2" data-path="figure4.html"><a href="figure4.html#mlimma-analysis"><i class="fa fa-check"></i><b>7.2</b> mlimma analysis</a></li>
<li class="chapter" data-level="7.3" data-path="figure4.html"><a href="figure4.html#epe-analysis"><i class="fa fa-check"></i><b>7.3</b> EPE analysis</a></li>
<li class="chapter" data-level="7.4" data-path="figure4.html"><a href="figure4.html#eda-analysis"><i class="fa fa-check"></i><b>7.4</b> EDA analysis</a></li>
<li class="chapter" data-level="7.5" data-path="figure4.html"><a href="figure4.html#esea-analysis"><i class="fa fa-check"></i><b>7.5</b> ESEA analysis</a></li>
<li class="chapter" data-level="7.6" data-path="figure4.html"><a href="figure4.html#aml-epe-analysis"><i class="fa fa-check"></i><b>7.6</b> AML EPE analysis</a></li>
<li class="chapter" data-level="7.7" data-path="figure4.html"><a href="figure4.html#volcano-plot-1"><i class="fa fa-check"></i><b>7.7</b> Volcano plot</a>
<ul>
<li class="chapter" data-level="7.7.1" data-path="figure4.html"><a href="figure4.html#differential-metabolites-and-genes"><i class="fa fa-check"></i><b>7.7.1</b> Differential metabolites and genes</a></li>
<li class="chapter" data-level="7.7.2" data-path="figure4.html"><a href="figure4.html#all"><i class="fa fa-check"></i><b>7.7.2</b> ALL</a></li>
<li class="chapter" data-level="7.7.3" data-path="figure4.html"><a href="figure4.html#aml"><i class="fa fa-check"></i><b>7.7.3</b> AML</a></li>
</ul></li>
<li class="chapter" data-level="7.8" data-path="figure4.html"><a href="figure4.html#heatmap-1"><i class="fa fa-check"></i><b>7.8</b> Heatmap</a></li>
<li class="chapter" data-level="7.9" data-path="figure4.html"><a href="figure4.html#subnetwork-1"><i class="fa fa-check"></i><b>7.9</b> Subnetwork</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>8</b> Session information</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">MNet</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">MNet</h1>
<p class="author"><em>Tuantuan Gui, Xuan Liu and Hai Fang</em></p>
<p class="date"><em>2024-02-01</em></p>
</div>
<div id="introduction-installation" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> Introduction &amp; Installation<a href="index.html#introduction-installation" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The surge in metabolomics and transcriptomic data has necessitated the development of innovative network analysis tools for the integrated analysis of multi-omics data. However, current tools often focus on individual pathways, extracting the subnetwork for the interested genes and metabolites or extracting the correlation subnetwork, neglecting the valuable information embedded in gene-metabolite or metabolite-metabolite knowledge networks across all metabolic pathways. In response to this gap, we introduce an integrated analytical tool MNet, an R package for mining metabolomics coupled with clinical and transcriptomic data. MNet introduces three key feature to enhance the analytical capabilities. Firstly, MNet incorporates a built-in knowledgebase called ‘dbMNet’ that has been curated over time by experts. It represents a unique reference resource for metabolomics, which includes all currently available gene-metabolite pairs in humans, as well as metabolite-metabolite pairs. Secondly, MNet facilitates extended pathway enrichment analysis (EPEA), extended differential abundance analysis (EDA) and extended data sets enrichment analysis (ESEA). These functionalities enable the identification of dysregulated metabolic pathways by simultaneously considering both metabolites and genes. Thirdly, based on ‘dbMNet’ knowledgebase, MNet supports metabolic-related network analysis, considering both gene-metabolite and metabolite-metabolite pairs. Building upon case presentations, MNet emerges as a pivotal tool, demonstrating its efficacy in delivering a comprehensive grasp of metabolic dysregulation. It stands out as an invaluable resource, empowering researchers with the capability to conduct thorough and insightful multi-omics investigations.</p>
<div id="overview-of-workflow" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Overview of Workflow<a href="index.html#overview-of-workflow" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The package MNet is designed to provide a collection of functionalities aiding in metabolome data interpretation together with clinical data and gene expression data. It not only supports the routine analyses and visualisations in terms of metabolomic datasets (including but not limited to: differential metabolites analyses/visualisations/feature selection), the extended pathway enrichment analyses which is designed to detect which pathways are differentially abundant between two conditions, but also is capable of carrying out integrated analysis of metabolic and gene expression data (the metabolite and gene pathway analyses and the subnetwork analyses of metabolite-gene network analyses).</p>
<p><img src="fig1-v0126-gtt-v2.png" width="120%" style="display: block; margin: auto;" /></p>
</div>
<div id="installation" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Installation<a href="index.html#installation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The R package MNet requires R version 4.0.0 or higher, please see <a href="https://cran.r-project.org" class="uri">https://cran.r-project.org</a>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="index.html#cb1-1" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(devtools)){</span>
<span id="cb1-2"><a href="index.html#cb1-2" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb1-3"><a href="index.html#cb1-3" tabindex="-1"></a>}</span></code></pre></div>
<p>MNet is available for all operating systems and can be installed via the Github.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="index.html#cb2-1" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;tuantuangui/MNet&quot;</span>)</span>
<span id="cb2-2"><a href="index.html#cb2-2" tabindex="-1"></a><span class="fu">library</span>(MNet)</span></code></pre></div>
<p>check the package ‘MNet’ successfully installed</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="index.html#cb3-1" tabindex="-1"></a><span class="fu">library</span>(<span class="at">help=</span>MNet)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="database-construction.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
