<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Session information | MNet</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.28 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Session information | MNet" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Session information | MNet" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Tuantuan Gui and Hai Fang" />


<meta name="date" content="2022-11-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="example-rountinely-analyses.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.24/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-buttons-1.11.3/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-buttons-1.11.3/js/dataTables.buttons.min.js"></script>
<script src="libs/dt-ext-buttons-1.11.3/js/buttons.html5.min.js"></script>
<script src="libs/dt-ext-buttons-1.11.3/js/buttons.colVis.min.js"></script>
<script src="libs/dt-ext-buttons-1.11.3/js/buttons.print.min.js"></script>
<link href="libs/dt-ext-fixedcolumns-1.11.3/css/fixedColumns.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedcolumns-1.11.3/js/dataTables.fixedColumns.min.js"></script>
<link href="libs/dt-plugin-searchhighlight-1.11.3/source.css" rel="stylesheet" />
<script src="libs/dt-plugin-searchhighlight-1.11.3/jquery.highlight.js"></script>
<script src="libs/dt-plugin-searchhighlight-1.11.3/source.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="quick-start.html"><a href="quick-start.html"><i class="fa fa-check"></i><b>2</b> Quick Start</a>
<ul>
<li class="chapter" data-level="2.1" data-path="quick-start.html"><a href="quick-start.html#overview-of-workflow"><i class="fa fa-check"></i><b>2.1</b> Overview of Workflow</a></li>
<li class="chapter" data-level="2.2" data-path="quick-start.html"><a href="quick-start.html#installation"><i class="fa fa-check"></i><b>2.2</b> Installation</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="example-uniquely-analyses.html"><a href="example-uniquely-analyses.html"><i class="fa fa-check"></i><b>3</b> Example-uniquely analyses</a>
<ul>
<li class="chapter" data-level="3.1" data-path="example-uniquely-analyses.html"><a href="example-uniquely-analyses.html#loading-the-data"><i class="fa fa-check"></i><b>3.1</b> Loading the data</a></li>
<li class="chapter" data-level="3.2" data-path="example-uniquely-analyses.html"><a href="example-uniquely-analyses.html#integrated-analyses"><i class="fa fa-check"></i><b>3.2</b> Integrated analyses</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="example-uniquely-analyses.html"><a href="example-uniquely-analyses.html#extended-pathway-analysis"><i class="fa fa-check"></i><b>3.2.1</b> Extended pathway analysis</a></li>
<li class="chapter" data-level="3.2.2" data-path="example-uniquely-analyses.html"><a href="example-uniquely-analyses.html#extended-da-score"><i class="fa fa-check"></i><b>3.2.2</b> Extended DA score</a></li>
<li class="chapter" data-level="3.2.3" data-path="example-uniquely-analyses.html"><a href="example-uniquely-analyses.html#all-network-in-our-database"><i class="fa fa-check"></i><b>3.2.3</b> All network in our database</a></li>
<li class="chapter" data-level="3.2.4" data-path="example-uniquely-analyses.html"><a href="example-uniquely-analyses.html#subnetwork-analysis-of-metabolism-and-transcriptome-data-from-tnbc"><i class="fa fa-check"></i><b>3.2.4</b> Subnetwork analysis of Metabolism and transcriptome data from TNBC:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html"><i class="fa fa-check"></i><b>4</b> Example-rountinely analyses</a>
<ul>
<li class="chapter" data-level="4.1" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#metabolite-name-change"><i class="fa fa-check"></i><b>4.1</b> Metabolite name change</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#metabolite-name-changed-to-refmet-name"><i class="fa fa-check"></i><b>4.1.1</b> Metabolite name changed to refmet name</a></li>
<li class="chapter" data-level="4.1.2" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#metabolite-name-changed-to-kegg-id"><i class="fa fa-check"></i><b>4.1.2</b> Metabolite name changed to kegg id</a></li>
<li class="chapter" data-level="4.1.3" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#metabolite-name-corresponding-to-kegg-pathway"><i class="fa fa-check"></i><b>4.1.3</b> Metabolite name corresponding to kegg pathway</a></li>
<li class="chapter" data-level="4.1.4" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#metabolite-keggid-to-kegg-pathway"><i class="fa fa-check"></i><b>4.1.4</b> Metabolite keggid to kegg pathway</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#group-wise-analyses"><i class="fa fa-check"></i><b>4.2</b> Group-wise analyses</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#pca-plot"><i class="fa fa-check"></i><b>4.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="4.2.2" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#differnetial-metabolite-analysis"><i class="fa fa-check"></i><b>4.2.2</b> Differnetial metabolite analysis</a></li>
<li class="chapter" data-level="4.2.3" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#differential-metabolites-volcano"><i class="fa fa-check"></i><b>4.2.3</b> Differential metabolites’ volcano</a></li>
<li class="chapter" data-level="4.2.4" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#differential-metabolites-heatmap"><i class="fa fa-check"></i><b>4.2.4</b> Differential metabolites’ heatmap</a></li>
<li class="chapter" data-level="4.2.5" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#differential-metabolites-zscore"><i class="fa fa-check"></i><b>4.2.5</b> Differential metabolites’ zscore</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#feature-selection"><i class="fa fa-check"></i><b>4.3</b> Feature selection</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#boruta"><i class="fa fa-check"></i><b>4.3.1</b> Boruta</a></li>
<li class="chapter" data-level="4.3.2" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#random-forest"><i class="fa fa-check"></i><b>4.3.2</b> Random Forest</a></li>
<li class="chapter" data-level="4.3.3" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#xgboost"><i class="fa fa-check"></i><b>4.3.3</b> XGBoost</a></li>
<li class="chapter" data-level="4.3.4" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#lasso"><i class="fa fa-check"></i><b>4.3.4</b> LASSO</a></li>
<li class="chapter" data-level="4.3.5" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#elastic-network"><i class="fa fa-check"></i><b>4.3.5</b> elastic network</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#pathway-analysis"><i class="fa fa-check"></i><b>4.4</b> Pathway analysis</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#pathview-only-metabolite"><i class="fa fa-check"></i><b>4.4.1</b> pathview only metabolite</a></li>
<li class="chapter" data-level="4.4.2" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#pathview-metabolite-and-gene"><i class="fa fa-check"></i><b>4.4.2</b> pathview metabolite and gene</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#time-series-analysis"><i class="fa fa-check"></i><b>4.5</b> Time series analysis</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#time-series-using-mfuzz"><i class="fa fa-check"></i><b>4.5.1</b> time series using mFuzz</a></li>
<li class="chapter" data-level="4.5.2" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#time-series-using-suprahex"><i class="fa fa-check"></i><b>4.5.2</b> time series using supraHex</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#clinical-analysis"><i class="fa fa-check"></i><b>4.6</b> Clinical analysis</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#time-series-of-clinical"><i class="fa fa-check"></i><b>4.6.1</b> Time series of clinical</a></li>
<li class="chapter" data-level="4.6.2" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#correlation-between-clinical-and-metabolites-using-mantel-test"><i class="fa fa-check"></i><b>4.6.2</b> Correlation between clinical and metabolites using mantel test</a></li>
<li class="chapter" data-level="4.6.3" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#survival-analysis"><i class="fa fa-check"></i><b>4.6.3</b> Survival analysis</a></li>
<li class="chapter" data-level="4.6.4" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#metabolites-survival-plot"><i class="fa fa-check"></i><b>4.6.4</b> Metabolites’ survival plot</a></li>
<li class="chapter" data-level="4.6.5" data-path="example-rountinely-analyses.html"><a href="example-rountinely-analyses.html#cox-analysis"><i class="fa fa-check"></i><b>4.6.5</b> Cox analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>5</b> Session information</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">MNet</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="session-information" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Session information<a href="session-information.html#session-information" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="session-information.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.2.1 (2022-06-23)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Catalina 10.15.6
## 
## Matrix products: default
## BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] tcltk     stats     graphics  grDevices utils    
## [6] datasets  methods   base     
## 
## other attached packages:
##  [1] WGCNA_1.71            fastcluster_1.2.3    
##  [3] dynamicTreeCut_1.63-1 e1071_1.7-11         
##  [5] Biobase_2.56.0        BiocGenerics_0.42.0  
##  [7] DynDoc_1.74.0         widgetTools_1.74.0   
##  [9] caret_6.0-93          lattice_0.20-45      
## [11] ggplot2_3.3.6         dplyr_1.0.9          
## [13] MNet_0.1.0           
## 
## loaded via a namespace (and not attached):
##   [1] utf8_1.2.2             tidyselect_1.1.2      
##   [3] RSQLite_2.2.15         AnnotationDbi_1.58.0  
##   [5] htmlwidgets_1.5.4      grid_4.2.1            
##   [7] ranger_0.14.1          pROC_1.18.0           
##   [9] XGR_1.1.8              munsell_0.5.0         
##  [11] preprocessCore_1.58.0  codetools_0.2-18      
##  [13] ragg_1.2.2             interp_1.1-3          
##  [15] xgboost_1.6.0.1        DT_0.24               
##  [17] future_1.27.0          withr_2.5.0           
##  [19] colorspace_2.0-3       Boruta_7.0.0          
##  [21] knitr_1.39             supraHex_1.34.0       
##  [23] rstudioapi_0.13        stats4_4.2.1          
##  [25] ggsignif_0.6.3         listenv_0.8.0         
##  [27] labeling_0.4.2         GenomeInfoDbData_1.2.8
##  [29] KMsurv_0.1-5           bit64_4.0.5           
##  [31] farver_2.1.1           parallelly_1.32.1     
##  [33] vctrs_0.4.1            generics_0.1.3        
##  [35] ipred_0.9-13           xfun_0.32             
##  [37] randomForest_4.7-1.1   R6_2.5.1              
##  [39] doParallel_1.0.17      GenomeInfoDb_1.32.3   
##  [41] bitops_1.0-7           cachem_1.0.6          
##  [43] assertthat_0.2.1       scales_1.2.0          
##  [45] vroom_1.5.7            nnet_7.3-17           
##  [47] gtable_0.3.0           globals_0.16.0        
##  [49] tidygraph_1.2.1        timeDate_4021.104     
##  [51] rlang_1.0.4            systemfonts_1.0.4     
##  [53] splines_4.2.1          rstatix_0.7.0         
##  [55] impute_1.70.0          ModelMetrics_1.2.2.2  
##  [57] hexbin_1.28.2          checkmate_2.1.0       
##  [59] broom_1.0.0            yaml_2.3.5            
##  [61] reshape2_1.4.4         abind_1.4-5           
##  [63] ggnetwork_0.5.10       crosstalk_1.2.0       
##  [65] backports_1.4.1        Hmisc_4.7-0           
##  [67] tools_4.2.1            lava_1.6.10           
##  [69] bookdown_0.28          ellipsis_0.3.2        
##  [71] jquerylib_0.1.4        RColorBrewer_1.1-3    
##  [73] proxy_0.4-27           ggcor_0.9.8.1         
##  [75] Rcpp_1.0.9             plyr_1.8.7            
##  [77] base64enc_0.1-3        zlibbioc_1.42.0       
##  [79] purrr_0.3.4            RCurl_1.98-1.8        
##  [81] ggpubr_0.4.0           rpart_4.1.16          
##  [83] deldir_1.0-6           S4Vectors_0.34.0      
##  [85] zoo_1.8-10             ggrepel_0.9.1         
##  [87] cluster_2.1.3          fs_1.5.2              
##  [89] crul_1.2.0             magrittr_2.0.3        
##  [91] data.table_1.14.2      dnet_1.1.7            
##  [93] survminer_0.4.9        ggnewscale_0.4.7      
##  [95] matrixStats_0.62.0     hms_1.1.1             
##  [97] patchwork_1.1.1        evaluate_0.16         
##  [99] xtable_1.8-4           jpeg_0.1-9            
## [101] readxl_1.4.0           IRanges_2.30.0        
## [103] gridExtra_2.3          shape_1.4.6           
## [105] compiler_4.2.1         tibble_3.1.8          
## [107] crayon_1.5.1           htmltools_0.5.3       
## [109] mgcv_1.8-40            tzdb_0.3.0            
## [111] Formula_1.2-4          tidyr_1.2.0           
## [113] lubridate_1.8.0        DBI_1.1.3             
## [115] MASS_7.3-58.1          Matrix_1.4-1          
## [117] car_3.1-0              readr_2.1.2           
## [119] permute_0.9-7          cli_3.3.0             
## [121] parallel_4.2.1         gower_1.0.0           
## [123] igraph_1.3.4           GenomicRanges_1.48.0  
## [125] pkgconfig_2.0.3        km.ci_0.5-6           
## [127] RCircos_1.2.2          osfr_0.2.8            
## [129] foreign_0.8-82         recipes_1.0.1         
## [131] foreach_1.5.2          tkWidgets_1.74.0      
## [133] bslib_0.4.0            hardhat_1.2.0         
## [135] XVector_0.36.0         prodlim_2019.11.13    
## [137] stringr_1.4.0          digest_0.6.29         
## [139] graph_1.74.0           vegan_2.6-2           
## [141] httpcode_0.3.0         Biostrings_2.64.0     
## [143] rmarkdown_2.14         cellranger_1.1.0      
## [145] survMisc_0.5.6         htmlTable_2.4.1       
## [147] curl_4.3.2             lifecycle_1.0.1       
## [149] nlme_3.1-157           jsonlite_1.8.0        
## [151] carData_3.0-5          fansi_1.0.3           
## [153] pillar_1.8.0           KEGGREST_1.36.3       
## [155] fastmap_1.1.0          httr_1.4.3            
## [157] survival_3.4-0         GO.db_3.15.0          
## [159] glue_1.6.2             png_0.1-7             
## [161] iterators_1.0.14       Rgraphviz_2.40.0      
## [163] glmnet_4.1-4           bit_4.0.4             
## [165] class_7.3-20           stringi_1.7.8         
## [167] sass_0.4.2             blob_1.2.3            
## [169] textshaping_0.3.6      Mfuzz_2.56.0          
## [171] latticeExtra_0.6-30    memoise_2.0.1         
## [173] future.apply_1.9.0     ape_5.6-2</code></pre>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="example-rountinely-analyses.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/03-example_routine.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
