<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Session information | MNet</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.28 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Session information | MNet" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Session information | MNet" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Tuantuan Gui and Hai Fang" />


<meta name="date" content="2023-02-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="example-rountine-analyses.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.24/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-buttons-1.11.3/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-buttons-1.11.3/js/dataTables.buttons.min.js"></script>
<script src="libs/dt-ext-buttons-1.11.3/js/buttons.html5.min.js"></script>
<script src="libs/dt-ext-buttons-1.11.3/js/buttons.colVis.min.js"></script>
<script src="libs/dt-ext-buttons-1.11.3/js/buttons.print.min.js"></script>
<link href="libs/dt-ext-fixedcolumns-1.11.3/css/fixedColumns.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedcolumns-1.11.3/js/dataTables.fixedColumns.min.js"></script>
<link href="libs/dt-plugin-searchhighlight-1.11.3/source.css" rel="stylesheet" />
<script src="libs/dt-plugin-searchhighlight-1.11.3/jquery.highlight.js"></script>
<script src="libs/dt-plugin-searchhighlight-1.11.3/source.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction &amp; Installation</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#overview-of-workflow"><i class="fa fa-check"></i><b>1.1</b> Overview of Workflow</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.2</b> Installation</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html"><i class="fa fa-check"></i><b>2</b> Example-unique analyses</a>
<ul>
<li class="chapter" data-level="2.1" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#loading-the-data"><i class="fa fa-check"></i><b>2.1</b> Loading the data</a></li>
<li class="chapter" data-level="2.2" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#pathway-enrichment-analysis"><i class="fa fa-check"></i><b>2.2</b> Pathway enrichment analysis</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#extended-pathway-enrichment"><i class="fa fa-check"></i><b>2.2.1</b> Extended pathway enrichment</a></li>
<li class="chapter" data-level="2.2.2" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#metabolite-pathway-enrichment"><i class="fa fa-check"></i><b>2.2.2</b> metabolite pathway enrichment</a></li>
<li class="chapter" data-level="2.2.3" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#gene-pathway-enrichment"><i class="fa fa-check"></i><b>2.2.3</b> gene pathway enrichment</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#pathway-differential-abundance-score"><i class="fa fa-check"></i><b>2.3</b> Pathway differential abundance score</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#extended-pathway-differential-abundance-score"><i class="fa fa-check"></i><b>2.3.1</b> Extended pathway differential abundance score</a></li>
<li class="chapter" data-level="2.3.2" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#metabolite-differential-abundance-score"><i class="fa fa-check"></i><b>2.3.2</b> metabolite differential abundance score</a></li>
<li class="chapter" data-level="2.3.3" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#gene-differential-abundance-score"><i class="fa fa-check"></i><b>2.3.3</b> gene differential abundance score</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#subnetwork-analysis-of-metabolism-and-transcriptome-data-from-tnbc"><i class="fa fa-check"></i><b>2.4</b> Subnetwork analysis of Metabolism and transcriptome data from TNBC:</a></li>
<li class="chapter" data-level="2.5" data-path="example-unique-analyses.html"><a href="example-unique-analyses.html#subnetwork-analysis-of-interested-metabolites-and-genes"><i class="fa fa-check"></i><b>2.5</b> Subnetwork analysis of interested metabolites and genes</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html"><i class="fa fa-check"></i><b>3</b> Example-rountine analyses</a>
<ul>
<li class="chapter" data-level="3.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-name-change"><i class="fa fa-check"></i><b>3.1</b> Metabolite name change</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-name-changed-to-refmet-name"><i class="fa fa-check"></i><b>3.1.1</b> Metabolite name changed to refmet name</a></li>
<li class="chapter" data-level="3.1.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-name-changed-to-kegg-id"><i class="fa fa-check"></i><b>3.1.2</b> Metabolite name changed to kegg id</a></li>
<li class="chapter" data-level="3.1.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-name-corresponding-to-kegg-pathway"><i class="fa fa-check"></i><b>3.1.3</b> Metabolite name corresponding to kegg pathway</a></li>
<li class="chapter" data-level="3.1.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolite-keggid-to-kegg-pathway"><i class="fa fa-check"></i><b>3.1.4</b> Metabolite keggid to kegg pathway</a></li>
<li class="chapter" data-level="3.1.5" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#the-pathway-information"><i class="fa fa-check"></i><b>3.1.5</b> the pathway information</a></li>
<li class="chapter" data-level="3.1.6" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathway-name-to-pathway-id"><i class="fa fa-check"></i><b>3.1.6</b> pathway name to pathway id</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#group-wise-analyses"><i class="fa fa-check"></i><b>3.2</b> Group-wise analyses</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pca-plot"><i class="fa fa-check"></i><b>3.2.1</b> PCA plot</a></li>
<li class="chapter" data-level="3.2.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differnetial-metabolite-analysis"><i class="fa fa-check"></i><b>3.2.2</b> Differnetial metabolite analysis</a></li>
<li class="chapter" data-level="3.2.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differential-metabolites-volcano"><i class="fa fa-check"></i><b>3.2.3</b> Differential metabolites’ volcano</a></li>
<li class="chapter" data-level="3.2.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differential-metabolites-heatmap"><i class="fa fa-check"></i><b>3.2.4</b> Differential metabolites’ heatmap</a></li>
<li class="chapter" data-level="3.2.5" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#differential-metabolites-zscore"><i class="fa fa-check"></i><b>3.2.5</b> Differential metabolites’ zscore</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#feature-selection"><i class="fa fa-check"></i><b>3.3</b> Feature selection</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#boruta"><i class="fa fa-check"></i><b>3.3.1</b> Boruta</a></li>
<li class="chapter" data-level="3.3.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#random-forest"><i class="fa fa-check"></i><b>3.3.2</b> Random Forest</a></li>
<li class="chapter" data-level="3.3.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#xgboost"><i class="fa fa-check"></i><b>3.3.3</b> XGBoost</a></li>
<li class="chapter" data-level="3.3.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#lasso"><i class="fa fa-check"></i><b>3.3.4</b> LASSO</a></li>
<li class="chapter" data-level="3.3.5" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#elastic-network"><i class="fa fa-check"></i><b>3.3.5</b> elastic network</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathway-analysis"><i class="fa fa-check"></i><b>3.4</b> Pathway analysis</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathview-only-metabolite"><i class="fa fa-check"></i><b>3.4.1</b> pathview only metabolite</a></li>
<li class="chapter" data-level="3.4.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#pathview-metabolite-and-gene"><i class="fa fa-check"></i><b>3.4.2</b> pathview metabolite and gene</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#clinical-analysis"><i class="fa fa-check"></i><b>3.5</b> Clinical analysis</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#time-series-of-clinical"><i class="fa fa-check"></i><b>3.5.1</b> Time series of clinical</a></li>
<li class="chapter" data-level="3.5.2" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#correlation-between-clinical-and-metabolites-using-mantel-test"><i class="fa fa-check"></i><b>3.5.2</b> Correlation between clinical and metabolites using mantel test</a></li>
<li class="chapter" data-level="3.5.3" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#survival-analysis"><i class="fa fa-check"></i><b>3.5.3</b> Survival analysis</a></li>
<li class="chapter" data-level="3.5.4" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#metabolites-survival-plot"><i class="fa fa-check"></i><b>3.5.4</b> Metabolites’ survival plot</a></li>
<li class="chapter" data-level="3.5.5" data-path="example-rountine-analyses.html"><a href="example-rountine-analyses.html#cox-analysis"><i class="fa fa-check"></i><b>3.5.5</b> Cox analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="session-information.html"><a href="session-information.html"><i class="fa fa-check"></i><b>4</b> Session information</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">MNet</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="session-information" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> Session information<a href="session-information.html#session-information" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="session-information.html#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.2.1 (2022-06-23)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Catalina 10.15.6
## 
## Matrix products: default
## BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] dplyr_1.0.9 knitr_1.39  MNet_0.1.0 
## 
## loaded via a namespace (and not attached):
##   [1] readxl_1.4.0           shadowtext_0.1.2       backports_1.4.1       
##   [4] fastmatch_1.1-3        RCircos_1.2.2          plyr_1.8.7            
##   [7] igraph_1.3.4           lazyeval_0.2.2         splines_4.2.1         
##  [10] crosstalk_1.2.0        BiocParallel_1.30.3    pathview_1.36.0       
##  [13] ggnetwork_0.5.10       GenomeInfoDb_1.32.3    ggplot2_3.3.6         
##  [16] digest_0.6.29          yulab.utils_0.0.5      htmltools_0.5.3       
##  [19] GOSemSim_2.22.0        viridis_0.6.2          GO.db_3.15.0          
##  [22] fansi_1.0.3            magrittr_2.0.3         memoise_2.0.1         
##  [25] strex_1.6.0            cluster_2.1.3          tzdb_0.3.0            
##  [28] limma_3.52.2           readr_2.1.2            Biostrings_2.64.0     
##  [31] graphlayouts_0.8.0     enrichplot_1.16.1      colorspace_2.0-3      
##  [34] blob_1.2.3             ggrepel_0.9.1          xfun_0.32             
##  [37] hexbin_1.28.2          crayon_1.5.1           RCurl_1.98-1.8        
##  [40] jsonlite_1.8.0         graph_1.74.0           scatterpie_0.1.7      
##  [43] supraHex_1.34.0        osfr_0.2.8             ape_5.6-2             
##  [46] glue_1.6.2             polyclip_1.10-0        gtable_0.3.0          
##  [49] zlibbioc_1.42.0        XVector_0.36.0         UpSetR_1.4.0          
##  [52] car_3.1-0              Rgraphviz_2.40.0       BiocGenerics_0.42.0   
##  [55] abind_1.4-5            scales_1.2.0           DOSE_3.22.0           
##  [58] DBI_1.1.3              rstatix_0.7.0          Rcpp_1.0.9            
##  [61] viridisLite_0.4.0      gridGraphics_0.5-1     tidytree_0.4.0        
##  [64] bit_4.0.4              proxy_0.4-27           stats4_4.2.1          
##  [67] DT_0.24                htmlwidgets_1.5.4      httr_1.4.3            
##  [70] fgsea_1.22.0           RColorBrewer_1.1-3     ellipsis_0.3.2        
##  [73] pkgconfig_2.0.3        XML_3.99-0.10          farver_2.1.1          
##  [76] sass_0.4.2             crul_1.2.0             utf8_1.2.2            
##  [79] ggplotify_0.1.0        tidyselect_1.1.2       rlang_1.0.6           
##  [82] reshape2_1.4.4         AnnotationDbi_1.58.0   munsell_0.5.0         
##  [85] cellranger_1.1.0       tools_4.2.1            cachem_1.0.6          
##  [88] downloader_0.4         cli_3.6.0              generics_0.1.3        
##  [91] RSQLite_2.2.15         broom_1.0.2            evaluate_0.16         
##  [94] stringr_1.5.0          fastmap_1.1.0          yaml_2.3.5            
##  [97] ggtree_3.4.2           fs_1.5.2               org.Hs.eg.db_3.15.0   
## [100] bit64_4.0.5            tidygraph_1.2.1        purrr_0.3.4           
## [103] KEGGREST_1.36.3        ggraph_2.0.6           nlme_3.1-157          
## [106] filesstrings_3.2.4     KEGGgraph_1.56.0       aplot_0.1.6           
## [109] DO.db_2.9              compiler_4.2.1         rstudioapi_0.13       
## [112] curl_4.3.2             png_0.1-7              e1071_1.7-11          
## [115] treeio_1.20.2          tibble_3.1.8           tweenr_1.0.2          
## [118] bslib_0.4.0            stringi_1.7.8          highr_0.9             
## [121] lattice_0.20-45        Matrix_1.4-1           vegan_2.6-2           
## [124] permute_0.9-7          vctrs_0.5.1            pillar_1.8.0          
## [127] lifecycle_1.0.3        jquerylib_0.1.4        data.table_1.14.2     
## [130] cowplot_1.1.1          bitops_1.0-7           dnet_1.1.7            
## [133] GenomicRanges_1.48.0   patchwork_1.1.1        qvalue_2.28.0         
## [136] R6_2.5.1               XGR_1.1.8              bookdown_0.28         
## [139] gridExtra_2.3          IRanges_2.30.0         codetools_0.2-18      
## [142] MASS_7.3-58.1          assertthat_0.2.1       httpcode_0.3.0        
## [145] S4Vectors_0.34.0       GenomeInfoDbData_1.2.8 hms_1.1.1             
## [148] mgcv_1.8-40            parallel_4.2.1         clusterProfiler_4.4.4 
## [151] grid_4.2.1             ggfun_0.0.6            tidyr_1.2.0           
## [154] class_7.3-20           rmarkdown_2.14         carData_3.0-5         
## [157] ggforce_0.3.3          Biobase_2.56.0</code></pre>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="example-rountine-analyses.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/03-example_routine.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
